// element-angular/__ivy_ngcc__/release/switch/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElSwitch } from './switch';
import { FormsModule } from '@angular/forms';
import * as ɵngcc0 from '@angular/core';
var ElSwitchModule = /** @class */ (function () {
    function ElSwitchModule() {
    }
    /**
     * @return {?}
     */
    ElSwitchModule.forRoot = function () {
        return { ngModule: ElSwitchModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElSwitchModule.ctorParameters = function () { return []; };
ElSwitchModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElSwitchModule });
ElSwitchModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElSwitchModule_Factory(t) { return new (t || ElSwitchModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElSwitchModule, { declarations: function () { return [ElSwitch]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [ElSwitch]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElSwitchModule, [{
        type: NgModule,
        args: [{
                declarations: [ElSwitch],
                exports: [ElSwitch],
                imports: [CommonModule, FormsModule],
                entryComponents: [ElSwitch]
            }]
    }], function () { return []; }, null); })();
    return ElSwitchModule;
}());
export { ElSwitchModule };
function ElSwitchModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElSwitchModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElSwitchModule.ctorParameters;
}
