// element-angular/__ivy_ngcc__/release/form/form.props.js
import { Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var ElFormProps = /** @class */ (function () {
    function ElFormProps() {
        this.inline = false;
        this.showIcon = false;
        this.showMessage = false;
        this.inlineMessage = false;
        // right / left / top
        this.labelPosition = 'right';
    }
    ElFormProps.propDecorators = {
        'inline': [{ type: Input },],
        'showIcon': [{ type: Input, args: ['show-icon',] },],
        'showMessage': [{ type: Input, args: ['show-message',] },],
        'inlineMessage': [{ type: Input, args: ['inline-message',] },],
        'size': [{ type: Input, args: ['size',] },],
        'labelPosition': [{ type: Input, args: ['label-position',] },],
        'labelWidth': [{ type: Input, args: ['label-width',] },],
        'labelSuffix': [{ type: Input, args: ['label-suffix',] },],
    };
ElFormProps.ɵfac = function ElFormProps_Factory(t) { return new (t || ElFormProps)(); };
ElFormProps.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElFormProps, inputs: { inline: "inline", showIcon: ["show-icon", "showIcon"], showMessage: ["show-message", "showMessage"], inlineMessage: ["inline-message", "inlineMessage"], labelPosition: ["label-position", "labelPosition"], size: "size", labelWidth: ["label-width", "labelWidth"], labelSuffix: ["label-suffix", "labelSuffix"] } });

    return ElFormProps;
}());
export { ElFormProps };
function ElFormProps_tsickle_Closure_declarations() {
    /** @type {?} */
    ElFormProps.propDecorators;
    /** @type {?} */
    ElFormProps.prototype.inline;
    /** @type {?} */
    ElFormProps.prototype.showIcon;
    /** @type {?} */
    ElFormProps.prototype.showMessage;
    /** @type {?} */
    ElFormProps.prototype.inlineMessage;
    /** @type {?} */
    ElFormProps.prototype.size;
    /** @type {?} */
    ElFormProps.prototype.labelPosition;
    /** @type {?} */
    ElFormProps.prototype.labelWidth;
    /** @type {?} */
    ElFormProps.prototype.labelSuffix;
}
