// element-angular/__ivy_ngcc__/release/form/form-item.js
import { Component, Input, Optional } from '@angular/core';
import { ElForm } from './form';
import { DomSanitizer } from '@angular/platform-browser';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './form';
import * as ɵngcc2 from '@angular/platform-browser';
import * as ɵngcc3 from '@angular/common';

function ElFormItem_label_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "label", 3);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleMap(ctx_r0.makeLabelStyle());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.label, " ");
} }
function ElFormItem_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("el-form-item__error--inline", ctx_r1.isInlineMessage());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r1.error, " ");
} }
var _c0 = ["*"];
var ElFormItem = /** @class */ (function () {
    /**
     * @param {?} root
     * @param {?} sanitizer
     */
    function ElFormItem(root, sanitizer) {
        this.root = root;
        this.sanitizer = sanitizer;
        this.required = false;
        this.showMessage = true;
        this.statusSubscriber = [];
    }
    /**
     * @return {?}
     */
    ElFormItem.prototype.showMessageEl = function () {
        if (this.status !== 'error')
            return false;
        return this.error && this.showMessage && this.root.showMessage;
    };
    /**
     * @return {?}
     */
    ElFormItem.prototype.isInlineMessage = function () {
        var /** @type {?} */ notDefault = typeof this.inlineMessage === 'boolean';
        return notDefault ? this.inlineMessage : this.root.inlineMessage;
    };
    /**
     * @return {?}
     */
    ElFormItem.prototype.makeSize = function () {
        var /** @type {?} */ appointed = this.size || this.root.size || '';
        return appointed ? " el-form-item--" + appointed + " " : '';
    };
    /**
     * @return {?}
     */
    ElFormItem.prototype.makeLabelStyle = function () {
        var /** @type {?} */ width = this.labelWidth || this.root.labelWidth || null;
        var /** @type {?} */ styles = width && this.root.labelPosition !== 'top' ? "width: " + width + ";" : '';
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @return {?}
     */
    ElFormItem.prototype.makeContentStyle = function () {
        var /** @type {?} */ skipStyle = (this.root.labelPosition === 'top' || this.root.inline)
            || (!this.label && !this.root.labelWidth);
        if (skipStyle)
            return this.sanitizer.bypassSecurityTrustStyle('');
        var /** @type {?} */ width = this.labelWidth || this.root.labelWidth;
        var /** @type {?} */ styles = "margin-left: " + width + ";";
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElFormItem.prototype.ngOnChanges = function (changes) {
        if (!this.root.showIcon)
            return;
        if (!changes || !changes.status)
            return;
        this.statusSubscriber.forEach(function (sub) {
            sub(changes.status.currentValue);
        });
    };
    /**
     * @nocollapse
     */
    ElFormItem.ctorParameters = function () { return [
        { type: ElForm, decorators: [{ type: Optional },] },
        { type: DomSanitizer, },
    ]; };
    ElFormItem.propDecorators = {
        'status': [{ type: Input },],
        'error': [{ type: Input },],
        'label': [{ type: Input },],
        'size': [{ type: Input },],
        'required': [{ type: Input },],
        'labelWidth': [{ type: Input, args: ['label-width',] },],
        'showMessage': [{ type: Input, args: ['show-message',] },],
        'inlineMessage': [{ type: Input, args: ['inline-message',] },],
    };
ElFormItem.ɵfac = function ElFormItem_Factory(t) { return new (t || ElFormItem)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElForm, 8), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.DomSanitizer)); };
ElFormItem.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElFormItem, selectors: [["el-form-item"]], inputs: { required: "required", showMessage: ["show-message", "showMessage"], status: "status", error: "error", label: "label", size: "size", labelWidth: ["label-width", "labelWidth"], inlineMessage: ["inline-message", "inlineMessage"] }, features: [ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c0, decls: 5, vars: 16, consts: [["class", "el-form-item__label", 3, "style", 4, "ngIf"], [1, "el-form-item__content"], ["class", "el-form-item__error", 3, "el-form-item__error--inline", 4, "ngIf"], [1, "el-form-item__label"], [1, "el-form-item__error"]], template: function ElFormItem_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵtemplate(1, ElFormItem_label_1_Template, 2, 3, "label", 0);
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵprojection(3);
        ɵngcc0.ɵɵtemplate(4, ElFormItem_div_4_Template, 2, 3, "div", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-form-item " + ctx.makeSize());
        ɵngcc0.ɵɵclassProp("el-form-item--feedback", ctx.root.showIcon)("is-error", ctx.status === "error")("is-success", ctx.status === "success")("is-validating", ctx.status === "validating")("is-required", ctx.required);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.label);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleMap(ctx.makeContentStyle());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showMessageEl);
    } }, directives: [ɵngcc3.NgIf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElFormItem, [{
        type: Component,
        args: [{
                selector: 'el-form-item',
                template: "\n  <div [class]=\"'el-form-item ' + makeSize()\"\n    [class.el-form-item--feedback]=\"root.showIcon\"\n    [class.is-error]=\"status === 'error'\"\n    [class.is-success]=\"status === 'success'\"\n    [class.is-validating]=\"status === 'validating'\"\n    [class.is-required]=\"required\">\n    <label class=\"el-form-item__label\" [style]=\"makeLabelStyle()\" *ngIf=\"label\">\n      {{label}}\n    </label>\n    <div class=\"el-form-item__content\" [style]=\"makeContentStyle()\">\n      <ng-content></ng-content>\n      <div *ngIf=\"showMessageEl\" class=\"el-form-item__error\"\n        [class.el-form-item__error--inline]=\"isInlineMessage()\">\n          {{error}}\n        </div>\n    </div>\n  </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElForm, decorators: [{
                type: Optional
            }] }, { type: ɵngcc2.DomSanitizer }]; }, { required: [{
            type: Input
        }], showMessage: [{
            type: Input,
            args: ['show-message']
        }], status: [{
            type: Input
        }], error: [{
            type: Input
        }], label: [{
            type: Input
        }], size: [{
            type: Input
        }], labelWidth: [{
            type: Input,
            args: ['label-width']
        }], inlineMessage: [{
            type: Input,
            args: ['inline-message']
        }] }); })();
    return ElFormItem;
}());
export { ElFormItem };
function ElFormItem_tsickle_Closure_declarations() {
    /** @type {?} */
    ElFormItem.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElFormItem.ctorParameters;
    /** @type {?} */
    ElFormItem.propDecorators;
    /** @type {?} */
    ElFormItem.prototype.status;
    /** @type {?} */
    ElFormItem.prototype.error;
    /** @type {?} */
    ElFormItem.prototype.label;
    /** @type {?} */
    ElFormItem.prototype.size;
    /** @type {?} */
    ElFormItem.prototype.required;
    /** @type {?} */
    ElFormItem.prototype.labelWidth;
    /** @type {?} */
    ElFormItem.prototype.showMessage;
    /** @type {?} */
    ElFormItem.prototype.inlineMessage;
    /** @type {?} */
    ElFormItem.prototype.statusSubscriber;
    /** @type {?} */
    ElFormItem.prototype.root;
    /** @type {?} */
    ElFormItem.prototype.sanitizer;
}
