// element-angular/__ivy_ngcc__/release/form/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElSharedModule } from '../shared/module';
import { ElButtonsModule } from '../button/module';
import { ElForm } from './form';
import { ElFormItem } from './form-item';
import * as ɵngcc0 from '@angular/core';
var ElFormModule = /** @class */ (function () {
    function ElFormModule() {
    }
    /**
     * @return {?}
     */
    ElFormModule.forRoot = function () {
        return { ngModule: ElFormModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElFormModule.ctorParameters = function () { return []; };
ElFormModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElFormModule });
ElFormModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElFormModule_Factory(t) { return new (t || ElFormModule)(); }, imports: [[CommonModule, ElButtonsModule, ElSharedModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElFormModule, { declarations: function () { return [ElForm, ElFormItem]; }, imports: function () { return [CommonModule, ElButtonsModule, ElSharedModule]; }, exports: function () { return [ElForm, ElFormItem]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElFormModule, [{
        type: NgModule,
        args: [{
                declarations: [ElForm, ElFormItem],
                exports: [ElForm, ElFormItem],
                imports: [CommonModule, ElButtonsModule, ElSharedModule],
                entryComponents: [ElForm]
            }]
    }], function () { return []; }, null); })();
    return ElFormModule;
}());
export { ElFormModule };
function ElFormModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElFormModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElFormModule.ctorParameters;
}
