// element-angular/__ivy_ngcc__/release/tag/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElTag } from './tag';
import * as ɵngcc0 from '@angular/core';
var ElTagsModule = /** @class */ (function () {
    function ElTagsModule() {
    }
    /**
     * @return {?}
     */
    ElTagsModule.forRoot = function () {
        return { ngModule: ElTagsModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElTagsModule.ctorParameters = function () { return []; };
ElTagsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElTagsModule });
ElTagsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElTagsModule_Factory(t) { return new (t || ElTagsModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElTagsModule, { declarations: function () { return [ElTag]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElTag]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElTagsModule, [{
        type: NgModule,
        args: [{
                declarations: [ElTag],
                exports: [ElTag],
                imports: [CommonModule],
                entryComponents: [ElTag]
            }]
    }], function () { return []; }, null); })();
    return ElTagsModule;
}());
export { ElTagsModule };
function ElTagsModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElTagsModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElTagsModule.ctorParameters;
}
