// element-angular/__ivy_ngcc__/release/shared/directives/class.directive.js
import { Directive, ElementRef, Input, Renderer2 } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var ElClassDirective = /** @class */ (function () {
    /**
     * @param {?} el
     * @param {?} renderer
     */
    function ElClassDirective(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.classNames = '';
    }
    /**
     * @return {?}
     */
    ElClassDirective.prototype.ngOnInit = function () {
        this.renderer.addClass(this.el.nativeElement, this.classNames);
    };
    /**
     * @nocollapse
     */
    ElClassDirective.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: Renderer2, },
    ]; };
    ElClassDirective.propDecorators = {
        'classNames': [{ type: Input, args: ['el-class',] },],
    };
ElClassDirective.ɵfac = function ElClassDirective_Factory(t) { return new (t || ElClassDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2)); };
ElClassDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElClassDirective, selectors: [["", "el-class", ""]], inputs: { classNames: ["el-class", "classNames"] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElClassDirective, [{
        type: Directive,
        args: [{
                selector: '[el-class]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }]; }, { classNames: [{
            type: Input,
            args: ['el-class']
        }] }); })();
    return ElClassDirective;
}());
export { ElClassDirective };
function ElClassDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    ElClassDirective.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElClassDirective.ctorParameters;
    /** @type {?} */
    ElClassDirective.propDecorators;
    /** @type {?} */
    ElClassDirective.prototype.classNames;
    /** @type {?} */
    ElClassDirective.prototype.el;
    /** @type {?} */
    ElClassDirective.prototype.renderer;
}
