// element-angular/__ivy_ngcc__/release/cascader/cascader.menu.js
import { Component, Optional } from '@angular/core';
import { ElCascader } from './cascader';
import { dropAnimation } from '../shared/animation/index';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './cascader';
import * as ɵngcc2 from '@angular/common';

function ElCascaderMenu_ul_1_li_1_Template(rf, ctx) { if (rf & 1) {
    var _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 4);
    ɵngcc0.ɵɵlistener("click", function ElCascaderMenu_ul_1_li_1_Template_li_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r8); var i_r5 = ctx.index; var step_r2 = ɵngcc0.ɵɵnextContext().index; var ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.root.selectHandle($event, step_r2, i_r5); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var listItem_r4 = ctx.$implicit;
    ɵngcc0.ɵɵclassProp("el-cascader-menu__item--extensible", listItem_r4.children)("is-active", listItem_r4.active)("is-disabled", listItem_r4.elDisabled);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", listItem_r4.label, " ");
} }
function ElCascaderMenu_ul_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ul", 2);
    ɵngcc0.ɵɵtemplate(1, ElCascaderMenu_ul_1_li_1_Template, 2, 7, "li", 3);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var menuItem_r1 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", menuItem_r1);
} }
var ElCascaderMenu = /** @class */ (function () {
    /**
     * @param {?} root
     */
    function ElCascaderMenu(root) {
        this.root = root;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    ElCascaderMenu.prototype.clickHandle = function (event) {
        event.stopPropagation();
    };
    /**
     * @nocollapse
     */
    ElCascaderMenu.ctorParameters = function () { return [
        { type: ElCascader, decorators: [{ type: Optional },] },
    ]; };
ElCascaderMenu.ɵfac = function ElCascaderMenu_Factory(t) { return new (t || ElCascaderMenu)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElCascader, 8)); };
ElCascaderMenu.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElCascaderMenu, selectors: [["el-cascader-menu"]], decls: 2, vars: 2, consts: [[1, "el-cascader-menus", 2, "z-index", "2007", "position", "absolute", "top", "100%", "left", "0", 3, "click"], ["class", "el-cascader-menu", 4, "ngFor", "ngForOf"], [1, "el-cascader-menu"], ["class", "el-cascader-menu__item", 3, "el-cascader-menu__item--extensible", "is-active", "is-disabled", "click", 4, "ngFor", "ngForOf"], [1, "el-cascader-menu__item", 3, "click"]], template: function ElCascaderMenu_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("click", function ElCascaderMenu_Template_div_click_0_listener($event) { return ctx.clickHandle($event); });
        ɵngcc0.ɵɵtemplate(1, ElCascaderMenu_ul_1_Template, 2, 1, "ul", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("@dropAnimation", ctx.root.menuVisible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.root.steps);
    } }, directives: [ɵngcc2.NgForOf], encapsulation: 2, data: { animation: [dropAnimation] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCascaderMenu, [{
        type: Component,
        args: [{
                selector: 'el-cascader-menu',
                template: "\n    <div class=\"el-cascader-menus\"\n      style=\"z-index: 2007; position: absolute; top: 100%; left: 0;\"\n      [@dropAnimation]=\"root.menuVisible\"\n      (click)=\"clickHandle($event)\">\n      <ul class=\"el-cascader-menu\" *ngFor=\"let menuItem of root.steps; let step = index\">\n        <li *ngFor=\"let listItem of menuItem; let i = index\"\n          class=\"el-cascader-menu__item\"\n          [class.el-cascader-menu__item--extensible]=\"listItem.children\"\n          [class.is-active]=\"listItem.active\"\n          [class.is-disabled]=\"listItem.elDisabled\"\n          (click)=\"root.selectHandle($event, step, i)\">\n          {{listItem.label}}\n        </li>\n      </ul>\n    </div>\n  ",
                animations: [dropAnimation]
            }]
    }], function () { return [{ type: ɵngcc1.ElCascader, decorators: [{
                type: Optional
            }] }]; }, null); })();
    return ElCascaderMenu;
}());
export { ElCascaderMenu };
function ElCascaderMenu_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCascaderMenu.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCascaderMenu.ctorParameters;
    /** @type {?} */
    ElCascaderMenu.prototype.root;
}
