// element-angular/__ivy_ngcc__/release/cascader/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElInputsModule } from '../input/module';
import { ElCascader } from './cascader';
import { ElCascaderMenu } from './cascader.menu';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../input/input';
import * as ɵngcc2 from '@angular/common';
var ElCascaderModule = /** @class */ (function () {
    function ElCascaderModule() {
    }
    /**
     * @return {?}
     */
    ElCascaderModule.forRoot = function () {
        return { ngModule: ElCascaderModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElCascaderModule.ctorParameters = function () { return []; };
ElCascaderModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElCascaderModule });
ElCascaderModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElCascaderModule_Factory(t) { return new (t || ElCascaderModule)(); }, imports: [[CommonModule, ElInputsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElCascaderModule, { declarations: function () { return [ElCascader, ElCascaderMenu]; }, imports: function () { return [CommonModule, ElInputsModule]; }, exports: function () { return [ElCascader, ElCascaderMenu]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCascaderModule, [{
        type: NgModule,
        args: [{
                declarations: [ElCascader, ElCascaderMenu],
                exports: [ElCascader, ElCascaderMenu],
                imports: [CommonModule, ElInputsModule],
                entryComponents: [ElCascader]
            }]
    }], function () { return []; }, null); })();
ɵngcc0.ɵɵsetComponentScope(ElCascader, [ɵngcc1.ElInput, ɵngcc2.NgIf, ɵngcc2.NgForOf, ElCascaderMenu], []);
    return ElCascaderModule;
}());
export { ElCascaderModule };
function ElCascaderModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCascaderModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCascaderModule.ctorParameters;
}
