// element-angular/__ivy_ngcc__/release/menu/menu.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

var _c0 = function (a0) { return { backgroundColor: a0 }; };
var _c1 = ["*"];
var ElMenu = /** @class */ (function () {
    function ElMenu() {
        this.mode = 'vertical';
        this.theme = 'light';
        this.uniqueOpened = false;
        this.menuTrigger = 'hover';
        this.modelChange = new EventEmitter();
        this.openedMenus = this.defaultOpeneds ? this.defaultOpeneds.slice(0) : [];
    }
    /**
     * @param {?} index
     * @return {?}
     */
    ElMenu.prototype.openMenu = function (index) {
        var /** @type {?} */ openedMenus = this.openedMenus;
        if (openedMenus.indexOf(index) !== -1)
            return;
        this.openedMenus.push(index);
    };
    /**
     * @param {?} index
     * @return {?}
     */
    ElMenu.prototype.closeMenu = function (index) {
        this.openedMenus.splice(this.openedMenus.indexOf(index), 1);
    };
    /**
     * @param {?} index
     * @param {?=} path
     * @return {?}
     */
    ElMenu.prototype.selectHandle = function (index, path) {
        var /** @type {?} */ main = path || index;
        this.model = main;
        this.modelChange.emit(main);
    };
    /**
     * @return {?}
     */
    ElMenu.prototype.hoverBackgroundColor = function () {
        return this.backgroundColor ? this.hexToRGB() : '';
    };
    /**
     * @return {?}
     */
    ElMenu.prototype.hexToRGB = function () {
        var /** @type {?} */ hex = +this.backgroundColor.replace('#', '0x');
        var /** @type {?} */ rgb = [(hex & 0xff0000) >> 16, (hex & 0x00ff00) >> 8, hex & 0x0000ff];
        return "rgb(" + rgb.map(function (v) { return ~~(0.8 * v); }).join(',') + ")";
    };
    /**
     * @nocollapse
     */
    ElMenu.ctorParameters = function () { return []; };
    ElMenu.propDecorators = {
        'mode': [{ type: Input },],
        'theme': [{ type: Input },],
        'model': [{ type: Input },],
        'nativeClass': [{ type: Input, args: ['class',] },],
        'defaultOpeneds': [{ type: Input, args: ['default-openeds',] },],
        'uniqueOpened': [{ type: Input, args: ['unique-opened',] },],
        'menuTrigger': [{ type: Input, args: ['menu-trigger',] },],
        'backgroundColor': [{ type: Input, args: ['background-color',] },],
        'textColor': [{ type: Input, args: ['text-color',] },],
        'activeTextColor': [{ type: Input, args: ['active-text-color',] },],
        'modelChange': [{ type: Output },],
    };
ElMenu.ɵfac = function ElMenu_Factory(t) { return new (t || ElMenu)(); };
ElMenu.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElMenu, selectors: [["el-menu"]], inputs: { mode: "mode", theme: "theme", uniqueOpened: ["unique-opened", "uniqueOpened"], menuTrigger: ["menu-trigger", "menuTrigger"], model: "model", nativeClass: ["class", "nativeClass"], defaultOpeneds: ["default-openeds", "defaultOpeneds"], backgroundColor: ["background-color", "backgroundColor"], textColor: ["text-color", "textColor"], activeTextColor: ["active-text-color", "activeTextColor"] }, outputs: { modelChange: "modelChange" }, ngContentSelectors: _c1, decls: 2, vars: 7, consts: [[3, "ngStyle"]], template: function ElMenu_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "ul", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-menu " + ctx.nativeClass);
        ɵngcc0.ɵɵclassProp("el-menu--horizontal", ctx.mode === "horizontal");
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(5, _c0, ctx.backgroundColor || ""));
    } }, directives: [ɵngcc1.NgStyle], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElMenu, [{
        type: Component,
        args: [{
                selector: 'el-menu',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: "\n    <ul [class]=\"'el-menu ' + nativeClass\"\n      [class.el-menu--horizontal]=\"mode === 'horizontal'\"\n      [ngStyle]=\"{ backgroundColor: backgroundColor || '' }\">\n      <ng-content></ng-content>\n    </ul>\n  "
            }]
    }], function () { return []; }, { mode: [{
            type: Input
        }], theme: [{
            type: Input
        }], uniqueOpened: [{
            type: Input,
            args: ['unique-opened']
        }], menuTrigger: [{
            type: Input,
            args: ['menu-trigger']
        }], modelChange: [{
            type: Output
        }], model: [{
            type: Input
        }], nativeClass: [{
            type: Input,
            args: ['class']
        }], defaultOpeneds: [{
            type: Input,
            args: ['default-openeds']
        }], backgroundColor: [{
            type: Input,
            args: ['background-color']
        }], textColor: [{
            type: Input,
            args: ['text-color']
        }], activeTextColor: [{
            type: Input,
            args: ['active-text-color']
        }] }); })();
    return ElMenu;
}());
export { ElMenu };
function ElMenu_tsickle_Closure_declarations() {
    /** @type {?} */
    ElMenu.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElMenu.ctorParameters;
    /** @type {?} */
    ElMenu.propDecorators;
    /** @type {?} */
    ElMenu.prototype.mode;
    /** @type {?} */
    ElMenu.prototype.theme;
    /** @type {?} */
    ElMenu.prototype.model;
    /** @type {?} */
    ElMenu.prototype.nativeClass;
    /** @type {?} */
    ElMenu.prototype.defaultOpeneds;
    /** @type {?} */
    ElMenu.prototype.uniqueOpened;
    /** @type {?} */
    ElMenu.prototype.menuTrigger;
    /** @type {?} */
    ElMenu.prototype.backgroundColor;
    /** @type {?} */
    ElMenu.prototype.textColor;
    /** @type {?} */
    ElMenu.prototype.activeTextColor;
    /** @type {?} */
    ElMenu.prototype.modelChange;
    /** @type {?} */
    ElMenu.prototype.openedMenus;
    /** @type {?} */
    ElMenu.prototype.showBorderIndex;
}
