// element-angular/__ivy_ngcc__/release/menu/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElTooltipModule } from '../tooltip/module';
import { ElMenuItemGroup } from './menu-item-group';
import { ElMenuItem } from './menu-item';
import { ElSubmenu } from './submenu';
import { ElMenu } from './menu';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../tooltip/module';
var ElMenusModule = /** @class */ (function () {
    function ElMenusModule() {
    }
    /**
     * @return {?}
     */
    ElMenusModule.forRoot = function () {
        return { ngModule: ElMenusModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElMenusModule.ctorParameters = function () { return []; };
ElMenusModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElMenusModule });
ElMenusModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElMenusModule_Factory(t) { return new (t || ElMenusModule)(); }, imports: [[CommonModule, ElTooltipModule.forRoot()]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElMenusModule, { declarations: function () { return [ElMenu, ElSubmenu, ElMenuItem, ElMenuItemGroup]; }, imports: function () { return [CommonModule, ɵngcc1.ElTooltipModule]; }, exports: function () { return [ElMenu, ElSubmenu, ElMenuItem, ElMenuItemGroup]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElMenusModule, [{
        type: NgModule,
        args: [{
                declarations: [ElMenu, ElSubmenu, ElMenuItem, ElMenuItemGroup],
                exports: [ElMenu, ElSubmenu, ElMenuItem, ElMenuItemGroup],
                imports: [CommonModule, ElTooltipModule.forRoot()],
                entryComponents: [ElMenu, ElSubmenu, ElMenuItem, ElMenuItemGroup]
            }]
    }], function () { return []; }, null); })();
    return ElMenusModule;
}());
export { ElMenusModule };
function ElMenusModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElMenusModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElMenusModule.ctorParameters;
}
