// element-angular/__ivy_ngcc__/release/dropdown/dropdown.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';
import * as ɵngcc2 from '../shared/services/index';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '../button/button';
import * as ɵngcc5 from './dropdown.item';

var _c0 = ["list"];
function ElDropdown_ng_template_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0);
} }
function ElDropdown_ng_container_3_ng_template_3_Template(rf, ctx) { }
function ElDropdown_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    var _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "el-button", 6);
    ɵngcc0.ɵɵlistener("click", function ElDropdown_ng_container_3_Template_el_button_click_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r8); var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.openMenu($event); });
    ɵngcc0.ɵɵelementStart(2, "span");
    ɵngcc0.ɵɵtemplate(3, ElDropdown_ng_container_3_ng_template_3_Template, 0, 0, "ng-template", 7);
    ɵngcc0.ɵɵelement(4, "i", 8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("type", ctx_r2.type)("size", ctx_r2.size);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", _r0);
} }
function ElDropdown_ng_container_4_ng_template_2_Template(rf, ctx) { }
function ElDropdown_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    var _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "span", 9);
    ɵngcc0.ɵɵlistener("click", function ElDropdown_ng_container_4_Template_span_click_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r11); var ctx_r10 = ɵngcc0.ɵɵnextContext(); return ctx_r10.openMenu($event); });
    ɵngcc0.ɵɵtemplate(2, ElDropdown_ng_container_4_ng_template_2_Template, 0, 0, "ng-template", 7);
    ɵngcc0.ɵɵelement(3, "i", 8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext();
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", _r0);
} }
function ElDropdown_el_dropdown_item_8_Template(rf, ctx) { if (rf & 1) {
    var _r14 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-dropdown-item", 10);
    ɵngcc0.ɵɵlistener("selected", function ElDropdown_el_dropdown_item_8_Template_el_dropdown_item_selected_0_listener() { ɵngcc0.ɵɵrestoreView(_r14); var item_r12 = ctx.$implicit; var ctx_r13 = ɵngcc0.ɵɵnextContext(); return ctx_r13.selectHandle(item_r12); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r12 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("model", item_r12);
} }
var _c1 = ["*"];
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, ElementRef, HostListener, Renderer2, ViewChild, } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { ElDropdownProps } from './dropdown.props';
import { dropAnimation } from './animation';
import { DocumentWrapper } from '../shared/services/index';
var ElDropdown = /** @class */ (function (_super) {
    __extends(ElDropdown, _super);
    /**
     * @param {?} renderer
     * @param {?} sanitizer
     * @param {?} document
     * @param {?} el
     */
    function ElDropdown(renderer, sanitizer, document, el) {
        var _this = _super.call(this) || this;
        _this.renderer = renderer;
        _this.sanitizer = sanitizer;
        _this.document = document;
        _this.el = el;
        _this.showMenu = false;
        _this.slideToBottom = true;
        _this.mouseleave = function () {
            if (_this.trigger !== 'hover')
                return;
            _this.timer = setTimeout(function () {
                _this.closeMenu();
                clearTimeout(_this.timer);
            }, 400);
        };
        _this.mouseenter = function () {
            if (_this.trigger !== 'hover')
                return;
            _this.timer && clearTimeout(_this.timer);
            !_this.showMenu && _this.visibleChange.emit();
            _this.showMenu = true;
        };
        return _this;
    }
    /**
     * @param {?=} event
     * @return {?}
     */
    ElDropdown.prototype.openMenu = function (event) {
        var _this = this;
        event && event.stopPropagation();
        this.showMenu = !this.showMenu;
        this.visibleChange.emit();
        this.globalListenFunc = this.renderer.listen('document', 'click', function () {
            _this.closeMenu();
        });
    };
    /**
     * @return {?}
     */
    ElDropdown.prototype.closeMenu = function () {
        this.showMenu = false;
        this.visibleChange.emit();
        this.globalListenFunc && this.globalListenFunc();
    };
    /**
     * @param {?} model
     * @return {?}
     */
    ElDropdown.prototype.selectHandle = function (model) {
        this.selected.emit(model);
        // select and hide menu (props)
        this.hideOnClick && this.closeMenu();
    };
    /**
     * @return {?}
     */
    ElDropdown.prototype.makeListStyles = function () {
        var /** @type {?} */ styles = "top: " + (this.slideToBottom ? '100%' : (0 - this.listHeight) + 'px') + ";" +
            ((this.menuAlign === 'end' ? 'right' : 'left') + ": 0; position: absolute; min-width: 100px;") +
            ("white-space: " + (this.menuNoWrap ? 'nowrap' : 'normal'));
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @return {?}
     */
    ElDropdown.prototype.ngAfterViewInit = function () {
        var _this = this;
        this.listHeight = this.list.nativeElement.offsetHeight + 12;
        this.globalScrollFunc = this.renderer.listen('window', 'scroll', function () {
            var /** @type {?} */ top = _this.el.nativeElement.getBoundingClientRect().top;
            var /** @type {?} */ nextValue = _this.document.documentElement.clientHeight - top > _this.listHeight;
            if (nextValue !== _this.slideToBottom) {
                _this.slideToBottom = nextValue;
            }
        });
    };
    /**
     * @return {?}
     */
    ElDropdown.prototype.ngOnDestroy = function () {
        this.globalListenFunc && this.globalListenFunc();
        this.globalScrollFunc && this.globalScrollFunc();
    };
    /**
     * @nocollapse
     */
    ElDropdown.ctorParameters = function () { return [
        { type: Renderer2, },
        { type: DomSanitizer, },
        { type: DocumentWrapper, },
        { type: ElementRef, },
    ]; };
    ElDropdown.propDecorators = {
        'list': [{ type: ViewChild, args: ['list',] },],
        'mouseleave': [{ type: HostListener, args: ['mouseleave',] },],
        'mouseenter': [{ type: HostListener, args: ['mouseenter',] },],
    };
ElDropdown.ɵfac = function ElDropdown_Factory(t) { return new (t || ElDropdown)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.DocumentWrapper), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElDropdown.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElDropdown, selectors: [["el-dropdown"]], viewQuery: function ElDropdown_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.list = _t.first);
    } }, hostBindings: function ElDropdown_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("mouseleave", function ElDropdown_mouseleave_HostBindingHandler() { return ctx.mouseleave(); })("mouseenter", function ElDropdown_mouseenter_HostBindingHandler() { return ctx.mouseenter(); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c1, decls: 9, vars: 6, consts: [["content", ""], [1, "el-dropdown"], [4, "ngIf"], [1, "el-dropdown-menu"], ["list", ""], [3, "model", "selected", 4, "ngFor", "ngForOf"], [3, "type", "size", "click"], [3, "ngTemplateOutlet"], [1, "el-icon-caret-bottom", "el-icon--right"], [1, "el-dropdown-link", 2, "cursor", "pointer", 3, "click"], [3, "model", "selected"]], template: function ElDropdown_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵtemplate(0, ElDropdown_ng_template_0_Template, 1, 0, "ng-template", null, 0, ɵngcc0.ɵɵtemplateRefExtractor);
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵtemplate(3, ElDropdown_ng_container_3_Template, 5, 3, "ng-container", 2);
        ɵngcc0.ɵɵtemplate(4, ElDropdown_ng_container_4_Template, 4, 1, "ng-container", 2);
        ɵngcc0.ɵɵelementStart(5, "div");
        ɵngcc0.ɵɵelementStart(6, "ul", 3, 4);
        ɵngcc0.ɵɵtemplate(8, ElDropdown_el_dropdown_item_8_Template, 1, 1, "el-dropdown-item", 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", ctx.splitButton);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.splitButton);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleMap(ctx.makeListStyles());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("@dropAnimation", ctx.showMenu);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.model);
    } }, directives: [ɵngcc3.NgIf, ɵngcc3.NgForOf, ɵngcc4.ElButton, ɵngcc3.NgTemplateOutlet, ɵngcc5.ElDropdownItem], styles: [".el-dropdown-menu[_ngcontent-%COMP%] { overflow: hidden; }"], data: { animation: [dropAnimation] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElDropdown, [{
        type: Component,
        args: [{
                selector: 'el-dropdown',
                template: "\n    <ng-template #content>\n      <ng-content></ng-content>\n    </ng-template>\n    <div class=\"el-dropdown\">\n      <ng-container *ngIf=\"splitButton\">\n        <el-button [type]=\"type\" [size]=\"size\" (click)=\"openMenu($event)\">\n          <span>\n            <ng-template [ngTemplateOutlet]=\"content\"></ng-template>\n            <i class=\"el-icon-caret-bottom el-icon--right\"></i>\n          </span>\n        </el-button>\n      </ng-container>\n      <ng-container *ngIf=\"!splitButton\">\n        <span class=\"el-dropdown-link\" (click)=\"openMenu($event)\" style=\"cursor: pointer;\">\n          <ng-template [ngTemplateOutlet]=\"content\"></ng-template>\n          <i class=\"el-icon-caret-bottom el-icon--right\"></i>\n        </span>\n      </ng-container>\n      <div [style]=\"makeListStyles()\">\n        <ul class=\"el-dropdown-menu\" #list [@dropAnimation]=\"showMenu\">\n          <el-dropdown-item *ngFor=\"let item of model\" [model]=\"item\" (selected)=\"selectHandle(item)\">\n          </el-dropdown-item>\n        </ul>\n      </div>\n    </div>\n  ",
                animations: [dropAnimation],
                styles: ['.el-dropdown-menu { overflow: hidden; }']
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc1.DomSanitizer }, { type: ɵngcc2.DocumentWrapper }, { type: ɵngcc0.ElementRef }]; }, { list: [{
            type: ViewChild,
            args: ['list']
        }], mouseleave: [{
            type: HostListener,
            args: ['mouseleave']
        }], mouseenter: [{
            type: HostListener,
            args: ['mouseenter']
        }] }); })();
    return ElDropdown;
}(ElDropdownProps));
export { ElDropdown };
function ElDropdown_tsickle_Closure_declarations() {
    /** @type {?} */
    ElDropdown.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElDropdown.ctorParameters;
    /** @type {?} */
    ElDropdown.propDecorators;
    /** @type {?} */
    ElDropdown.prototype.list;
    /** @type {?} */
    ElDropdown.prototype.showMenu;
    /** @type {?} */
    ElDropdown.prototype.timer;
    /** @type {?} */
    ElDropdown.prototype.slideToBottom;
    /** @type {?} */
    ElDropdown.prototype.listHeight;
    /** @type {?} */
    ElDropdown.prototype.globalListenFunc;
    /** @type {?} */
    ElDropdown.prototype.globalScrollFunc;
    /** @type {?} */
    ElDropdown.prototype.mouseleave;
    /** @type {?} */
    ElDropdown.prototype.mouseenter;
    /** @type {?} */
    ElDropdown.prototype.renderer;
    /** @type {?} */
    ElDropdown.prototype.sanitizer;
    /** @type {?} */
    ElDropdown.prototype.document;
    /** @type {?} */
    ElDropdown.prototype.el;
}
