// element-angular/__ivy_ngcc__/release/dropdown/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElSharedModule } from '../shared/module';
import { ElButtonsModule } from '../button/module';
import { ElDropdownItem } from './dropdown.item';
import { ElDropdown } from './dropdown';
import * as ɵngcc0 from '@angular/core';
var ElDropdownModule = /** @class */ (function () {
    function ElDropdownModule() {
    }
    /**
     * @return {?}
     */
    ElDropdownModule.forRoot = function () {
        return { ngModule: ElDropdownModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElDropdownModule.ctorParameters = function () { return []; };
ElDropdownModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElDropdownModule });
ElDropdownModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElDropdownModule_Factory(t) { return new (t || ElDropdownModule)(); }, imports: [[CommonModule, ElButtonsModule, ElSharedModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElDropdownModule, { declarations: function () { return [ElDropdown, ElDropdownItem]; }, imports: function () { return [CommonModule, ElButtonsModule, ElSharedModule]; }, exports: function () { return [ElDropdown, ElDropdownItem]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElDropdownModule, [{
        type: NgModule,
        args: [{
                declarations: [ElDropdown, ElDropdownItem],
                exports: [ElDropdown, ElDropdownItem],
                imports: [CommonModule, ElButtonsModule, ElSharedModule],
                entryComponents: [ElDropdown]
            }]
    }], function () { return []; }, null); })();
    return ElDropdownModule;
}());
export { ElDropdownModule };
function ElDropdownModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElDropdownModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElDropdownModule.ctorParameters;
}
