// element-angular/__ivy_ngcc__/release/radio/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ElRadioGroup } from './radio-group';
import { ElRadioButton } from './radio-button';
import { ElRadio } from './radio';
import * as ɵngcc0 from '@angular/core';
var ElRadiosModule = /** @class */ (function () {
    function ElRadiosModule() {
    }
    /**
     * @return {?}
     */
    ElRadiosModule.forRoot = function () {
        return { ngModule: ElRadiosModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElRadiosModule.ctorParameters = function () { return []; };
ElRadiosModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElRadiosModule });
ElRadiosModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElRadiosModule_Factory(t) { return new (t || ElRadiosModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElRadiosModule, { declarations: function () { return [ElRadio, ElRadioGroup, ElRadioButton]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [ElRadio, ElRadioGroup, ElRadioButton]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRadiosModule, [{
        type: NgModule,
        args: [{
                declarations: [ElRadio, ElRadioGroup, ElRadioButton],
                exports: [ElRadio, ElRadioGroup, ElRadioButton],
                imports: [CommonModule, FormsModule],
                entryComponents: [ElRadio, ElRadioGroup, ElRadioButton]
            }]
    }], function () { return []; }, null); })();
    return ElRadiosModule;
}());
export { ElRadiosModule };
function ElRadiosModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRadiosModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRadiosModule.ctorParameters;
}
