// element-angular/__ivy_ngcc__/release/radio/radio-button.js
import { Component, Input, Output, EventEmitter, Optional, ElementRef, } from '@angular/core';
import { ElRadioGroup } from './radio-group';
import { isParentTag, removeNgTag } from '../shared/utils/index';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './radio-group';
import * as ɵngcc2 from '@angular/forms';
import * as ɵngcc3 from '@angular/common';

var _c0 = ["*"];
var ElRadioButton = /** @class */ (function () {
    /**
     * @param {?} rootGroup
     * @param {?} el
     */
    function ElRadioButton(rootGroup, el) {
        this.rootGroup = rootGroup;
        this.el = el;
        this.elDisabled = false;
        this.nativeName = '';
        this.modelChange = new EventEmitter();
        this.showLabel = false;
        this.parentIsGroup = false;
    }
    Object.defineProperty(ElRadioButton.prototype, "disabled", {
        /**
         * @param {?} val
         * @return {?}
         */
        set: function (val) {
            console.warn('Element Angular: (disabled) is discarded, use [elDisabled] replace it.');
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    ElRadioButton.prototype.changeHandle = function () {
        if (this.parentIsGroup) {
            return this.rootGroup.changeHandle(this.label);
        }
        this.modelChange.emit(this.label);
    };
    /**
     * @return {?}
     */
    ElRadioButton.prototype.ngOnInit = function () {
        var _this = this;
        var /** @type {?} */ nativeElement = this.el.nativeElement;
        var /** @type {?} */ update = function () {
            _this.elDisabled = _this.rootGroup.elDisabled;
            _this.model = _this.rootGroup.model;
            _this.size = _this.rootGroup.buttonSize;
            _this.activeStyles = {
                backgroundColor: _this.rootGroup.fillColor || '',
                borderColor: _this.rootGroup.fillColor || '',
                boxShadow: _this.rootGroup.fillColor ? "-1px 0 0 0 " + _this.rootGroup.fillColor : '',
                color: _this.rootGroup.textColor || '',
            };
        };
        this.parentIsGroup = isParentTag(nativeElement, 'el-radio-group');
        removeNgTag(nativeElement);
        if (this.parentIsGroup && this.rootGroup) {
            update();
            this.rootGroup.subscriber.push(update);
        }
    };
    /**
     * @nocollapse
     */
    ElRadioButton.ctorParameters = function () { return [
        { type: ElRadioGroup, decorators: [{ type: Optional },] },
        { type: ElementRef, },
    ]; };
    ElRadioButton.propDecorators = {
        'disabled': [{ type: Input },],
        'elDisabled': [{ type: Input },],
        'label': [{ type: Input },],
        'nativeName': [{ type: Input, args: ['name',] },],
        'model': [{ type: Input },],
        'modelChange': [{ type: Output },],
    };
ElRadioButton.ɵfac = function ElRadioButton_Factory(t) { return new (t || ElRadioButton)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElRadioGroup, 8), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElRadioButton.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElRadioButton, selectors: [["el-radio-button"]], inputs: { elDisabled: "elDisabled", nativeName: ["name", "nativeName"], disabled: "disabled", label: "label", model: "model" }, outputs: { modelChange: "modelChange" }, ngContentSelectors: _c0, decls: 4, vars: 11, consts: [["type", "radio", 1, "el-radio-button__orig-radio", 3, "value", "name", "disabled", "ngModel", "ngModelChange"], [1, "el-radio-button__inner", 3, "ngStyle"]], template: function ElRadioButton_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "label");
        ɵngcc0.ɵɵelementStart(1, "input", 0);
        ɵngcc0.ɵɵlistener("ngModelChange", function ElRadioButton_Template_input_ngModelChange_1_listener() { return ctx.changeHandle(); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(2, "span", 1);
        ɵngcc0.ɵɵprojection(3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-radio-button" + (ctx.size ? " el-radio-button--" + ctx.size : ""));
        ɵngcc0.ɵɵclassProp("is-disabled", ctx.elDisabled)("is-active", ctx.model === ctx.label);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("value", ctx.label)("name", ctx.nativeName)("disabled", ctx.elDisabled)("ngModel", ctx.model);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngStyle", ctx.model === ctx.label && ctx.activeStyles);
    } }, directives: [ɵngcc2.RadioControlValueAccessor, ɵngcc2.DefaultValueAccessor, ɵngcc2.NgControlStatus, ɵngcc2.NgModel, ɵngcc3.NgStyle], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRadioButton, [{
        type: Component,
        args: [{
                selector: 'el-radio-button',
                template: "\n    <label [class]=\"'el-radio-button' + (size ? ' el-radio-button--' + size : '')\"\n      [class.is-disabled]=\"elDisabled\"\n      [class.is-active]=\"model === label\">\n      <input class=\"el-radio-button__orig-radio\" type=\"radio\"\n        [value]=\"label\" [name]=\"nativeName\" [disabled]=\"elDisabled\"\n        [ngModel]=\"model\" (ngModelChange)=\"changeHandle()\">\n      <span class=\"el-radio-button__inner\" [ngStyle]=\"model === label && activeStyles\">\n        <ng-content></ng-content>\n      </span>\n    </label>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElRadioGroup, decorators: [{
                type: Optional
            }] }, { type: ɵngcc0.ElementRef }]; }, { elDisabled: [{
            type: Input
        }], nativeName: [{
            type: Input,
            args: ['name']
        }], modelChange: [{
            type: Output
        }], disabled: [{
            type: Input
        }], label: [{
            type: Input
        }], model: [{
            type: Input
        }] }); })();
    return ElRadioButton;
}());
export { ElRadioButton };
function ElRadioButton_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRadioButton.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRadioButton.ctorParameters;
    /** @type {?} */
    ElRadioButton.propDecorators;
    /** @type {?} */
    ElRadioButton.prototype.elDisabled;
    /** @type {?} */
    ElRadioButton.prototype.label;
    /** @type {?} */
    ElRadioButton.prototype.nativeName;
    /** @type {?} */
    ElRadioButton.prototype.model;
    /** @type {?} */
    ElRadioButton.prototype.modelChange;
    /** @type {?} */
    ElRadioButton.prototype.size;
    /** @type {?} */
    ElRadioButton.prototype.showLabel;
    /** @type {?} */
    ElRadioButton.prototype.parentIsGroup;
    /** @type {?} */
    ElRadioButton.prototype.activeStyles;
    /** @type {?} */
    ElRadioButton.prototype.rootGroup;
    /** @type {?} */
    ElRadioButton.prototype.el;
}
