// element-angular/__ivy_ngcc__/release/element-angular.module.js
import { NgModule } from '@angular/core';
import { ElMessageService } from './message/message.service';
import { ElNotificationService } from './notification/notification.service';
import { ElSharedModule } from './shared/module';
import { ElButtonsModule } from './button/module';
import { ElCheckboxsModule } from './checkbox/module';
import { ElIconsModule } from './icon/module';
import { ElRadiosModule } from './radio/module';
import { ElMenusModule } from './menu/module';
import { ElTooltipModule } from './tooltip/module';
import { ElRowModule } from './row/module';
import { ElColModule } from './col/module';
import { ElInputsModule } from './input/module';
import { ElInputNumberModule } from './input-number/module';
import { ElTagsModule } from './tag/module';
import { ElSelectModule } from './select/module';
import { ElSwitchModule } from './switch/module';
import { ElRateModule } from './rate/module';
import { ElProgressModule } from './progress/module';
import { ElStepsModule } from './steps/module';
import { ElLoadingModule } from './loading/module';
import { ElMessagesModule } from './message/module';
import { ElNotificationModule } from './notification/module';
import { ElCascaderModule } from './cascader/module';
import { ElBadgesModule } from './badge/module';
import { ElCardsModule } from './card/module';
import { ElDropdownModule } from './dropdown/module';
import { ElBreadcrumbsModule } from './breadcrumb/module';
import { ElDateModule } from './date-picker/module';
import { ElSliderModule } from './slider/module';
import { ElDialogModule } from './dialog/module';
import { ElCarouselModule } from './carousel/module';
import { ElCollapseModule } from './collapse/module';
import { ElAlertModule } from './alert/module';
import { ElPaginationModule } from './pagination/module';
import { ElUploadModule } from './upload/module';
import { ElTableModule } from './table/module';
import { ElContainerModule } from './container/module';
import { ElFormModule } from './form/module';
import { ElTreeModule } from './tree/module';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './button/module';
import * as ɵngcc2 from './icon/module';
import * as ɵngcc3 from './radio/module';
import * as ɵngcc4 from './menu/module';
import * as ɵngcc5 from './tooltip/module';
import * as ɵngcc6 from './row/module';
import * as ɵngcc7 from './col/module';
import * as ɵngcc8 from './checkbox/module';
import * as ɵngcc9 from './input/module';
import * as ɵngcc10 from './input-number/module';
import * as ɵngcc11 from './tag/module';
import * as ɵngcc12 from './select/module';
import * as ɵngcc13 from './switch/module';
import * as ɵngcc14 from './rate/module';
import * as ɵngcc15 from './progress/module';
import * as ɵngcc16 from './steps/module';
import * as ɵngcc17 from './loading/module';
import * as ɵngcc18 from './message/module';
import * as ɵngcc19 from './shared/module';
import * as ɵngcc20 from './notification/module';
import * as ɵngcc21 from './cascader/module';
import * as ɵngcc22 from './badge/module';
import * as ɵngcc23 from './card/module';
import * as ɵngcc24 from './dropdown/module';
import * as ɵngcc25 from './breadcrumb/module';
import * as ɵngcc26 from './date-picker/module';
import * as ɵngcc27 from './slider/module';
import * as ɵngcc28 from './dialog/module';
import * as ɵngcc29 from './carousel/module';
import * as ɵngcc30 from './collapse/module';
import * as ɵngcc31 from './alert/module';
import * as ɵngcc32 from './pagination/module';
import * as ɵngcc33 from './upload/module';
import * as ɵngcc34 from './table/module';
import * as ɵngcc35 from './container/module';
import * as ɵngcc36 from './form/module';
import * as ɵngcc37 from './tree/module';
export var /** @type {?} */ ElChildModules = {
    ElButtonsModule: ElButtonsModule, ElIconsModule: ElIconsModule, ElRadiosModule: ElRadiosModule, ElMenusModule: ElMenusModule, ElTooltipModule: ElTooltipModule, ElRowModule: ElRowModule,
    ElColModule: ElColModule, ElCheckboxsModule: ElCheckboxsModule, ElInputsModule: ElInputsModule, ElInputNumberModule: ElInputNumberModule, ElTagsModule: ElTagsModule, ElSelectModule: ElSelectModule,
    ElSwitchModule: ElSwitchModule, ElRateModule: ElRateModule, ElProgressModule: ElProgressModule, ElStepsModule: ElStepsModule, ElLoadingModule: ElLoadingModule, ElMessagesModule: ElMessagesModule,
    ElSharedModule: ElSharedModule, ElNotificationModule: ElNotificationModule, ElCascaderModule: ElCascaderModule, ElBadgesModule: ElBadgesModule, ElCardsModule: ElCardsModule, ElDropdownModule: ElDropdownModule,
    ElBreadcrumbsModule: ElBreadcrumbsModule, ElDateModule: ElDateModule, ElSliderModule: ElSliderModule, ElDialogModule: ElDialogModule, ElCarouselModule: ElCarouselModule, ElCollapseModule: ElCollapseModule,
    ElAlertModule: ElAlertModule, ElPaginationModule: ElPaginationModule, ElUploadModule: ElUploadModule, ElTableModule: ElTableModule, ElContainerModule: ElContainerModule, ElFormModule: ElFormModule,
    ElTreeModule: ElTreeModule,
};
export var /** @type {?} */ ELMODULES_GROUP = [
    ElButtonsModule, ElIconsModule, ElRadiosModule, ElMenusModule, ElTooltipModule, ElRowModule,
    ElColModule, ElCheckboxsModule, ElInputsModule, ElInputNumberModule, ElTagsModule, ElSelectModule,
    ElSwitchModule, ElRateModule, ElProgressModule, ElStepsModule, ElLoadingModule, ElMessagesModule,
    ElSharedModule, ElNotificationModule, ElCascaderModule, ElBadgesModule, ElCardsModule, ElDropdownModule,
    ElBreadcrumbsModule, ElDateModule, ElSliderModule, ElDialogModule, ElCarouselModule, ElCollapseModule,
    ElAlertModule, ElPaginationModule, ElUploadModule, ElTableModule, ElContainerModule, ElFormModule,
    ElTreeModule,
];
var ElModule = /** @class */ (function () {
    function ElModule() {
    }
    /**
     * @return {?}
     */
    ElModule.forRoot = function () {
        return {
            ngModule: ElModule,
            providers: [],
        };
    };
    /**
     * @nocollapse
     */
    ElModule.ctorParameters = function () { return []; };
ElModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElModule });
ElModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElModule_Factory(t) { return new (t || ElModule)(); }, imports: [[
            ElButtonsModule.forRoot(), ElIconsModule.forRoot(), ElRadiosModule.forRoot(), ElMenusModule.forRoot(),
            ElTooltipModule.forRoot(), ElRowModule.forRoot(), ElColModule.forRoot(), ElCheckboxsModule.forRoot(),
            ElInputsModule.forRoot(), ElInputNumberModule.forRoot(), ElTagsModule.forRoot(), ElSelectModule.forRoot(),
            ElSwitchModule.forRoot(), ElRateModule.forRoot(), ElProgressModule.forRoot(), ElStepsModule.forRoot(),
            ElLoadingModule.forRoot(), ElMessagesModule.forRoot(), ElSharedModule.forRoot(), ElNotificationModule.forRoot(),
            ElCascaderModule.forRoot(), ElBadgesModule.forRoot(), ElCardsModule.forRoot(), ElDropdownModule.forRoot(),
            ElBreadcrumbsModule.forRoot(), ElDateModule.forRoot(), ElSliderModule.forRoot(), ElDialogModule.forRoot(),
            ElCarouselModule.forRoot(), ElCollapseModule.forRoot(), ElAlertModule.forRoot(), ElPaginationModule.forRoot(),
            ElUploadModule.forRoot(), ElTableModule.forRoot(), ElContainerModule.forRoot(), ElFormModule.forRoot(),
            ElTreeModule.forRoot(),
        ], ElButtonsModule, ElIconsModule, ElRadiosModule, ElMenusModule, ElTooltipModule, ElRowModule,
        ElColModule, ElCheckboxsModule, ElInputsModule, ElInputNumberModule, ElTagsModule, ElSelectModule,
        ElSwitchModule, ElRateModule, ElProgressModule, ElStepsModule, ElLoadingModule, ElMessagesModule,
        ElSharedModule, ElNotificationModule, ElCascaderModule, ElBadgesModule, ElCardsModule, ElDropdownModule,
        ElBreadcrumbsModule, ElDateModule, ElSliderModule, ElDialogModule, ElCarouselModule, ElCollapseModule,
        ElAlertModule, ElPaginationModule, ElUploadModule, ElTableModule, ElContainerModule, ElFormModule,
        ElTreeModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElModule, { imports: [ɵngcc1.ElButtonsModule, ɵngcc2.ElIconsModule, ɵngcc3.ElRadiosModule, ɵngcc4.ElMenusModule, ɵngcc5.ElTooltipModule, ɵngcc6.ElRowModule, ɵngcc7.ElColModule, ɵngcc8.ElCheckboxsModule, ɵngcc9.ElInputsModule, ɵngcc10.ElInputNumberModule, ɵngcc11.ElTagsModule, ɵngcc12.ElSelectModule, ɵngcc13.ElSwitchModule, ɵngcc14.ElRateModule, ɵngcc15.ElProgressModule, ɵngcc16.ElStepsModule, ɵngcc17.ElLoadingModule, ɵngcc18.ElMessagesModule, ɵngcc19.ElSharedModule, ɵngcc20.ElNotificationModule, ɵngcc21.ElCascaderModule, ɵngcc22.ElBadgesModule, ɵngcc23.ElCardsModule, ɵngcc24.ElDropdownModule, ɵngcc25.ElBreadcrumbsModule, ɵngcc26.ElDateModule, ɵngcc27.ElSliderModule, ɵngcc28.ElDialogModule, ɵngcc29.ElCarouselModule, ɵngcc30.ElCollapseModule, ɵngcc31.ElAlertModule, ɵngcc32.ElPaginationModule, ɵngcc33.ElUploadModule, ɵngcc34.ElTableModule, ɵngcc35.ElContainerModule, ɵngcc36.ElFormModule, ɵngcc37.ElTreeModule], exports: [ElButtonsModule, ElIconsModule, ElRadiosModule, ElMenusModule, ElTooltipModule, ElRowModule,
        ElColModule, ElCheckboxsModule, ElInputsModule, ElInputNumberModule, ElTagsModule, ElSelectModule,
        ElSwitchModule, ElRateModule, ElProgressModule, ElStepsModule, ElLoadingModule, ElMessagesModule,
        ElSharedModule, ElNotificationModule, ElCascaderModule, ElBadgesModule, ElCardsModule, ElDropdownModule,
        ElBreadcrumbsModule, ElDateModule, ElSliderModule, ElDialogModule, ElCarouselModule, ElCollapseModule,
        ElAlertModule, ElPaginationModule, ElUploadModule, ElTableModule, ElContainerModule, ElFormModule,
        ElTreeModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElModule, [{
        type: NgModule,
        args: [{
                imports: [
                    ElButtonsModule.forRoot(), ElIconsModule.forRoot(), ElRadiosModule.forRoot(), ElMenusModule.forRoot(),
                    ElTooltipModule.forRoot(), ElRowModule.forRoot(), ElColModule.forRoot(), ElCheckboxsModule.forRoot(),
                    ElInputsModule.forRoot(), ElInputNumberModule.forRoot(), ElTagsModule.forRoot(), ElSelectModule.forRoot(),
                    ElSwitchModule.forRoot(), ElRateModule.forRoot(), ElProgressModule.forRoot(), ElStepsModule.forRoot(),
                    ElLoadingModule.forRoot(), ElMessagesModule.forRoot(), ElSharedModule.forRoot(), ElNotificationModule.forRoot(),
                    ElCascaderModule.forRoot(), ElBadgesModule.forRoot(), ElCardsModule.forRoot(), ElDropdownModule.forRoot(),
                    ElBreadcrumbsModule.forRoot(), ElDateModule.forRoot(), ElSliderModule.forRoot(), ElDialogModule.forRoot(),
                    ElCarouselModule.forRoot(), ElCollapseModule.forRoot(), ElAlertModule.forRoot(), ElPaginationModule.forRoot(),
                    ElUploadModule.forRoot(), ElTableModule.forRoot(), ElContainerModule.forRoot(), ElFormModule.forRoot(),
                    ElTreeModule.forRoot(),
                ],
                exports: ELMODULES_GROUP
            }]
    }], function () { return []; }, null); })();
    return ElModule;
}());
function ElModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElModule.ctorParameters;
}
export { ElModule, ElMessageService, ElNotificationService, };

export {ElCSSValuePipe} from './shared/pipe/css-value';
export {ElClassDirective} from './shared/directives/class.directive';
export {ElButton} from './button/button';
export {ElButtonGroup} from './button/button-group';
export {ElCheckbox} from './checkbox/checkbox';
export {ElCheckboxButton} from './checkbox/checkbox-button';
export {ElCheckboxGroup} from './checkbox/checkbox-group';
export {ElIcon} from './icon/icon';
export {ElRadio} from './radio/radio';
export {ElRadioGroup} from './radio/radio-group';
export {ElRadioButton} from './radio/radio-button';
export {ElTooltip} from './tooltip/tooltip';
export {ElMenu} from './menu/menu';
export {ElSubmenu} from './menu/submenu';
export {ElMenuItem} from './menu/menu-item';
export {ElMenuItemGroup} from './menu/menu-item-group';
export {ElRowDirective} from './row/row.directive';
export {ElColDirective} from './col/col.directive';
export {ElInput} from './input/input';
export {ElInputNumber} from './input-number/input-number';
export {ElTag} from './tag/tag';
export {ElSelectDropdown} from './select/select-dropdown';
export {ElOption} from './select/option';
export {ElSelect} from './select/select';
export {ElSwitch} from './switch/switch';
export {ElRate} from './rate/rate';
export {Elprogress} from './progress/progress';
export {ElStep} from './steps/step';
export {ElSteps} from './steps/steps';
export {ElLoadingDirective} from './loading/loading.directive';
export {ElMessageContainer} from './message/message';
export {ElNotificationContainer} from './notification/notification';
export {ElCascader} from './cascader/cascader';
export {ElCascaderMenu} from './cascader/cascader.menu';
export {ElBadge} from './badge/badge';
export {ElCard} from './card/card';
export {ElDropdown} from './dropdown/dropdown';
export {ElDropdownItem} from './dropdown/dropdown.item';
export {ElBreadcrumb} from './breadcrumb/breadcrumb';
export {ElBreadcrumbItem} from './breadcrumb/breadcrumb-item';
export {ElDataPicker} from './date-picker/picker';
export {ElDatePickerPanel} from './date-picker/picker-panel';
export {ElDateTable} from './date-picker/children/date-table';
export {ElYearTable} from './date-picker/children/year-table';
export {EMonthTable} from './date-picker/children/month-table';
export {ElSlider} from './slider/slider';
export {ElSliderButton} from './slider/slider-button';
export {ElDialog} from './dialog/dialog';
export {ElCarousel} from './carousel/carousel';
export {ElCarouselItem} from './carousel/carousel-item';
export {ElCollapse} from './collapse/collapse';
export {ElCollapseItem} from './collapse/collapse-item';
export {ElAlert} from './alert/alert';
export {ElPagination} from './pagination/pagination';
export {ElUpload} from './upload/upload';
export {ElTable} from './table/table';
export {ElTableColumn} from './table/children/column';
export {ElContainer} from './container/container';
export {ElHeader} from './container/children/header';
export {ElAside} from './container/children/aside';
export {ElMain} from './container/children/main';
export {ElFooter} from './container/children/footer';
export {ElContainerDirective} from './container/container.directive';
export {ElHeaderDirective} from './container/directives/header.directive';
export {ElMainDirective} from './container/directives/main.directive';
export {ElFooterDirective} from './container/directives/footer.directive';
export {ElAsideDirective} from './container/directives/aside.directive';
export {ElForm} from './form/form';
export {ElFormItem} from './form/form-item';
export {ElTree} from './tree/tree';
export {ElTreeItem} from './tree/tree-item';
export {ElButtonsModule} from './button/module';
export {ElIconsModule} from './icon/module';
export {ElRadiosModule} from './radio/module';
export {ElMenusModule} from './menu/module';
export {ElTooltipModule} from './tooltip/module';
export {ElRowModule} from './row/module';
export {ElColModule} from './col/module';
export {ElCheckboxsModule} from './checkbox/module';
export {ElInputsModule} from './input/module';
export {ElInputNumberModule} from './input-number/module';
export {ElTagsModule} from './tag/module';
export {ElSelectModule} from './select/module';
export {ElSwitchModule} from './switch/module';
export {ElRateModule} from './rate/module';
export {ElProgressModule} from './progress/module';
export {ElStepsModule} from './steps/module';
export {ElLoadingModule} from './loading/module';
export {ElMessagesModule} from './message/module';
export {ElSharedModule} from './shared/module';
export {ElNotificationModule} from './notification/module';
export {ElCascaderModule} from './cascader/module';
export {ElBadgesModule} from './badge/module';
export {ElCardsModule} from './card/module';
export {ElDropdownModule} from './dropdown/module';
export {ElBreadcrumbsModule} from './breadcrumb/module';
export {ElDateModule} from './date-picker/module';
export {ElSliderModule} from './slider/module';
export {ElDialogModule} from './dialog/module';
export {ElCarouselModule} from './carousel/module';
export {ElCollapseModule} from './collapse/module';
export {ElAlertModule} from './alert/module';
export {ElPaginationModule} from './pagination/module';
export {ElUploadModule} from './upload/module';
export {ElTableModule} from './table/module';
export {ElContainerModule} from './container/module';
export {ElFormModule} from './form/module';
export {ElTreeModule} from './tree/module';
