// element-angular/__ivy_ngcc__/release/card/card.js
import { Component, Input, ChangeDetectionStrategy, ContentChild } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';
import * as ɵngcc2 from '@angular/common';

var _c0 = ["header"];
function ElCard_div_1_ng_container_1_ng_template_1_Template(rf, ctx) { }
function ElCard_div_1_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, ElCard_div_1_ng_container_1_ng_template_1_Template, 0, 0, "ng-template", 5);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r1.header);
} }
function ElCard_div_1_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r2.headerStr, " ");
} }
function ElCard_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵtemplate(1, ElCard_div_1_ng_container_1_Template, 2, 1, "ng-container", 4);
    ɵngcc0.ɵɵtemplate(2, ElCard_div_1_ng_container_2_Template, 2, 1, "ng-container", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.header);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.header);
} }
var _c1 = ["*"];
var ElCard = /** @class */ (function () {
    /**
     * @param {?} sanitizer
     */
    function ElCard(sanitizer) {
        this.sanitizer = sanitizer;
        this.bodyStyle = '';
    }
    /**
     * @return {?}
     */
    ElCard.prototype.makeSafebodyStyle = function () {
        return this.sanitizer.bypassSecurityTrustStyle(this.bodyStyle);
    };
    /**
     * @nocollapse
     */
    ElCard.ctorParameters = function () { return [
        { type: DomSanitizer, },
    ]; };
    ElCard.propDecorators = {
        'header': [{ type: ContentChild, args: ['header',] },],
        'headerStr': [{ type: Input, args: ['header',] },],
        'bodyStyle': [{ type: Input, args: ['body-style',] },],
    };
ElCard.ɵfac = function ElCard_Factory(t) { return new (t || ElCard)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer)); };
ElCard.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElCard, selectors: [["el-card"]], contentQueries: function ElCard_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.header = _t.first);
    } }, inputs: { bodyStyle: ["body-style", "bodyStyle"], headerStr: ["header", "headerStr"] }, ngContentSelectors: _c1, decls: 4, vars: 3, consts: [[1, "el-card"], ["class", "el-card__header", 4, "ngIf"], [1, "el-card__body"], [1, "el-card__header"], [4, "ngIf"], [3, "ngTemplateOutlet"]], template: function ElCard_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, ElCard_div_1_Template, 3, 2, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵprojection(3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.header || ctx.headerStr);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleMap(ctx.makeSafebodyStyle());
    } }, directives: [ɵngcc2.NgIf, ɵngcc2.NgTemplateOutlet], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCard, [{
        type: Component,
        args: [{
                selector: 'el-card',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: "\n    <div class=\"el-card\">\n      <div class=\"el-card__header\" *ngIf=\"header || headerStr\">\n        <ng-container *ngIf=\"header\">\n          <ng-template [ngTemplateOutlet]=\"header\">\n          </ng-template>\n        </ng-container>\n        <ng-container *ngIf=\"!header\">\n          {{headerStr}}\n        </ng-container>\n      </div>\n      <div class=\"el-card__body\" [style]=\"makeSafebodyStyle()\">\n        <ng-content></ng-content>\n      </div>\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.DomSanitizer }]; }, { bodyStyle: [{
            type: Input,
            args: ['body-style']
        }], header: [{
            type: ContentChild,
            args: ['header']
        }], headerStr: [{
            type: Input,
            args: ['header']
        }] }); })();
    return ElCard;
}());
export { ElCard };
function ElCard_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCard.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCard.ctorParameters;
    /** @type {?} */
    ElCard.propDecorators;
    /** @type {?} */
    ElCard.prototype.header;
    /** @type {?} */
    ElCard.prototype.headerStr;
    /** @type {?} */
    ElCard.prototype.bodyStyle;
    /** @type {?} */
    ElCard.prototype.sanitizer;
}
