// element-angular/__ivy_ngcc__/release/progress/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Elprogress } from './progress';
import * as ɵngcc0 from '@angular/core';
var ElProgressModule = /** @class */ (function () {
    function ElProgressModule() {
    }
    /**
     * @return {?}
     */
    ElProgressModule.forRoot = function () {
        return { ngModule: ElProgressModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElProgressModule.ctorParameters = function () { return []; };
ElProgressModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElProgressModule });
ElProgressModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElProgressModule_Factory(t) { return new (t || ElProgressModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElProgressModule, { declarations: function () { return [Elprogress]; }, imports: function () { return [CommonModule]; }, exports: function () { return [Elprogress]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElProgressModule, [{
        type: NgModule,
        args: [{
                declarations: [Elprogress],
                exports: [Elprogress],
                imports: [CommonModule],
                entryComponents: [Elprogress]
            }]
    }], function () { return []; }, null); })();
    return ElProgressModule;
}());
export { ElProgressModule };
function ElProgressModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElProgressModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElProgressModule.ctorParameters;
}
