// element-angular/__ivy_ngcc__/release/checkbox/checkbox-button.js
import { Component, Input, Output, EventEmitter, ElementRef, Optional, ViewChild, ViewEncapsulation, } from '@angular/core';
import { ElCheckboxGroup } from './checkbox-group';
import { isParentTag, removeNgTag } from '../shared/utils/index';
import { DomSanitizer } from '@angular/platform-browser';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './checkbox-group';
import * as ɵngcc2 from '@angular/platform-browser';
import * as ɵngcc3 from '@angular/forms';
import * as ɵngcc4 from '@angular/common';

var _c0 = ["content"];
function ElCheckboxButton_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.label);
} }
function ElCheckboxButton_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", null, 4);
    ɵngcc0.ɵɵprojection(2);
    ɵngcc0.ɵɵelementEnd();
} }
var _c1 = ["*"];
var ElCheckboxButton = /** @class */ (function () {
    /**
     * @param {?} hostGroup
     * @param {?} el
     * @param {?} domSanitizer
     */
    function ElCheckboxButton(hostGroup, el, domSanitizer) {
        this.hostGroup = hostGroup;
        this.el = el;
        this.domSanitizer = domSanitizer;
        this.elDisabled = false;
        this.indeterminate = false;
        this.checked = false;
        this.modelChange = new EventEmitter();
        this.parentIsGroup = false;
        this.isFocus = false;
        this.showLabel = false;
    }
    Object.defineProperty(ElCheckboxButton.prototype, "disabled", {
        /**
         * @param {?} val
         * @return {?}
         */
        set: function (val) {
            console.warn('Element Angular: (disabled) is discarded, use [elDisabled] replace it.');
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    ElCheckboxButton.prototype.activeStyle = function () {
        if (!this.hostGroup)
            return this.domSanitizer.bypassSecurityTrustStyle('');
        var /** @type {?} */ styles = "backgroundColor: " + this.hostGroup.fill + ";" +
            ("borderColor: " + this.hostGroup.fill + ";color: " + this.hostGroup.textColor + ";") +
            ("box-shadow: -1px 0 0 0 " + this.hostGroup.fill + ";");
        return this.domSanitizer.bypassSecurityTrustStyle(this.checked && !this.elDisabled ? styles : '');
    };
    /**
     * @return {?}
     */
    ElCheckboxButton.prototype.isChecked = function () {
        if (this.parentIsGroup) {
            return this.hostGroup.model.indexOf(this.label) > -1;
        }
        return this.model;
    };
    /**
     * @param {?} t
     * @return {?}
     */
    ElCheckboxButton.prototype.changeHandle = function (t) {
        this.parentIsGroup && this.hostGroup.updateModelFromChildren(t, this.label);
        this.model = t;
        this.checked = this.isChecked();
    };
    /**
     * @return {?}
     */
    ElCheckboxButton.prototype.ngOnInit = function () {
        var _this = this;
        var /** @type {?} */ nativeElement = this.el.nativeElement;
        this.parentIsGroup = isParentTag(nativeElement, 'el-checkbox-group');
        removeNgTag(nativeElement);
        // update model from group
        if (this.parentIsGroup) {
            this.labels = this.hostGroup.model;
            this.size = this.hostGroup.size;
            this.model = this.isChecked();
            // update handle
            this.hostGroup.subscriber.push(function () { return _this.model = _this.isChecked(); });
        }
        this.checked = this.isChecked();
    };
    /**
     * @return {?}
     */
    ElCheckboxButton.prototype.ngAfterViewInit = function () {
        var _this = this;
        var /** @type {?} */ contentText = this.content && this.content.nativeElement.innerText;
        setTimeout(function () {
            _this.showLabel = !contentText || contentText.length < 1;
        }, 0);
    };
    /**
     * @nocollapse
     */
    ElCheckboxButton.ctorParameters = function () { return [
        { type: ElCheckboxGroup, decorators: [{ type: Optional },] },
        { type: ElementRef, },
        { type: DomSanitizer, },
    ]; };
    ElCheckboxButton.propDecorators = {
        'content': [{ type: ViewChild, args: ['content',] },],
        'disabled': [{ type: Input },],
        'elDisabled': [{ type: Input },],
        'label': [{ type: Input },],
        'model': [{ type: Input },],
        'indeterminate': [{ type: Input },],
        'checked': [{ type: Input },],
        'name': [{ type: Input },],
        'trueLabel': [{ type: Input, args: ['true-label',] },],
        'modelChange': [{ type: Output },],
    };
ElCheckboxButton.ɵfac = function ElCheckboxButton_Factory(t) { return new (t || ElCheckboxButton)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElCheckboxGroup, 8), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.DomSanitizer)); };
ElCheckboxButton.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElCheckboxButton, selectors: [["el-checkbox-button"]], viewQuery: function ElCheckboxButton_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.content = _t.first);
    } }, inputs: { elDisabled: "elDisabled", indeterminate: "indeterminate", checked: "checked", disabled: "disabled", model: "model", label: "label", name: "name", trueLabel: ["true-label", "trueLabel"] }, outputs: { modelChange: "modelChange" }, ngContentSelectors: _c1, decls: 5, vars: 18, consts: [["role", "checkbox"], ["type", "checkbox", 1, "el-checkbox-button__original", 3, "disabled", "value", "name", "ngModel", "ngModelChange", "focus", "blur"], [1, "el-checkbox-button__inner"], [4, "ngIf"], ["content", ""]], template: function ElCheckboxButton_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "label", 0);
        ɵngcc0.ɵɵelementStart(1, "input", 1);
        ɵngcc0.ɵɵlistener("ngModelChange", function ElCheckboxButton_Template_input_ngModelChange_1_listener($event) { return ctx.changeHandle($event); })("focus", function ElCheckboxButton_Template_input_focus_1_listener() { return ctx.isFocus = true; })("blur", function ElCheckboxButton_Template_input_blur_1_listener() { return ctx.isFocus = false; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(2, "span", 2);
        ɵngcc0.ɵɵtemplate(3, ElCheckboxButton_ng_container_3_Template, 2, 1, "ng-container", 3);
        ɵngcc0.ɵɵtemplate(4, ElCheckboxButton_span_4_Template, 3, 0, "span", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-checkbox-button" + (ctx.size ? " el-checkbox-button--" + ctx.size : ""));
        ɵngcc0.ɵɵclassProp("is-disabled", ctx.elDisabled)("is-focus", ctx.isFocus)("is-indeterminate", ctx.indeterminate)("is-checked", ctx.checked);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("disabled", ctx.elDisabled)("value", ctx.label)("name", ctx.name)("ngModel", ctx.model);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleMap(ctx.activeStyle());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showLabel);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.showLabel);
    } }, directives: [ɵngcc3.CheckboxControlValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel, ɵngcc4.NgIf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCheckboxButton, [{
        type: Component,
        args: [{
                selector: 'el-checkbox-button',
                template: "\n    <label [class]=\"'el-checkbox-button' + (size ? ' el-checkbox-button--' + size : '') \"\n      role=\"checkbox\"\n      [class.is-disabled]=\"elDisabled\" [class.is-focus]=\"isFocus\"\n      [class.is-indeterminate]=\"indeterminate\" [class.is-checked]=\"checked\">\n      <input class=\"el-checkbox-button__original\" type=\"checkbox\"\n        [disabled]=\"elDisabled\" [value]=\"label\" [name]=\"name\"\n        [ngModel]=\"model\" (ngModelChange)=\"changeHandle($event)\"\n        (focus)=\"isFocus = true\" (blur)=\"isFocus = false\">\n      <span class=\"el-checkbox-button__inner\"\n        [style]=\"activeStyle()\">\n        <ng-container *ngIf=\"showLabel\">{{label}}</ng-container>\n        <span *ngIf=\"!showLabel\" #content>\n          <ng-content></ng-content>\n        </span>\n      </span>\n    </label>\n  ",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.ElCheckboxGroup, decorators: [{
                type: Optional
            }] }, { type: ɵngcc0.ElementRef }, { type: ɵngcc2.DomSanitizer }]; }, { elDisabled: [{
            type: Input
        }], indeterminate: [{
            type: Input
        }], checked: [{
            type: Input
        }], modelChange: [{
            type: Output
        }], disabled: [{
            type: Input
        }], model: [{
            type: Input
        }], content: [{
            type: ViewChild,
            args: ['content']
        }], label: [{
            type: Input
        }], name: [{
            type: Input
        }], trueLabel: [{
            type: Input,
            args: ['true-label']
        }] }); })();
    return ElCheckboxButton;
}());
export { ElCheckboxButton };
function ElCheckboxButton_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCheckboxButton.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCheckboxButton.ctorParameters;
    /** @type {?} */
    ElCheckboxButton.propDecorators;
    /** @type {?} */
    ElCheckboxButton.prototype.content;
    /** @type {?} */
    ElCheckboxButton.prototype.elDisabled;
    /** @type {?} */
    ElCheckboxButton.prototype.label;
    /** @type {?} */
    ElCheckboxButton.prototype.model;
    /** @type {?} */
    ElCheckboxButton.prototype.indeterminate;
    /** @type {?} */
    ElCheckboxButton.prototype.checked;
    /** @type {?} */
    ElCheckboxButton.prototype.name;
    /** @type {?} */
    ElCheckboxButton.prototype.trueLabel;
    /** @type {?} */
    ElCheckboxButton.prototype.modelChange;
    /** @type {?} */
    ElCheckboxButton.prototype.labels;
    /** @type {?} */
    ElCheckboxButton.prototype.parentIsGroup;
    /** @type {?} */
    ElCheckboxButton.prototype.isFocus;
    /** @type {?} */
    ElCheckboxButton.prototype.showLabel;
    /** @type {?} */
    ElCheckboxButton.prototype.size;
    /** @type {?} */
    ElCheckboxButton.prototype.hostGroup;
    /** @type {?} */
    ElCheckboxButton.prototype.el;
    /** @type {?} */
    ElCheckboxButton.prototype.domSanitizer;
}
