// element-angular/__ivy_ngcc__/release/radio/radio.js
import { Component, Input, Output, EventEmitter, ElementRef, Optional, forwardRef, } from '@angular/core';
import { ElRadioGroup } from './radio-group';
import { isParentTag, removeNgTag } from '../shared/utils/index';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './radio-group';
import * as ɵngcc2 from '@angular/forms';

var _c0 = ["*"];
var ElRadio = /** @class */ (function () {
    /**
     * @param {?} rootGroup
     * @param {?} el
     */
    function ElRadio(rootGroup, el) {
        this.rootGroup = rootGroup;
        this.el = el;
        this.elDisabled = false;
        this.label = '';
        this.nativeName = '';
        this.modelChange = new EventEmitter();
        this.isFocus = false;
        this.parentIsGroup = false;
        this.controlChange = function () { };
        this.controlTouch = function () { };
        console.log(this.elDisabled);
    }
    Object.defineProperty(ElRadio.prototype, "disabled", {
        /**
         * @param {?} val
         * @return {?}
         */
        set: function (val) {
            console.warn('Element Angular: (disabled) is discarded, use [elDisabled] replace it.');
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    ElRadio.prototype.changeHandle = function () {
        if (this.parentIsGroup) {
            return this.rootGroup.changeHandle(this.label);
        }
        this.model = this.label;
        this.modelChange.emit(this.label);
        this.controlChange(this.label);
    };
    /**
     * @return {?}
     */
    ElRadio.prototype.ngOnInit = function () {
        var _this = this;
        var /** @type {?} */ nativeElement = this.el.nativeElement;
        var /** @type {?} */ update = function () {
            if (_this.rootGroup.elDisabled) {
                _this.elDisabled = _this.rootGroup.elDisabled;
            }
            _this.model = _this.rootGroup.model;
        };
        this.parentIsGroup = isParentTag(nativeElement, 'el-radio-group');
        removeNgTag(nativeElement);
        if (this.parentIsGroup && this.rootGroup) {
            update();
            this.rootGroup.subscriber.push(update);
        }
    };
    /**
     * @param {?} value
     * @return {?}
     */
    ElRadio.prototype.writeValue = function (value) {
        this.model = value;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElRadio.prototype.registerOnChange = function (fn) {
        this.controlChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElRadio.prototype.registerOnTouched = function (fn) {
        this.controlTouch = fn;
    };
    /**
     * @nocollapse
     */
    ElRadio.ctorParameters = function () { return [
        { type: ElRadioGroup, decorators: [{ type: Optional },] },
        { type: ElementRef, },
    ]; };
    ElRadio.propDecorators = {
        'disabled': [{ type: Input },],
        'elDisabled': [{ type: Input },],
        'label': [{ type: Input },],
        'nativeName': [{ type: Input, args: ['name',] },],
        'model': [{ type: Input },],
        'modelChange': [{ type: Output },],
    };
ElRadio.ɵfac = function ElRadio_Factory(t) { return new (t || ElRadio)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElRadioGroup, 8), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElRadio.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElRadio, selectors: [["el-radio"]], inputs: { elDisabled: "elDisabled", label: "label", nativeName: ["name", "nativeName"], disabled: "disabled", model: "model" }, outputs: { modelChange: "modelChange" }, features: [ɵngcc0.ɵɵProvidersFeature([{
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return ElRadio; }),
                multi: true
            }])], ngContentSelectors: _c0, decls: 6, vars: 10, consts: [["role", "radio", "tabindex", "0", 1, "el-radio"], [1, "el-radio__input", 2, "float", "left"], [1, "el-radio__inner"], ["type", "radio", 1, "el-radio__original", 3, "value", "name", "disabled", "ngModel", "focus", "blur", "ngModelChange"], [1, "el-radio__label"]], template: function ElRadio_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "label", 0);
        ɵngcc0.ɵɵelementStart(1, "span", 1);
        ɵngcc0.ɵɵelement(2, "span", 2);
        ɵngcc0.ɵɵelementStart(3, "input", 3);
        ɵngcc0.ɵɵlistener("focus", function ElRadio_Template_input_focus_3_listener() { return ctx.isFocus = true; })("blur", function ElRadio_Template_input_blur_3_listener() { return ctx.isFocus = false; })("ngModelChange", function ElRadio_Template_input_ngModelChange_3_listener() { return ctx.changeHandle(); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "span", 4);
        ɵngcc0.ɵɵprojection(5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassProp("is-disabled", ctx.elDisabled)("is-checked", ctx.model === ctx.label)("is-focus", ctx.isFocus);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("value", ctx.label)("name", ctx.nativeName)("disabled", ctx.elDisabled)("ngModel", ctx.model);
    } }, directives: [ɵngcc2.RadioControlValueAccessor, ɵngcc2.DefaultValueAccessor, ɵngcc2.NgControlStatus, ɵngcc2.NgModel], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRadio, [{
        type: Component,
        args: [{
                selector: 'el-radio',
                providers: [{
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(function () { return ElRadio; }),
                        multi: true
                    }],
                template: "\n    <label class=\"el-radio\" role=\"radio\" tabindex=\"0\">\n      <span class=\"el-radio__input\"\n        style=\"float: left;\"\n        [class.is-disabled]=\"elDisabled\"\n        [class.is-checked]=\"model === label\"\n        [class.is-focus]=\"isFocus\">\n        <span class=\"el-radio__inner\"></span>\n        <input class=\"el-radio__original\" type=\"radio\"\n          [value]=\"label\" [name]=\"nativeName\" [disabled]=\"elDisabled\"\n          (focus)=\"isFocus = true\" (blur)=\"isFocus = false\"\n          [ngModel]=\"model\" (ngModelChange)=\"changeHandle()\">\n      </span>\n      <span class=\"el-radio__label\"><ng-content></ng-content></span>\n    </label>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElRadioGroup, decorators: [{
                type: Optional
            }] }, { type: ɵngcc0.ElementRef }]; }, { elDisabled: [{
            type: Input
        }], label: [{
            type: Input
        }], nativeName: [{
            type: Input,
            args: ['name']
        }], modelChange: [{
            type: Output
        }], disabled: [{
            type: Input
        }], model: [{
            type: Input
        }] }); })();
    return ElRadio;
}());
export { ElRadio };
function ElRadio_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRadio.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRadio.ctorParameters;
    /** @type {?} */
    ElRadio.propDecorators;
    /** @type {?} */
    ElRadio.prototype.elDisabled;
    /** @type {?} */
    ElRadio.prototype.label;
    /** @type {?} */
    ElRadio.prototype.nativeName;
    /** @type {?} */
    ElRadio.prototype.model;
    /** @type {?} */
    ElRadio.prototype.modelChange;
    /** @type {?} */
    ElRadio.prototype.isFocus;
    /** @type {?} */
    ElRadio.prototype.parentIsGroup;
    /** @type {?} */
    ElRadio.prototype.controlChange;
    /** @type {?} */
    ElRadio.prototype.controlTouch;
    /** @type {?} */
    ElRadio.prototype.rootGroup;
    /** @type {?} */
    ElRadio.prototype.el;
}
