// element-angular/__ivy_ngcc__/release/radio/radio-group.js
import { Component, Input, EventEmitter, Output, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var ElRadioGroup = /** @class */ (function () {
    function ElRadioGroup() {
        this.elDisabled = false;
        this.fillColor = '#20a0ff';
        this.textColor = '#ffffff';
        this.modelChange = new EventEmitter();
        this.subscriber = [];
        this.controlChange = function () { };
        this.controlTouch = function () { };
    }
    Object.defineProperty(ElRadioGroup.prototype, "disabled", {
        /**
         * @param {?} val
         * @return {?}
         */
        set: function (val) {
            console.warn('Element Angular: (disabled) is discarded, use [elDisabled] replace it.');
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} nextValue
     * @return {?}
     */
    ElRadioGroup.prototype.changeHandle = function (nextValue) {
        var _this = this;
        setTimeout(function () {
            _this.model = nextValue;
            _this.modelChange.emit(nextValue);
            _this.controlChange(nextValue);
            _this.subscriber.forEach(function (sub) { return sub(); });
        }, 0);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    ElRadioGroup.prototype.writeValue = function (value) {
        this.model = value;
        this.changeHandle(this.model);
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElRadioGroup.prototype.registerOnChange = function (fn) {
        this.controlChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElRadioGroup.prototype.registerOnTouched = function (fn) {
        this.controlTouch = fn;
    };
    /**
     * @nocollapse
     */
    ElRadioGroup.ctorParameters = function () { return []; };
    ElRadioGroup.propDecorators = {
        'disabled': [{ type: Input },],
        'elDisabled': [{ type: Input },],
        'buttonSize': [{ type: Input, args: ['size',] },],
        'fillColor': [{ type: Input, args: ['fill',] },],
        'textColor': [{ type: Input },],
        'model': [{ type: Input },],
        'modelChange': [{ type: Output },],
    };
ElRadioGroup.ɵfac = function ElRadioGroup_Factory(t) { return new (t || ElRadioGroup)(); };
ElRadioGroup.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElRadioGroup, selectors: [["el-radio-group"]], inputs: { elDisabled: "elDisabled", fillColor: ["fill", "fillColor"], textColor: "textColor", disabled: "disabled", model: "model", buttonSize: ["size", "buttonSize"] }, outputs: { modelChange: "modelChange" }, features: [ɵngcc0.ɵɵProvidersFeature([{
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return ElRadioGroup; }),
                multi: true
            }])], ngContentSelectors: _c0, decls: 2, vars: 0, consts: [["role", "radiogroup", 1, "el-radio-group"]], template: function ElRadioGroup_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRadioGroup, [{
        type: Component,
        args: [{
                selector: 'el-radio-group',
                providers: [{
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(function () { return ElRadioGroup; }),
                        multi: true
                    }],
                template: "\n    <div class=\"el-radio-group\" role=\"radiogroup\">\n      <ng-content></ng-content>\n    </div>\n  "
            }]
    }], function () { return []; }, { elDisabled: [{
            type: Input
        }], fillColor: [{
            type: Input,
            args: ['fill']
        }], textColor: [{
            type: Input
        }], modelChange: [{
            type: Output
        }], disabled: [{
            type: Input
        }], model: [{
            type: Input
        }], buttonSize: [{
            type: Input,
            args: ['size']
        }] }); })();
    return ElRadioGroup;
}());
export { ElRadioGroup };
function ElRadioGroup_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRadioGroup.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRadioGroup.ctorParameters;
    /** @type {?} */
    ElRadioGroup.propDecorators;
    /** @type {?} */
    ElRadioGroup.prototype.elDisabled;
    /** @type {?} */
    ElRadioGroup.prototype.buttonSize;
    /** @type {?} */
    ElRadioGroup.prototype.fillColor;
    /** @type {?} */
    ElRadioGroup.prototype.textColor;
    /** @type {?} */
    ElRadioGroup.prototype.model;
    /** @type {?} */
    ElRadioGroup.prototype.modelChange;
    /** @type {?} */
    ElRadioGroup.prototype.subscriber;
    /** @type {?} */
    ElRadioGroup.prototype.controlChange;
    /** @type {?} */
    ElRadioGroup.prototype.controlTouch;
}
