// element-angular/__ivy_ngcc__/release/input-number/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ElInputNumber } from './input-number';
import * as ɵngcc0 from '@angular/core';
var ElInputNumberModule = /** @class */ (function () {
    function ElInputNumberModule() {
    }
    /**
     * @return {?}
     */
    ElInputNumberModule.forRoot = function () {
        return { ngModule: ElInputNumberModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElInputNumberModule.ctorParameters = function () { return []; };
ElInputNumberModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElInputNumberModule });
ElInputNumberModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElInputNumberModule_Factory(t) { return new (t || ElInputNumberModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElInputNumberModule, { declarations: function () { return [ElInputNumber]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [ElInputNumber]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElInputNumberModule, [{
        type: NgModule,
        args: [{
                declarations: [ElInputNumber],
                exports: [ElInputNumber],
                imports: [CommonModule, FormsModule],
                entryComponents: [ElInputNumber]
            }]
    }], function () { return []; }, null); })();
    return ElInputNumberModule;
}());
export { ElInputNumberModule };
function ElInputNumberModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElInputNumberModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElInputNumberModule.ctorParameters;
}
