// element-angular/__ivy_ngcc__/release/message/message.js
import { Component } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import * as Icons from './images';
import { MessageAnimation } from './message.animation';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';
import * as ɵngcc2 from '@angular/common';

function ElMessageContainer_i_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "i");
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(ctx_r0.iconClass);
} }
function ElMessageContainer_i_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "i");
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(ctx_r1.makeTypeClass());
} }
function ElMessageContainer_div_5_Template(rf, ctx) { if (rf & 1) {
    var _r4 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵlistener("click", function ElMessageContainer_div_5_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r4); var ctx_r3 = ɵngcc0.ɵɵnextContext(); return ctx_r3.close(); });
    ɵngcc0.ɵɵelementEnd();
} }
var _c0 = function (a0) { return { "z-index": a0 }; };
var ElMessageContainer = /** @class */ (function () {
    /**
     * @param {?} sanitizer
     */
    function ElMessageContainer(sanitizer) {
        this.sanitizer = sanitizer;
        this.showClose = false;
        this.type = 'info';
        this.center = false;
        this.duration = 3000;
        // user setting
        this.iconClass = '';
        this.customClass = '';
        this.zIndex = 1000;
        this.message = '';
        this.showBox = false;
        this.onClose = function () { };
        this.onDestroy = function () { };
    }
    /**
     * @return {?}
     */
    ElMessageContainer.prototype.makeTypeClass = function () {
        return this.type && !this.iconClass ? "el-message__icon el-icon-" + this.type : '';
    };
    /**
     * @return {?}
     */
    ElMessageContainer.prototype.makeLink = function () {
        return this.sanitizer.bypassSecurityTrustUrl(Icons[this.type]);
    };
    /**
     * @param {?} message
     * @return {?}
     */
    ElMessageContainer.prototype.show = function (message) {
        var _this = this;
        this.message = message;
        this.showBox = true;
        this.timer = setTimeout(function () {
            _this.close();
        }, this.duration);
    };
    /**
     * @return {?}
     */
    ElMessageContainer.prototype.close = function () {
        this.timer && clearTimeout(this.timer);
        this.showBox = false;
        this.onClose();
        this.onDestroy();
    };
    /**
     * @return {?}
     */
    ElMessageContainer.prototype.startTimer = function () {
        var _this = this;
        if (!this.showBox)
            return;
        this.timer = setTimeout(function () {
            _this.close();
        }, this.duration);
    };
    /**
     * @return {?}
     */
    ElMessageContainer.prototype.clearTimer = function () {
        this.timer && clearTimeout(this.timer);
    };
    /**
     * @nocollapse
     */
    ElMessageContainer.ctorParameters = function () { return [
        { type: DomSanitizer, },
    ]; };
ElMessageContainer.ɵfac = function ElMessageContainer_Factory(t) { return new (t || ElMessageContainer)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer)); };
ElMessageContainer.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElMessageContainer, selectors: [["el-message-container"]], decls: 6, vars: 12, consts: [["role", "alertdialog", 2, "visibility", "hidden", "transition", "all .25s", 3, "ngStyle", "mouseenter", "mouseleave"], [3, "class", 4, "ngIf"], ["tabindex", "0", 1, "el-message__content"], ["class", "el-message__closeBtn el-icon-close", "role", "button", 3, "click", 4, "ngIf"], ["role", "button", 1, "el-message__closeBtn", "el-icon-close", 3, "click"]], template: function ElMessageContainer_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("mouseenter", function ElMessageContainer_Template_div_mouseenter_0_listener() { return ctx.clearTimer(); })("mouseleave", function ElMessageContainer_Template_div_mouseleave_0_listener() { return ctx.startTimer(); });
        ɵngcc0.ɵɵtemplate(1, ElMessageContainer_i_1_Template, 1, 2, "i", 1);
        ɵngcc0.ɵɵtemplate(2, ElMessageContainer_i_2_Template, 1, 2, "i", 1);
        ɵngcc0.ɵɵelementStart(3, "p", 2);
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(5, ElMessageContainer_div_5_Template, 1, 0, "div", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-message " + ctx.customClass + (ctx.type && !ctx.iconClass ? " el-message--" + ctx.type : ""));
        ɵngcc0.ɵɵclassProp("is-center", ctx.center);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(10, _c0, ctx.zIndex))("@messageAnimation", ctx.showBox);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.iconClass);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.iconClass);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.message);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showClose);
    } }, directives: [ɵngcc2.NgStyle, ɵngcc2.NgIf], styles: [""], data: { animation: [MessageAnimation] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElMessageContainer, [{
        type: Component,
        args: [{
                selector: 'el-message-container',
                styles: [""],
                template: "\n    <div [class]=\"'el-message ' + customClass + (type && !iconClass ? ' el-message--' + type : '')\"\n      [class.is-center]=\"center\"\n      style=\"visibility: hidden; transition: all .25s\" role=\"alertdialog\"\n      [ngStyle]=\"{ 'z-index': zIndex }\"\n      (mouseenter)=\"clearTimer()\" (mouseleave)=\"startTimer()\"\n      [@messageAnimation]=\"showBox\">\n      <i [class]=\"iconClass\" *ngIf=\"iconClass\"></i>\n      <i [class]=\"makeTypeClass()\" *ngIf=\"!iconClass\"></i>\n      \n      <p class=\"el-message__content\" tabindex=\"0\">{{ message }}</p>\n      <div *ngIf=\"showClose\" class=\"el-message__closeBtn el-icon-close\" (click)=\"close()\" role=\"button\"></div>\n    </div>\n  ",
                animations: [MessageAnimation]
            }]
    }], function () { return [{ type: ɵngcc1.DomSanitizer }]; }, null); })();
    return ElMessageContainer;
}());
export { ElMessageContainer };
function ElMessageContainer_tsickle_Closure_declarations() {
    /** @type {?} */
    ElMessageContainer.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElMessageContainer.ctorParameters;
    /** @type {?} */
    ElMessageContainer.prototype.id;
    /** @type {?} */
    ElMessageContainer.prototype.showClose;
    /** @type {?} */
    ElMessageContainer.prototype.type;
    /** @type {?} */
    ElMessageContainer.prototype.center;
    /** @type {?} */
    ElMessageContainer.prototype.duration;
    /** @type {?} */
    ElMessageContainer.prototype.iconClass;
    /** @type {?} */
    ElMessageContainer.prototype.customClass;
    /** @type {?} */
    ElMessageContainer.prototype.zIndex;
    /** @type {?} */
    ElMessageContainer.prototype.message;
    /** @type {?} */
    ElMessageContainer.prototype.showBox;
    /** @type {?} */
    ElMessageContainer.prototype.timer;
    /** @type {?} */
    ElMessageContainer.prototype.onClose;
    /** @type {?} */
    ElMessageContainer.prototype.onDestroy;
    /** @type {?} */
    ElMessageContainer.prototype.sanitizer;
}
