// element-angular/__ivy_ngcc__/release/message/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElMessageContainer } from './message';
import { ElMessageService } from './message.service';
import * as ɵngcc0 from '@angular/core';
var ElMessagesModule = /** @class */ (function () {
    function ElMessagesModule() {
    }
    /**
     * @return {?}
     */
    ElMessagesModule.forRoot = function () {
        return { ngModule: ElMessagesModule, providers: [
                ElMessageService,
            ] };
    };
    /**
     * @nocollapse
     */
    ElMessagesModule.ctorParameters = function () { return []; };
ElMessagesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElMessagesModule });
ElMessagesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElMessagesModule_Factory(t) { return new (t || ElMessagesModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElMessagesModule, { declarations: function () { return [ElMessageContainer]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElMessageContainer]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElMessagesModule, [{
        type: NgModule,
        args: [{
                declarations: [ElMessageContainer],
                exports: [ElMessageContainer],
                imports: [CommonModule],
                entryComponents: [ElMessageContainer]
            }]
    }], function () { return []; }, null); })();
    return ElMessagesModule;
}());
export { ElMessagesModule };
function ElMessagesModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElMessagesModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElMessagesModule.ctorParameters;
}
