// element-angular/__ivy_ngcc__/release/card/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElCard } from './card';
import * as ɵngcc0 from '@angular/core';
var ElCardsModule = /** @class */ (function () {
    function ElCardsModule() {
    }
    /**
     * @return {?}
     */
    ElCardsModule.forRoot = function () {
        return { ngModule: ElCardsModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElCardsModule.ctorParameters = function () { return []; };
ElCardsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElCardsModule });
ElCardsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElCardsModule_Factory(t) { return new (t || ElCardsModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElCardsModule, { declarations: function () { return [ElCard]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElCard]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCardsModule, [{
        type: NgModule,
        args: [{
                declarations: [ElCard],
                exports: [ElCard],
                imports: [CommonModule],
                entryComponents: [ElCard]
            }]
    }], function () { return []; }, null); })();
    return ElCardsModule;
}());
export { ElCardsModule };
function ElCardsModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCardsModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCardsModule.ctorParameters;
}
