// element-angular/__ivy_ngcc__/release/date-picker/picker-panel.js
import { Component, EventEmitter, Input, Optional, Output } from '@angular/core';
import { ElDataPicker } from './picker';
import { dropAnimation } from '../shared/animation/index';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './picker';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from './children/date-table';
import * as ɵngcc4 from './children/year-table';
import * as ɵngcc5 from './children/month-table';

function ElDatePickerPanel_button_5_Template(rf, ctx) { if (rf & 1) {
    var _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 14);
    ɵngcc0.ɵɵlistener("click", function ElDatePickerPanel_button_5_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r9); var ctx_r8 = ɵngcc0.ɵɵnextContext(); return ctx_r8.nextMonth(-1); });
    ɵngcc0.ɵɵelementEnd();
} }
function ElDatePickerPanel_span_6_Template(rf, ctx) { if (rf & 1) {
    var _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span", 15);
    ɵngcc0.ɵɵlistener("click", function ElDatePickerPanel_span_6_Template_span_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r11); var ctx_r10 = ɵngcc0.ɵɵnextContext(); return ctx_r10.showPicker("year"); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r1.dateShowModels.year, " \u5E74");
} }
function ElDatePickerPanel_span_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 16);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate2(" ", ctx_r2.dateShowModels.yearRange[0], " \u5E74 - ", ctx_r2.dateShowModels.yearRange[1], " \u5E74 ");
} }
function ElDatePickerPanel_span_8_Template(rf, ctx) { if (rf & 1) {
    var _r13 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span", 15);
    ɵngcc0.ɵɵlistener("click", function ElDatePickerPanel_span_8_Template_span_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r13); var ctx_r12 = ɵngcc0.ɵɵnextContext(); return ctx_r12.showPicker("month"); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("active", ctx_r3.currentView === "month");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r3.dateShowModels.month + 1, " \u6708");
} }
function ElDatePickerPanel_button_10_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 17);
    ɵngcc0.ɵɵlistener("click", function ElDatePickerPanel_button_10_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.nextMonth(1); });
    ɵngcc0.ɵɵelementEnd();
} }
function ElDatePickerPanel_el_date_table_12_Template(rf, ctx) { if (rf & 1) {
    var _r17 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-date-table", 18);
    ɵngcc0.ɵɵlistener("modelChange", function ElDatePickerPanel_el_date_table_12_Template_el_date_table_modelChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r17); var ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16.datePickChangeHandle($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("model", ctx_r5.model);
} }
function ElDatePickerPanel_el_year_table_13_Template(rf, ctx) { if (rf & 1) {
    var _r19 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-year-table", 19);
    ɵngcc0.ɵɵlistener("modelChange", function ElDatePickerPanel_el_year_table_13_Template_el_year_table_modelChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r19); var ctx_r18 = ɵngcc0.ɵɵnextContext(); return ctx_r18.yearPickChangeHandle($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("model", ctx_r6.model)("disabled-date", ctx_r6.disabledDate);
} }
function ElDatePickerPanel_el_month_table_14_Template(rf, ctx) { if (rf & 1) {
    var _r21 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-month-table", 19);
    ɵngcc0.ɵɵlistener("modelChange", function ElDatePickerPanel_el_month_table_14_Template_el_month_table_modelChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r21); var ctx_r20 = ɵngcc0.ɵɵnextContext(); return ctx_r20.monthPickChangeHandle($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r7 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("model", ctx_r7.model)("disabled-date", ctx_r7.disabledDate);
} }
var _c0 = function (a0, a1) { return { width: a0, "z-index": a1 }; };
var ElDatePickerPanel = /** @class */ (function () {
    /**
     * @param {?} root
     */
    function ElDatePickerPanel(root) {
        this.root = root;
        this.show = false;
        this.hiddenDay = false;
        this.panelAbsolute = true;
        this.panelIndex = 200;
        this.modelChange = new EventEmitter();
        this.shortcuts = false;
        this.showTime = false;
        this.currentView = 'date';
    }
    /**
     * @param {?} pickPath
     * @return {?}
     */
    ElDatePickerPanel.prototype.showPicker = function (pickPath) {
        this.currentView = pickPath;
    };
    /**
     * @return {?}
     */
    ElDatePickerPanel.prototype.updateDate = function () {
        var /** @type {?} */ date = new Date(this.model);
        var /** @type {?} */ startYear = ~~(date.getFullYear() / 10) * 10;
        this.dateShowModels = {
            month: date.getMonth(),
            year: date.getFullYear(),
            yearRange: [startYear, startYear + 10],
        };
    };
    /**
     * @param {?} time
     * @return {?}
     */
    ElDatePickerPanel.prototype.datePickChangeHandle = function (time) {
        this.model = time;
        this.modelChange.emit(time);
        this.updateDate();
    };
    /**
     * @param {?} time
     * @return {?}
     */
    ElDatePickerPanel.prototype.monthPickChangeHandle = function (time) {
        this.model = time;
        // hidden day, only show month
        if (this.hiddenDay) {
            this.modelChange.emit(time);
        }
        else {
            this.currentView = 'date';
        }
        this.updateDate();
    };
    /**
     * @param {?} time
     * @return {?}
     */
    ElDatePickerPanel.prototype.yearPickChangeHandle = function (time) {
        this.model = time;
        this.currentView = 'month';
        this.updateDate();
    };
    /**
     * @param {?} step
     * @return {?}
     */
    ElDatePickerPanel.prototype.nextYear = function (step) {
        // year table component opened, edit year range
        if (this.currentView === 'year') {
            step = step * 10;
        }
        var /** @type {?} */ date = new Date(this.model);
        date.setFullYear(this.dateShowModels.year + step);
        this.model = date.getTime();
        this.updateDate();
    };
    /**
     * @param {?} step
     * @return {?}
     */
    ElDatePickerPanel.prototype.nextMonth = function (step) {
        var /** @type {?} */ date = new Date(this.model);
        date.setMonth(this.dateShowModels.month + step);
        this.model = date.getTime();
        this.updateDate();
    };
    /**
     * @return {?}
     */
    ElDatePickerPanel.prototype.ngOnInit = function () {
        // hidden day
        this.currentView = this.hiddenDay ? 'month' : 'date';
        this.model = this.model || Date.now();
        this.updateDate();
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElDatePickerPanel.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.model)
            return;
        // first change
        if (changes.model.isFirstChange())
            return;
        this.model = changes.model.currentValue;
        this.model = this.model || Date.now();
        this.updateDate();
    };
    /**
     * @nocollapse
     */
    ElDatePickerPanel.ctorParameters = function () { return [
        { type: ElDataPicker, decorators: [{ type: Optional },] },
    ]; };
    ElDatePickerPanel.propDecorators = {
        'show': [{ type: Input },],
        'width': [{ type: Input },],
        'model': [{ type: Input },],
        'hiddenDay': [{ type: Input, args: ['hidden-day',] },],
        'panelAbsolute': [{ type: Input, args: ['panel-absolute',] },],
        'panelIndex': [{ type: Input, args: ['panel-index',] },],
        'modelChange': [{ type: Output },],
    };
ElDatePickerPanel.ɵfac = function ElDatePickerPanel_Factory(t) { return new (t || ElDatePickerPanel)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElDataPicker, 8)); };
ElDatePickerPanel.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElDatePickerPanel, selectors: [["el-data-picker-panel"]], inputs: { show: "show", hiddenDay: ["hidden-day", "hiddenDay"], panelAbsolute: ["panel-absolute", "panelAbsolute"], panelIndex: ["panel-index", "panelIndex"], model: "model", width: "width" }, outputs: { modelChange: "modelChange" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 15, vars: 19, consts: [[3, "ngStyle"], [1, "el-picker-panel__body-wrapper"], [1, "el-picker-panel__body"], [1, "el-date-picker__header"], ["type", "button", 1, "el-picker-panel__icon-btn", "el-date-picker__prev-btn", "el-icon-d-arrow-left", 3, "click"], ["class", "el-picker-panel__icon-btn el-date-picker__prev-btn el-icon-arrow-left", "type", "button", 3, "click", 4, "ngIf"], ["class", "el-date-picker__header-label", 3, "click", 4, "ngIf"], ["class", "el-date-picker__header-label", 4, "ngIf"], ["class", "el-date-picker__header-label", 3, "active", "click", 4, "ngIf"], ["type", "button", 1, "el-picker-panel__icon-btn", "el-date-picker__next-btn", "el-icon-d-arrow-right", 3, "click"], ["class", "el-picker-panel__icon-btn el-date-picker__next-btn el-icon-arrow-right", "type", "button", 3, "click", 4, "ngIf"], [1, "el-picker-panel__content"], [3, "model", "modelChange", 4, "ngIf"], [3, "model", "disabled-date", "modelChange", 4, "ngIf"], ["type", "button", 1, "el-picker-panel__icon-btn", "el-date-picker__prev-btn", "el-icon-arrow-left", 3, "click"], [1, "el-date-picker__header-label", 3, "click"], [1, "el-date-picker__header-label"], ["type", "button", 1, "el-picker-panel__icon-btn", "el-date-picker__next-btn", "el-icon-arrow-right", 3, "click"], [3, "model", "modelChange"], [3, "model", "disabled-date", "modelChange"]], template: function ElDatePickerPanel_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "div", 3);
        ɵngcc0.ɵɵelementStart(4, "button", 4);
        ɵngcc0.ɵɵlistener("click", function ElDatePickerPanel_Template_button_click_4_listener() { return ctx.nextYear(-1); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(5, ElDatePickerPanel_button_5_Template, 1, 0, "button", 5);
        ɵngcc0.ɵɵtemplate(6, ElDatePickerPanel_span_6_Template, 2, 1, "span", 6);
        ɵngcc0.ɵɵtemplate(7, ElDatePickerPanel_span_7_Template, 2, 2, "span", 7);
        ɵngcc0.ɵɵtemplate(8, ElDatePickerPanel_span_8_Template, 2, 3, "span", 8);
        ɵngcc0.ɵɵelementStart(9, "button", 9);
        ɵngcc0.ɵɵlistener("click", function ElDatePickerPanel_Template_button_click_9_listener() { return ctx.nextYear(1); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(10, ElDatePickerPanel_button_10_Template, 1, 0, "button", 10);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 11);
        ɵngcc0.ɵɵtemplate(12, ElDatePickerPanel_el_date_table_12_Template, 1, 1, "el-date-table", 12);
        ɵngcc0.ɵɵtemplate(13, ElDatePickerPanel_el_year_table_13_Template, 1, 2, "el-year-table", 13);
        ɵngcc0.ɵɵtemplate(14, ElDatePickerPanel_el_month_table_14_Template, 1, 2, "el-month-table", 13);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-picker-panel el-date-picker ");
        ɵngcc0.ɵɵclassProp("has-time", ctx.showTime)("el-picker-panel-absolute", ctx.panelAbsolute);
        ɵngcc0.ɵɵproperty("@dropAnimation", ctx.show)("ngStyle", ɵngcc0.ɵɵpureFunction2(16, _c0, ctx.width ? ctx.width + "px" : "auto", ctx.panelIndex));
        ɵngcc0.ɵɵadvance(5);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView === "date");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView !== "year");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView === "year");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView === "date");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView === "date");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView === "date" && !ctx.hiddenDay);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView === "year");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentView === "month");
    } }, directives: [ɵngcc2.NgStyle, ɵngcc2.NgIf, ɵngcc3.ElDateTable, ɵngcc4.ElYearTable, ɵngcc5.EMonthTable], styles: [".el-picker-panel-absolute[_ngcontent-%COMP%] { position: absolute; }"], data: { animation: [dropAnimation] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElDatePickerPanel, [{
        type: Component,
        args: [{
                selector: 'el-data-picker-panel',
                animations: [dropAnimation],
                styles: [' .el-picker-panel-absolute { position: absolute; } '],
                template: "\n    <div [@dropAnimation]=\"show\"\n      [ngStyle]=\"{ width: width ? width + 'px' : 'auto', 'z-index': panelIndex}\"\n      [class]=\"'el-picker-panel el-date-picker '\"\n      [class.has-time]=\"showTime\"\n      [class.el-picker-panel-absolute]=\"panelAbsolute\">\n      <div class=\"el-picker-panel__body-wrapper\">\n        <!--<div class=\"el-picker-panel__sidebar\" *ngIf=\"shortcuts\">-->\n          <!--<button type=\"button\" class=\"el-picker-panel__shortcut\"-->\n            <!--*ngFor=\"shortcut in shortcuts\"-->\n            <!--(click)=\"handleShortcutClick(shortcut)\">-->\n            <!--{{ shortcut.text }}-->\n          <!--</button>-->\n        <!--</div>-->\n        <div class=\"el-picker-panel__body\">\n          <div class=\"el-date-picker__header\">\n            <button class=\"el-picker-panel__icon-btn el-date-picker__prev-btn el-icon-d-arrow-left\"\n              type=\"button\" (click)=\"nextYear(-1)\">\n            </button>\n            <button class=\"el-picker-panel__icon-btn el-date-picker__prev-btn el-icon-arrow-left\"\n              type=\"button\" (click)=\"nextMonth(-1)\"\n              *ngIf=\"currentView === 'date'\">\n            </button>\n            \n            <!--year label-->\n            <span class=\"el-date-picker__header-label\" *ngIf=\"currentView !== 'year'\"\n              (click)=\"showPicker('year')\">{{dateShowModels.year}} \u5E74</span>\n            <!--year range label-->\n            <span class=\"el-date-picker__header-label\" *ngIf=\"currentView === 'year'\">\n              {{dateShowModels.yearRange[0]}} \u5E74 - {{dateShowModels.yearRange[1]}} \u5E74\n            </span>\n            \n            <span class=\"el-date-picker__header-label\"\n              [class.active]=\"currentView === 'month'\"\n              (click)=\"showPicker('month')\"\n              *ngIf=\"currentView === 'date'\">{{dateShowModels.month + 1}} \u6708</span>\n            <button class=\"el-picker-panel__icon-btn el-date-picker__next-btn el-icon-d-arrow-right\"\n              type=\"button\" (click)=\"nextYear(1)\">\n            </button>\n            <button class=\"el-picker-panel__icon-btn el-date-picker__next-btn el-icon-arrow-right\"\n              type=\"button\" (click)=\"nextMonth(1)\"\n              *ngIf=\"currentView === 'date'\">\n            </button>\n          </div>\n\n          <div class=\"el-picker-panel__content\">\n            <el-date-table *ngIf=\"currentView === 'date' && !hiddenDay\"\n              (modelChange)=\"datePickChangeHandle($event)\"\n              [model]=\"model\">\n            </el-date-table>\n            <el-year-table *ngIf=\"currentView === 'year'\"\n              [model]=\"model\"\n              (modelChange)=\"yearPickChangeHandle($event)\"\n              [disabled-date]=\"disabledDate\">\n            </el-year-table>\n            <el-month-table *ngIf=\"currentView === 'month'\"\n              [model]=\"model\"\n              (modelChange)=\"monthPickChangeHandle($event)\"\n              [disabled-date]=\"disabledDate\">\n            </el-month-table>\n          </div>\n        </div>\n      </div>\n\n      <!--<div class=\"el-picker-panel__footer\" *ngIf=\"footerVisible && currentView === 'date'\">-->\n        <!--<a href=\"JavaScript:\" class=\"el-picker-panel__link-btn\" (click)=\"changeToNow()\">556</a>-->\n        <!--<button class=\"el-picker-panel__btn\" type=\"button\"-->\n          <!--(click)=\"confirm()\">667</button>-->\n      <!--</div>-->\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElDataPicker, decorators: [{
                type: Optional
            }] }]; }, { show: [{
            type: Input
        }], hiddenDay: [{
            type: Input,
            args: ['hidden-day']
        }], panelAbsolute: [{
            type: Input,
            args: ['panel-absolute']
        }], panelIndex: [{
            type: Input,
            args: ['panel-index']
        }], modelChange: [{
            type: Output
        }], model: [{
            type: Input
        }], width: [{
            type: Input
        }] }); })();
    return ElDatePickerPanel;
}());
export { ElDatePickerPanel };
function ElDatePickerPanel_tsickle_Closure_declarations() {
    /** @type {?} */
    ElDatePickerPanel.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElDatePickerPanel.ctorParameters;
    /** @type {?} */
    ElDatePickerPanel.propDecorators;
    /** @type {?} */
    ElDatePickerPanel.prototype.show;
    /** @type {?} */
    ElDatePickerPanel.prototype.width;
    /** @type {?} */
    ElDatePickerPanel.prototype.model;
    /** @type {?} */
    ElDatePickerPanel.prototype.hiddenDay;
    /** @type {?} */
    ElDatePickerPanel.prototype.panelAbsolute;
    /** @type {?} */
    ElDatePickerPanel.prototype.panelIndex;
    /** @type {?} */
    ElDatePickerPanel.prototype.modelChange;
    /** @type {?} */
    ElDatePickerPanel.prototype.shortcuts;
    /** @type {?} */
    ElDatePickerPanel.prototype.showTime;
    /** @type {?} */
    ElDatePickerPanel.prototype.currentView;
    /** @type {?} */
    ElDatePickerPanel.prototype.dateShowModels;
    /** @type {?} */
    ElDatePickerPanel.prototype.root;
}
