// element-angular/__ivy_ngcc__/release/select/option.js
import { Component, Input, Optional } from '@angular/core';
import { ElSelect } from './select';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './select';
var ElOption = /** @class */ (function () {
    /**
     * @param {?} rootSelect
     */
    function ElOption(rootSelect) {
        this.rootSelect = rootSelect;
        this.elDisabled = false;
        this.rootDisabled = false;
        this.itemSelected = false;
    }
    Object.defineProperty(ElOption.prototype, "disabled", {
        /**
         * @param {?} val
         * @return {?}
         */
        set: function (val) {
            console.warn('Element Angular: (disabled) is discarded, use [elDisabled] replace it.');
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} event
     * @return {?}
     */
    ElOption.prototype.clickHandle = function (event) {
        event.stopPropagation();
        if (this.elDisabled || this.rootDisabled)
            return;
        this.rootSelect.changeLabel(this.label, this.value);
    };
    /**
     * @return {?}
     */
    ElOption.prototype.ngOnInit = function () {
        var _this = this;
        var /** @type {?} */ updateHandle = function () {
            if (_this.rootSelect.multiple) {
                _this.itemSelected = Array.isArray(_this.rootSelect.model) && _this.rootSelect.model.indexOf(_this.value) > -1;
            }
            else {
                _this.itemSelected = _this.value === _this.rootSelect.model;
            }
            _this.itemSelected && _this.rootSelect.changeLabel(_this.label);
        };
        this.rootDisabled = this.rootSelect.elDisabled;
        updateHandle();
        this.rootSelect.subscriber.push(updateHandle);
        this.rootSelect.appendOptions({
            value: this.value,
            label: this.label,
        });
    };
    /**
     * @nocollapse
     */
    ElOption.ctorParameters = function () { return [
        { type: ElSelect, decorators: [{ type: Optional },] },
    ]; };
    ElOption.propDecorators = {
        'disabled': [{ type: Input },],
        'elDisabled': [{ type: Input },],
        'value': [{ type: Input },],
        'label': [{ type: Input },],
    };
ElOption.ɵfac = function ElOption_Factory(t) { return new (t || ElOption)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElSelect, 8)); };
ElOption.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElOption, selectors: [["el-option"]], inputs: { elDisabled: "elDisabled", disabled: "disabled", value: "value", label: "label" }, decls: 3, vars: 5, consts: [[1, "el-select-dropdown__item", 3, "click"]], template: function ElOption_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "li", 0);
        ɵngcc0.ɵɵlistener("click", function ElOption_Template_li_click_0_listener($event) { return ctx.clickHandle($event); });
        ɵngcc0.ɵɵelementStart(1, "span");
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("is-disabled", ctx.elDisabled || ctx.rootDisabled)("selected", ctx.itemSelected);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.label);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElOption, [{
        type: Component,
        args: [{
                selector: 'el-option',
                template: "\n    <li class=\"el-select-dropdown__item\"\n      [class.is-disabled]=\"elDisabled || rootDisabled\"\n      [class.selected]=\"itemSelected\"\n      (click)=\"clickHandle($event)\">\n      <span>{{ label }}</span>\n    </li>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElSelect, decorators: [{
                type: Optional
            }] }]; }, { elDisabled: [{
            type: Input
        }], disabled: [{
            type: Input
        }], value: [{
            type: Input
        }], label: [{
            type: Input
        }] }); })();
    return ElOption;
}());
export { ElOption };
function ElOption_tsickle_Closure_declarations() {
    /** @type {?} */
    ElOption.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElOption.ctorParameters;
    /** @type {?} */
    ElOption.propDecorators;
    /** @type {?} */
    ElOption.prototype.elDisabled;
    /** @type {?} */
    ElOption.prototype.value;
    /** @type {?} */
    ElOption.prototype.label;
    /** @type {?} */
    ElOption.prototype.rootDisabled;
    /** @type {?} */
    ElOption.prototype.itemSelected;
    /** @type {?} */
    ElOption.prototype.rootSelect;
}
