// element-angular/__ivy_ngcc__/release/dialog/dialog.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';
import * as ɵngcc2 from '../shared/services/index';
import * as ɵngcc3 from '@angular/common';

var _c0 = ["title"];
var _c1 = ["footer"];
function ElDialog_ng_container_3_ng_template_1_Template(rf, ctx) { }
function ElDialog_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, ElDialog_ng_container_3_ng_template_1_Template, 0, 0, "ng-template", 8);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r0.titleTmp);
} }
function ElDialog_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 9);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.title);
} }
function ElDialog_button_5_Template(rf, ctx) { if (rf & 1) {
    var _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 10);
    ɵngcc0.ɵɵlistener("click", function ElDialog_button_5_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.closeHandler(); });
    ɵngcc0.ɵɵelement(1, "i", 11);
    ɵngcc0.ɵɵelementEnd();
} }
function ElDialog_ng_container_9_ng_template_1_Template(rf, ctx) { }
function ElDialog_ng_container_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, ElDialog_ng_container_9_ng_template_1_Template, 0, 0, "ng-template", 8);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r3.footerTmp);
} }
var _c2 = function (a0) { return { "z-index": a0 }; };
var _c3 = ["*"];
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, ContentChild, Input, Renderer2, } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { DocumentWrapper, WindowWrapper } from '../shared/services/index';
import { ElDialogProps } from './dialog.props';
import { dialogFadeAnimation } from '../shared/animation/index';
var ElDialog = /** @class */ (function (_super) {
    __extends(ElDialog, _super);
    /**
     * @param {?} sanitizer
     * @param {?} renderer
     * @param {?} window
     * @param {?} document
     */
    function ElDialog(sanitizer, renderer, window, document) {
        var _this = _super.call(this) || this;
        _this.sanitizer = sanitizer;
        _this.renderer = renderer;
        _this.window = window;
        _this.document = document;
        return _this;
    }
    /**
     * @return {?}
     */
    ElDialog.prototype.makeDialogStyles = function () {
        var /** @type {?} */ width = this.width ? "width: " + this.width + ";" : '';
        var /** @type {?} */ styles = "top: " + this.top + ";" + width;
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @return {?}
     */
    ElDialog.prototype.closeHandler = function () {
        var _this = this;
        this.beforeClose(function () {
            _this.visible = false;
            _this.visibleChange.emit(_this.visible);
            _this.updateWithVisibleChange();
        });
    };
    /**
     * @return {?}
     */
    ElDialog.prototype.wrapperClickHandle = function () {
        // close dialog
        if (this.closeOnClickModal) {
            this.closeHandler();
        }
    };
    /**
     * @return {?}
     */
    ElDialog.prototype.updateWithVisibleChange = function () {
        // update modal
        if (this.cacheModalElement) {
            // this.cacheModalElement.innerHTML = this.makeModalTmp(this.visible)
            this.renderer.setStyle(this.cacheModalElement, 'display', this.visible ? 'block' : 'none');
        }
        // lock body scroll
        if (this.lockScroll) {
            var /** @type {?} */ nextValue = this.visible ? 'hidden' : this.cacheOverflow;
            this.renderer.setStyle(this.document.body, 'overflow', nextValue);
        }
    };
    /**
     * @return {?}
     */
    ElDialog.prototype.ngOnInit = function () {
        var _this = this;
        // save old overflow
        if (this.lockScroll) {
            this.cacheOverflow = this.window.getComputedStyle(this.document.body).overflow;
        }
        this.cacheModalElement = this.renderer.createElement('div');
        this.renderer.setAttribute(this.cacheModalElement, 'class', 'v-modal');
        this.renderer.setStyle(this.cacheModalElement, 'z-index', this.modalZindex);
        this.renderer.setStyle(this.cacheModalElement, 'display', this.visible ? 'block' : 'none');
        this.document.body.appendChild(this.cacheModalElement);
        // listen esc
        if (this.closeOnPressEscape) {
            this.globalListenFunc = this.renderer.listen('document', 'keydown', function (event) {
                _this.visible && event.keyCode === 27 && _this.closeHandler();
            });
        }
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElDialog.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.visible)
            return;
        this.visible = changes.visible.currentValue;
        this.updateWithVisibleChange();
    };
    /**
     * @return {?}
     */
    ElDialog.prototype.ngOnDestroy = function () {
        this.globalListenFunc && this.globalListenFunc();
        this.cacheModalElement.parentElement.removeChild(this.cacheModalElement);
        this.renderer.setStyle(this.document.body, 'overflow', this.cacheOverflow);
    };
    /**
     * @nocollapse
     */
    ElDialog.ctorParameters = function () { return [
        { type: DomSanitizer, },
        { type: Renderer2, },
        { type: WindowWrapper, },
        { type: DocumentWrapper, },
    ]; };
    ElDialog.propDecorators = {
        'titleTmp': [{ type: ContentChild, args: ['title',] },],
        'footerTmp': [{ type: ContentChild, args: ['footer',] },],
        'model': [{ type: Input },],
    };
ElDialog.ɵfac = function ElDialog_Factory(t) { return new (t || ElDialog)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.WindowWrapper), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.DocumentWrapper)); };
ElDialog.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElDialog, selectors: [["el-dialog"]], contentQueries: function ElDialog_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c0, true);
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c1, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.titleTmp = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.footerTmp = _t.first);
    } }, inputs: { model: "model" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c3, decls: 10, vars: 14, consts: [[1, "el-dialog__wrapper", 3, "ngStyle", "click"], [3, "click"], [1, "el-dialog__header"], [4, "ngIf"], ["class", "el-dialog__title", 4, "ngIf"], ["type", "button", "class", "el-dialog__headerbtn", "aria-label", "Close", 3, "click", 4, "ngIf"], [1, "el-dialog__body"], [1, "el-dialog__footer"], [3, "ngTemplateOutlet"], [1, "el-dialog__title"], ["type", "button", "aria-label", "Close", 1, "el-dialog__headerbtn", 3, "click"], [1, "el-dialog__close", "el-icon", "el-icon-close"]], template: function ElDialog_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("click", function ElDialog_Template_div_click_0_listener() { return ctx.wrapperClickHandle(); });
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵlistener("click", function ElDialog_Template_div_click_1_listener($event) { return $event.stopPropagation(); });
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵtemplate(3, ElDialog_ng_container_3_Template, 2, 1, "ng-container", 3);
        ɵngcc0.ɵɵtemplate(4, ElDialog_span_4_Template, 2, 1, "span", 4);
        ɵngcc0.ɵɵtemplate(5, ElDialog_button_5_Template, 2, 0, "button", 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "div", 6);
        ɵngcc0.ɵɵprojection(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "div", 7);
        ɵngcc0.ɵɵtemplate(9, ElDialog_ng_container_9_Template, 2, 1, "ng-container", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("@dialogFadeAnimation", ctx.visible)("ngStyle", ɵngcc0.ɵɵpureFunction1(12, _c2, ctx.dialogZindex));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleMap(ctx.makeDialogStyles());
        ɵngcc0.ɵɵclassMap("el-dialog " + ctx.customClass);
        ɵngcc0.ɵɵclassProp("el-dialog--center", ctx.center);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.titleTmp);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.titleTmp);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showClose);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngIf", ctx.footerTmp);
    } }, directives: [ɵngcc3.NgStyle, ɵngcc3.NgIf, ɵngcc3.NgTemplateOutlet], encapsulation: 2, data: { animation: [dialogFadeAnimation] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElDialog, [{
        type: Component,
        args: [{
                selector: 'el-dialog',
                animations: [dialogFadeAnimation],
                template: "\n    <div class=\"el-dialog__wrapper\" [@dialogFadeAnimation]=\"visible\"\n      [ngStyle]=\"{ 'z-index': dialogZindex }\"\n      (click)=\"wrapperClickHandle()\">\n      <div [class]=\"'el-dialog ' + customClass\"\n        [class.el-dialog--center]=\"center\"\n        [style]=\"makeDialogStyles()\"\n        (click)=\"$event.stopPropagation()\">\n        <div class=\"el-dialog__header\">\n          <ng-container *ngIf=\"titleTmp\">\n            <ng-template [ngTemplateOutlet]=\"titleTmp\"></ng-template>\n          </ng-container>\n          <span class=\"el-dialog__title\" *ngIf=\"!titleTmp\">{{title}}</span>\n          <button type=\"button\" class=\"el-dialog__headerbtn\"\n            aria-label=\"Close\"\n            *ngIf=\"showClose\"\n            (click)=\"closeHandler()\">\n            <i class=\"el-dialog__close el-icon el-icon-close\"></i>\n          </button>\n        </div>\n        <div class=\"el-dialog__body\">\n          <ng-content></ng-content>\n        </div>\n        <div class=\"el-dialog__footer\">\n          <ng-container *ngIf=\"footerTmp\">\n            <ng-template [ngTemplateOutlet]=\"footerTmp\"></ng-template>\n          </ng-container>\n        </div>\n      </div>\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.DomSanitizer }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc2.WindowWrapper }, { type: ɵngcc2.DocumentWrapper }]; }, { titleTmp: [{
            type: ContentChild,
            args: ['title']
        }], footerTmp: [{
            type: ContentChild,
            args: ['footer']
        }], model: [{
            type: Input
        }] }); })();
    return ElDialog;
}(ElDialogProps));
export { ElDialog };
function ElDialog_tsickle_Closure_declarations() {
    /** @type {?} */
    ElDialog.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElDialog.ctorParameters;
    /** @type {?} */
    ElDialog.propDecorators;
    /** @type {?} */
    ElDialog.prototype.titleTmp;
    /** @type {?} */
    ElDialog.prototype.footerTmp;
    /** @type {?} */
    ElDialog.prototype.model;
    /** @type {?} */
    ElDialog.prototype.cacheOverflow;
    /** @type {?} */
    ElDialog.prototype.cacheModalElement;
    /** @type {?} */
    ElDialog.prototype.globalListenFunc;
    /** @type {?} */
    ElDialog.prototype.sanitizer;
    /** @type {?} */
    ElDialog.prototype.renderer;
    /** @type {?} */
    ElDialog.prototype.window;
    /** @type {?} */
    ElDialog.prototype.document;
}
