// element-angular/__ivy_ngcc__/release/dialog/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElSharedModule } from '../shared/module';
import { ElDialog } from './dialog';
/**
 * @return {?}
 */
import * as ɵngcc0 from '@angular/core';
export function getDocument() { return document; }
/**
 * @return {?}
 */
export function getWindow() { return window; }
var ElDialogModule = /** @class */ (function () {
    function ElDialogModule() {
    }
    /**
     * @return {?}
     */
    ElDialogModule.forRoot = function () {
        return { ngModule: ElDialogModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElDialogModule.ctorParameters = function () { return []; };
ElDialogModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElDialogModule });
ElDialogModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElDialogModule_Factory(t) { return new (t || ElDialogModule)(); }, imports: [[CommonModule, ElSharedModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElDialogModule, { declarations: function () { return [ElDialog]; }, imports: function () { return [CommonModule, ElSharedModule]; }, exports: function () { return [ElDialog]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElDialogModule, [{
        type: NgModule,
        args: [{
                declarations: [ElDialog],
                exports: [ElDialog],
                imports: [CommonModule, ElSharedModule],
                entryComponents: [ElDialog]
            }]
    }], function () { return []; }, null); })();
    return ElDialogModule;
}());
export { ElDialogModule };
function ElDialogModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElDialogModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElDialogModule.ctorParameters;
}
