// element-angular/__ivy_ngcc__/release/alert/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElAlert } from './alert';
import * as ɵngcc0 from '@angular/core';
var ElAlertModule = /** @class */ (function () {
    function ElAlertModule() {
    }
    /**
     * @return {?}
     */
    ElAlertModule.forRoot = function () {
        return { ngModule: ElAlertModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElAlertModule.ctorParameters = function () { return []; };
ElAlertModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElAlertModule });
ElAlertModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElAlertModule_Factory(t) { return new (t || ElAlertModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElAlertModule, { declarations: function () { return [ElAlert]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElAlert]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElAlertModule, [{
        type: NgModule,
        args: [{
                declarations: [ElAlert],
                exports: [ElAlert],
                imports: [CommonModule],
                entryComponents: [ElAlert]
            }]
    }], function () { return []; }, null); })();
    return ElAlertModule;
}());
export { ElAlertModule };
function ElAlertModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElAlertModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElAlertModule.ctorParameters;
}
