// element-angular/__ivy_ngcc__/release/input/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElInput } from './input';
import { FormsModule } from '@angular/forms';
import { ElFormItem } from '../form/form-item';
import * as ɵngcc0 from '@angular/core';
var ElInputsModule = /** @class */ (function () {
    function ElInputsModule() {
    }
    /**
     * @return {?}
     */
    ElInputsModule.forRoot = function () {
        return { ngModule: ElInputsModule, providers: [ElFormItem] };
    };
    /**
     * @nocollapse
     */
    ElInputsModule.ctorParameters = function () { return []; };
ElInputsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElInputsModule });
ElInputsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElInputsModule_Factory(t) { return new (t || ElInputsModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElInputsModule, { declarations: function () { return [ElInput]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [ElInput]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElInputsModule, [{
        type: NgModule,
        args: [{
                declarations: [ElInput],
                exports: [ElInput],
                imports: [CommonModule, FormsModule],
                entryComponents: [ElInput]
            }]
    }], function () { return []; }, null); })();
    return ElInputsModule;
}());
export { ElInputsModule };
function ElInputsModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElInputsModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElInputsModule.ctorParameters;
}
