// element-angular/__ivy_ngcc__/release/tree/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElTree } from './tree';
import { ElTreeItem } from './tree-item';
import { ElCheckboxsModule } from '../checkbox/module';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
var ElTreeModule = /** @class */ (function () {
    function ElTreeModule() {
    }
    /**
     * @return {?}
     */
    ElTreeModule.forRoot = function () {
        return { ngModule: ElTreeModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElTreeModule.ctorParameters = function () { return []; };
ElTreeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElTreeModule });
ElTreeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElTreeModule_Factory(t) { return new (t || ElTreeModule)(); }, imports: [[CommonModule, ElCheckboxsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElTreeModule, { declarations: function () { return [ElTree, ElTreeItem]; }, imports: function () { return [CommonModule, ElCheckboxsModule]; }, exports: function () { return [ElTree, ElTreeItem]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElTreeModule, [{
        type: NgModule,
        args: [{
                declarations: [ElTree, ElTreeItem],
                exports: [ElTree, ElTreeItem],
                imports: [CommonModule, ElCheckboxsModule],
                entryComponents: [ElTree]
            }]
    }], function () { return []; }, null); })();
ɵngcc0.ɵɵsetComponentScope(ElTree, [ɵngcc1.NgForOf, ElTreeItem, ɵngcc1.NgIf], []);
    return ElTreeModule;
}());
export { ElTreeModule };
function ElTreeModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElTreeModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElTreeModule.ctorParameters;
}
