// element-angular/__ivy_ngcc__/release/container/container.js
import { Component, ElementRef, Input } from '@angular/core';
import { removeNgTag } from '../shared/utils/host';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var ElContainer = /** @class */ (function () {
    /**
     * @param {?} el
     */
    function ElContainer(el) {
        this.el = el;
        this.direction = '';
        this.isVertical = false;
    }
    /**
     * @param {?=} isVertical
     * @return {?}
     */
    ElContainer.prototype.setVertical = function (isVertical) {
        if (isVertical && this.direction !== 'horizontal') {
            this.isVertical = true;
        }
    };
    /**
     * @return {?}
     */
    ElContainer.prototype.ngOnInit = function () {
        console.log('Element Angular: Container Component is discarded, use [Container directive] replace it.');
        this.isVertical = this.direction === 'vertical';
        removeNgTag(this.el.nativeElement);
    };
    /**
     * @nocollapse
     */
    ElContainer.ctorParameters = function () { return [
        { type: ElementRef, },
    ]; };
    ElContainer.propDecorators = {
        'direction': [{ type: Input },],
        'class': [{ type: Input },],
    };
ElContainer.ɵfac = function ElContainer_Factory(t) { return new (t || ElContainer)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElContainer.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElContainer, selectors: [["el-container"]], inputs: { direction: "direction", class: "class" }, ngContentSelectors: _c0, decls: 2, vars: 4, template: function ElContainer_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "section");
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-container " + ctx.class);
        ɵngcc0.ɵɵclassProp("is-vertical", ctx.isVertical);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElContainer, [{
        type: Component,
        args: [{
                selector: 'el-container',
                template: "\n    <section [class]=\"'el-container ' + class\" [class.is-vertical]=\"isVertical\">\n      <ng-content></ng-content>\n    </section>\n  "
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { direction: [{
            type: Input
        }], class: [{
            type: Input
        }] }); })();
    return ElContainer;
}());
export { ElContainer };
function ElContainer_tsickle_Closure_declarations() {
    /** @type {?} */
    ElContainer.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElContainer.ctorParameters;
    /** @type {?} */
    ElContainer.propDecorators;
    /** @type {?} */
    ElContainer.prototype.direction;
    /** @type {?} */
    ElContainer.prototype.class;
    /** @type {?} */
    ElContainer.prototype.isVertical;
    /** @type {?} */
    ElContainer.prototype.el;
}
