// element-angular/__ivy_ngcc__/release/container/container.directive.js
import { Directive, Input, Self } from '@angular/core';
import { NgClass } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
var ElContainerDirective = /** @class */ (function () {
    /**
     * @param {?} ngClass
     */
    function ElContainerDirective(ngClass) {
        this.ngClass = ngClass;
        this.direction = '';
    }
    /**
     * @return {?}
     */
    ElContainerDirective.prototype.ngOnChanges = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElContainerDirective.prototype.ngOnInit = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElContainerDirective.prototype.colletClasses = function () {
        this.hostClasses = {
            'el-container': true,
            'is-vertical': this.direction === 'vertical',
        };
        this.ngClass.ngClass = this.hostClasses;
        this.ngClass.ngDoCheck();
    };
    /**
     * @nocollapse
     */
    ElContainerDirective.ctorParameters = function () { return [
        { type: NgClass, decorators: [{ type: Self },] },
    ]; };
    ElContainerDirective.propDecorators = {
        'direction': [{ type: Input },],
    };
ElContainerDirective.ɵfac = function ElContainerDirective_Factory(t) { return new (t || ElContainerDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NgClass, 2)); };
ElContainerDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElContainerDirective, selectors: [["", "el-container", ""]], inputs: { direction: "direction" }, features: [ɵngcc0.ɵɵProvidersFeature([NgClass]), ɵngcc0.ɵɵNgOnChangesFeature] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElContainerDirective, [{
        type: Directive,
        args: [{
                selector: '[el-container]',
                providers: [NgClass]
            }]
    }], function () { return [{ type: ɵngcc1.NgClass, decorators: [{
                type: Self
            }] }]; }, { direction: [{
            type: Input
        }] }); })();
    return ElContainerDirective;
}());
export { ElContainerDirective };
function ElContainerDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    ElContainerDirective.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElContainerDirective.ctorParameters;
    /** @type {?} */
    ElContainerDirective.propDecorators;
    /** @type {?} */
    ElContainerDirective.prototype.direction;
    /** @type {?} */
    ElContainerDirective.prototype.hostClasses;
    /** @type {?} */
    ElContainerDirective.prototype.ngClass;
}
