// element-angular/__ivy_ngcc__/release/steps/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElStep } from './step';
import { ElSteps } from './steps';
import * as ɵngcc0 from '@angular/core';
var ElStepsModule = /** @class */ (function () {
    function ElStepsModule() {
    }
    /**
     * @return {?}
     */
    ElStepsModule.forRoot = function () {
        return { ngModule: ElStepsModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElStepsModule.ctorParameters = function () { return []; };
ElStepsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElStepsModule });
ElStepsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElStepsModule_Factory(t) { return new (t || ElStepsModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElStepsModule, { declarations: function () { return [ElStep, ElSteps]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElStep, ElSteps]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElStepsModule, [{
        type: NgModule,
        args: [{
                declarations: [ElStep, ElSteps],
                exports: [ElStep, ElSteps],
                imports: [CommonModule],
                entryComponents: [ElStep, ElSteps]
            }]
    }], function () { return []; }, null); })();
    return ElStepsModule;
}());
export { ElStepsModule };
function ElStepsModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElStepsModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElStepsModule.ctorParameters;
}
