// element-angular/__ivy_ngcc__/release/row/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElRowDirective } from './row.directive';
import * as ɵngcc0 from '@angular/core';
var ElRowModule = /** @class */ (function () {
    function ElRowModule() {
    }
    /**
     * @return {?}
     */
    ElRowModule.forRoot = function () {
        return { ngModule: ElRowModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElRowModule.ctorParameters = function () { return []; };
ElRowModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElRowModule });
ElRowModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElRowModule_Factory(t) { return new (t || ElRowModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElRowModule, { declarations: function () { return [ElRowDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElRowDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRowModule, [{
        type: NgModule,
        args: [{
                declarations: [ElRowDirective],
                exports: [ElRowDirective],
                imports: [CommonModule],
                entryComponents: []
            }]
    }], function () { return []; }, null); })();
    return ElRowModule;
}());
export { ElRowModule };
function ElRowModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRowModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRowModule.ctorParameters;
}
