// element-angular/__ivy_ngcc__/release/row/row.directive.js
import { Input, Directive, ElementRef } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';
var ElRowDirective = /** @class */ (function () {
    /**
     * @param {?} sanitizer
     * @param {?} el
     */
    function ElRowDirective(sanitizer, el) {
        this.sanitizer = sanitizer;
        this.el = el;
        this.gutter = 0;
        this.justify = 'start';
        this.align = 'top';
        this.nativeClass = ' ';
        this.nativeClass += this.el.nativeElement.classList.value;
    }
    /**
     * @return {?}
     */
    ElRowDirective.prototype.justifyClass = function () {
        return this.justify !== 'start' ? " is-justify-" + this.justify : '';
    };
    /**
     * @return {?}
     */
    ElRowDirective.prototype.alignClass = function () {
        return this.align !== 'top' ? " is-align-" + this.align : '';
    };
    /**
     * @return {?}
     */
    ElRowDirective.prototype.gutterStyle = function () {
        var /** @type {?} */ styleStr = '';
        if (this.gutter) {
            styleStr += "margin-left: -" + this.gutter / 2 + "px;";
            styleStr += "margin-right: -" + this.gutter / 2 + "px;";
        }
        return this.sanitizer.bypassSecurityTrustStyle(styleStr);
    };
    /**
     * @nocollapse
     */
    ElRowDirective.ctorParameters = function () { return [
        { type: DomSanitizer, },
        { type: ElementRef, },
    ]; };
    ElRowDirective.propDecorators = {
        'type': [{ type: Input },],
        'gutter': [{ type: Input },],
        'justify': [{ type: Input },],
        'align': [{ type: Input },],
    };
ElRowDirective.ɵfac = function ElRowDirective_Factory(t) { return new (t || ElRowDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElRowDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElRowDirective, selectors: [["", "el-row", ""]], hostVars: 6, hostBindings: function ElRowDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵstyleMap(ctx.gutterStyle());
        ɵngcc0.ɵɵclassMap("el-row" + ctx.justifyClass() + ctx.alignClass() + ctx.nativeClass);
        ɵngcc0.ɵɵclassProp("el-row--flex", ctx.type === "flex");
    } }, inputs: { gutter: "gutter", justify: "justify", align: "align", type: "type" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRowDirective, [{
        type: Directive,
        args: [{
                selector: '[el-row]',
                host: {
                    '[class]': '"el-row" + justifyClass() + alignClass() + nativeClass',
                    '[class.el-row--flex]': 'type === "flex"',
                    '[style]': 'gutterStyle()'
                }
            }]
    }], function () { return [{ type: ɵngcc1.DomSanitizer }, { type: ɵngcc0.ElementRef }]; }, { gutter: [{
            type: Input
        }], justify: [{
            type: Input
        }], align: [{
            type: Input
        }], type: [{
            type: Input
        }] }); })();
    return ElRowDirective;
}());
export { ElRowDirective };
function ElRowDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRowDirective.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRowDirective.ctorParameters;
    /** @type {?} */
    ElRowDirective.propDecorators;
    /** @type {?} */
    ElRowDirective.prototype.type;
    /** @type {?} */
    ElRowDirective.prototype.gutter;
    /** @type {?} */
    ElRowDirective.prototype.justify;
    /** @type {?} */
    ElRowDirective.prototype.align;
    /** @type {?} */
    ElRowDirective.prototype.nativeClass;
    /** @type {?} */
    ElRowDirective.prototype.sanitizer;
    /** @type {?} */
    ElRowDirective.prototype.el;
}
