// element-angular/__ivy_ngcc__/release/shared/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElClassDirective } from './directives/index';
import { ExDynamicService, DocumentWrapper, WindowWrapper } from '../shared/services/index';
import { ElCSSValuePipe } from './pipe/index';
/**
 * @return {?}
 */
import * as ɵngcc0 from '@angular/core';
export function getDocument() { return document; }
/**
 * @return {?}
 */
export function getWindow() { return window; }
var ElSharedModule = /** @class */ (function () {
    function ElSharedModule() {
    }
    /**
     * @return {?}
     */
    ElSharedModule.forRoot = function () {
        return { ngModule: ElSharedModule, providers: [
                ExDynamicService,
                ElCSSValuePipe,
                { provide: DocumentWrapper, useFactory: getDocument },
                { provide: WindowWrapper, useFactory: getWindow },
            ] };
    };
    /**
     * @nocollapse
     */
    ElSharedModule.ctorParameters = function () { return []; };
ElSharedModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElSharedModule });
ElSharedModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElSharedModule_Factory(t) { return new (t || ElSharedModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElSharedModule, { declarations: function () { return [ElCSSValuePipe,
        ElClassDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElCSSValuePipe,
        ElClassDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElSharedModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    ElCSSValuePipe,
                    ElClassDirective,
                ],
                exports: [
                    ElCSSValuePipe,
                    ElClassDirective,
                ],
                imports: [CommonModule],
                entryComponents: []
            }]
    }], function () { return []; }, null); })();
    return ElSharedModule;
}());
export { ElSharedModule };
function ElSharedModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElSharedModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElSharedModule.ctorParameters;
}
