// element-angular/__ivy_ngcc__/release/container/children/aside.js
import { Component, ElementRef, Input } from '@angular/core';
import { removeNgTag } from '../../shared/utils/index';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

var _c0 = function (a0) { return { width: a0 }; };
var _c1 = ["*"];
var ElAside = /** @class */ (function () {
    /**
     * @param {?} el
     */
    function ElAside(el) {
        this.el = el;
        this.width = '300px';
    }
    /**
     * @return {?}
     */
    ElAside.prototype.ngOnInit = function () {
        console.log('Element Angular: Container Component is discarded, use [Container directive] replace it.');
        removeNgTag(this.el.nativeElement);
    };
    /**
     * @nocollapse
     */
    ElAside.ctorParameters = function () { return [
        { type: ElementRef, },
    ]; };
    ElAside.propDecorators = {
        'width': [{ type: Input },],
        'class': [{ type: Input },],
    };
ElAside.ɵfac = function ElAside_Factory(t) { return new (t || ElAside)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElAside.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElAside, selectors: [["el-aside"]], inputs: { width: "width", class: "class" }, ngContentSelectors: _c1, decls: 2, vars: 5, consts: [[3, "ngStyle"]], template: function ElAside_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "aside", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-aside " + ctx.class);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c0, ctx.width));
    } }, directives: [ɵngcc1.NgStyle], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElAside, [{
        type: Component,
        args: [{
                selector: 'el-aside',
                template: "\n    <aside [class]=\"'el-aside ' + class\" [ngStyle]=\"{width: width}\">\n      <ng-content></ng-content>\n    </aside>\n  "
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { width: [{
            type: Input
        }], class: [{
            type: Input
        }] }); })();
    return ElAside;
}());
export { ElAside };
function ElAside_tsickle_Closure_declarations() {
    /** @type {?} */
    ElAside.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElAside.ctorParameters;
    /** @type {?} */
    ElAside.propDecorators;
    /** @type {?} */
    ElAside.prototype.width;
    /** @type {?} */
    ElAside.prototype.class;
    /** @type {?} */
    ElAside.prototype.el;
}
