// element-angular/__ivy_ngcc__/release/container/children/footer.js
import { Component, ElementRef, Input } from '@angular/core';
import { ElContainer } from '../container';
import { removeNgTag } from '../../shared/utils/host';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../container';
import * as ɵngcc2 from '@angular/common';

var _c0 = function (a0) { return { height: a0 }; };
var _c1 = ["*"];
var ElFooter = /** @class */ (function () {
    /**
     * @param {?} root
     * @param {?} el
     */
    function ElFooter(root, el) {
        this.root = root;
        this.el = el;
        this.height = '60px';
    }
    /**
     * @return {?}
     */
    ElFooter.prototype.ngOnInit = function () {
        console.log('Element Angular: Container Component is discarded, use [Container directive] replace it.');
        this.root.setVertical(true);
        removeNgTag(this.el.nativeElement);
    };
    /**
     * @nocollapse
     */
    ElFooter.ctorParameters = function () { return [
        { type: ElContainer, },
        { type: ElementRef, },
    ]; };
    ElFooter.propDecorators = {
        'height': [{ type: Input },],
        'class': [{ type: Input },],
    };
ElFooter.ɵfac = function ElFooter_Factory(t) { return new (t || ElFooter)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElContainer), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElFooter.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElFooter, selectors: [["el-footer"]], inputs: { height: "height", class: "class" }, ngContentSelectors: _c1, decls: 2, vars: 5, consts: [[3, "ngStyle"]], template: function ElFooter_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "footer", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-footer " + ctx.class);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c0, ctx.height));
    } }, directives: [ɵngcc2.NgStyle], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElFooter, [{
        type: Component,
        args: [{
                selector: 'el-footer',
                template: "\n    <footer [class]=\"'el-footer ' + class\" [ngStyle]=\"{height: height}\">\n      <ng-content></ng-content>\n    </footer>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElContainer }, { type: ɵngcc0.ElementRef }]; }, { height: [{
            type: Input
        }], class: [{
            type: Input
        }] }); })();
    return ElFooter;
}());
export { ElFooter };
function ElFooter_tsickle_Closure_declarations() {
    /** @type {?} */
    ElFooter.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElFooter.ctorParameters;
    /** @type {?} */
    ElFooter.propDecorators;
    /** @type {?} */
    ElFooter.prototype.height;
    /** @type {?} */
    ElFooter.prototype.class;
    /** @type {?} */
    ElFooter.prototype.root;
    /** @type {?} */
    ElFooter.prototype.el;
}
