// element-angular/__ivy_ngcc__/release/container/children/header.js
import { Component, ElementRef, Input } from '@angular/core';
import { ElContainer } from '../container';
import { removeNgTag } from '../../shared/utils/host';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../container';
import * as ɵngcc2 from '@angular/common';

var _c0 = function (a0) { return { height: a0 }; };
var _c1 = ["*"];
var ElHeader = /** @class */ (function () {
    /**
     * @param {?} root
     * @param {?} el
     */
    function ElHeader(root, el) {
        this.root = root;
        this.el = el;
        this.height = '60px';
    }
    /**
     * @return {?}
     */
    ElHeader.prototype.ngOnInit = function () {
        console.log('Element Angular: Container Component is discarded, use [Container directive] replace it.');
        this.root.setVertical(true);
        removeNgTag(this.el.nativeElement);
    };
    /**
     * @nocollapse
     */
    ElHeader.ctorParameters = function () { return [
        { type: ElContainer, },
        { type: ElementRef, },
    ]; };
    ElHeader.propDecorators = {
        'height': [{ type: Input },],
        'class': [{ type: Input },],
    };
ElHeader.ɵfac = function ElHeader_Factory(t) { return new (t || ElHeader)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElContainer), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElHeader.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElHeader, selectors: [["el-header"]], inputs: { height: "height", class: "class" }, ngContentSelectors: _c1, decls: 2, vars: 5, consts: [[3, "ngStyle"]], template: function ElHeader_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "header", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-header " + ctx.class);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c0, ctx.height));
    } }, directives: [ɵngcc2.NgStyle], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElHeader, [{
        type: Component,
        args: [{
                selector: 'el-header',
                template: "\n    <header [class]=\"'el-header ' + class\" [ngStyle]=\"{height: height}\">\n      <ng-content></ng-content>\n    </header>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElContainer }, { type: ɵngcc0.ElementRef }]; }, { height: [{
            type: Input
        }], class: [{
            type: Input
        }] }); })();
    return ElHeader;
}());
export { ElHeader };
function ElHeader_tsickle_Closure_declarations() {
    /** @type {?} */
    ElHeader.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElHeader.ctorParameters;
    /** @type {?} */
    ElHeader.propDecorators;
    /** @type {?} */
    ElHeader.prototype.height;
    /** @type {?} */
    ElHeader.prototype.class;
    /** @type {?} */
    ElHeader.prototype.root;
    /** @type {?} */
    ElHeader.prototype.el;
}
