// element-angular/__ivy_ngcc__/release/container/children/main.js
import { Component, ElementRef, Input } from '@angular/core';
import { removeNgTag } from '../../shared/utils/host';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var ElMain = /** @class */ (function () {
    /**
     * @param {?} el
     */
    function ElMain(el) {
        this.el = el;
    }
    /**
     * @return {?}
     */
    ElMain.prototype.ngOnInit = function () {
        console.log('Element Angular: Container Component is discarded, use [Container directive] replace it.');
        removeNgTag(this.el.nativeElement);
    };
    /**
     * @nocollapse
     */
    ElMain.ctorParameters = function () { return [
        { type: ElementRef, },
    ]; };
    ElMain.propDecorators = {
        'class': [{ type: Input },],
    };
ElMain.ɵfac = function ElMain_Factory(t) { return new (t || ElMain)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ElMain.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElMain, selectors: [["el-main"]], inputs: { class: "class" }, ngContentSelectors: _c0, decls: 2, vars: 2, template: function ElMain_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "main");
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-main " + ctx.class);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElMain, [{
        type: Component,
        args: [{
                selector: 'el-main',
                template: "\n    <main [class]=\"'el-main ' + class\">\n      <ng-content></ng-content>\n    </main>\n  "
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { class: [{
            type: Input
        }] }); })();
    return ElMain;
}());
export { ElMain };
function ElMain_tsickle_Closure_declarations() {
    /** @type {?} */
    ElMain.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElMain.ctorParameters;
    /** @type {?} */
    ElMain.propDecorators;
    /** @type {?} */
    ElMain.prototype.class;
    /** @type {?} */
    ElMain.prototype.el;
}
