// element-angular/__ivy_ngcc__/release/date-picker/children/date-table.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { DateFormat } from '../utils/format';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function ElDateTable_th_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var week_r2 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(week_r2);
} }
function ElDateTable_tr_4_td_1_Template(rf, ctx) { if (rf & 1) {
    var _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 5);
    ɵngcc0.ɵɵlistener("click", function ElDateTable_tr_4_td_1_Template_td_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r7); var item_r5 = ctx.$implicit; var ctx_r6 = ɵngcc0.ɵɵnextContext(2); return ctx_r6.clickHandle(item_r5); });
    ɵngcc0.ɵɵelementStart(1, "div");
    ɵngcc0.ɵɵelementStart(2, "span");
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r5 = ctx.$implicit;
    var ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassProp("available", item_r5.monthOffset === 0)("next-month", item_r5.monthOffset === 1)("prev-month", item_r5.monthOffset === -1)("normal", item_r5.monthOffset === 0)("today", ctx_r4.isToday(item_r5))("current", ctx_r4.isTargetDay(item_r5));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(item_r5.day);
} }
function ElDateTable_tr_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr", 3);
    ɵngcc0.ɵɵtemplate(1, ElDateTable_tr_4_td_1_Template, 4, 13, "td", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var row_r3 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", row_r3);
} }
var ElDateTable = /** @class */ (function () {
    function ElDateTable() {
        this.modelChange = new EventEmitter();
        this.weeks = ['日', '一', '二', '三', '四', '五', '六'];
        this.tableRows = [[], [], [], [], [], []];
        this.targetMonthOffset = 0; // default: current month, offset = 0
    }
    /**
     * @param {?} first
     * @param {?} lastCount
     * @return {?}
     */
    ElDateTable.BuildMonthStartRow = function (first, lastCount) {
        var /** @type {?} */ lastday = 7 - first;
        // first loop
        lastCount++;
        lastday++;
        return [0, 1, 2, 3, 4, 5, 6].map(function () {
            lastday--;
            if (lastday > 0)
                return { day: lastday, monthOffset: 0 };
            lastCount--;
            return { day: lastCount, monthOffset: -1 };
        }).reverse();
    };
    /**
     * @param {?} item
     * @return {?}
     */
    ElDateTable.prototype.isToday = function (item) {
        if (this.currentMonthOffset === null)
            return false;
        return item.monthOffset === this.currentMonthOffset && this.today === item.day;
    };
    /**
     * @param {?} item
     * @return {?}
     */
    ElDateTable.prototype.isTargetDay = function (item) {
        return item.monthOffset === this.targetMonthOffset && item.day === this.targetDay;
    };
    /**
     * @param {?} item
     * @return {?}
     */
    ElDateTable.prototype.clickHandle = function (item) {
        var /** @type {?} */ date = this.date;
        var /** @type {?} */ currentMonth = date.getMonth() + 1;
        var /** @type {?} */ targetMonth = currentMonth + item.monthOffset;
        // update target and update view
        this.targetDay = item.day;
        this.targetMonthOffset = item.monthOffset;
        // get time and emit a number
        date.setMonth(targetMonth - 1);
        date.setDate(item.day);
        this.modelChange.emit(date.getTime());
    };
    /**
     * @return {?}
     */
    ElDateTable.prototype.getRows = function () {
        var /** @type {?} */ date = this.date;
        this.targetDay = date.getDate();
        this.today = new Date().getDate();
        this.currentMonthOffset = DateFormat.getCurrentMonthOffset(date);
        var /** @type {?} */ lastMonth = date.getMonth() - 1;
        var /** @type {?} */ lastYear = lastMonth < 0 ? date.getFullYear() - 1 : date.getFullYear();
        var /** @type {?} */ currentMonthdayCount = DateFormat.getDayCountOfMonth(date.getFullYear(), date.getMonth());
        var /** @type {?} */ lastMonthDayCount = DateFormat.getDayCountOfMonth(lastYear, lastMonth < 0 ? 12 : lastMonth);
        var /** @type {?} */ firstDay = DateFormat.getFirstDayOfMonth(date);
        var /** @type {?} */ nextMonthDay = 0;
        this.tableRows = this.tableRows.map(function (row, index) {
            if (index === 0) {
                return ElDateTable.BuildMonthStartRow(firstDay, lastMonthDayCount);
            }
            var /** @type {?} */ thisWeekFirstDay = 7 - firstDay + 7 * (index - 1);
            return new Array(7).fill(0).map(function (v, i) {
                var /** @type {?} */ start = thisWeekFirstDay + i + 1;
                if (start <= currentMonthdayCount)
                    return { day: start, monthOffset: 0 };
                nextMonthDay++;
                return { day: nextMonthDay, monthOffset: 1 };
            });
        });
    };
    /**
     * @return {?}
     */
    ElDateTable.prototype.ngOnInit = function () {
        this.date = new Date(this.model);
        this.getRows();
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElDateTable.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.model)
            return;
        // first change
        if (changes.model.isFirstChange())
            return;
        this.model = changes.model.currentValue;
        this.date = new Date(this.model);
        this.getRows();
    };
    /**
     * @nocollapse
     */
    ElDateTable.ctorParameters = function () { return []; };
    ElDateTable.propDecorators = {
        'model': [{ type: Input },],
        'disabledDate': [{ type: Input, args: ['disabled-date',] },],
        'modelChange': [{ type: Output },],
    };
ElDateTable.ɵfac = function ElDateTable_Factory(t) { return new (t || ElDateTable)(); };
ElDateTable.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElDateTable, selectors: [["el-date-table"]], inputs: { model: "model", disabledDate: ["disabled-date", "disabledDate"] }, outputs: { modelChange: "modelChange" }, features: [ɵngcc0.ɵɵProvidersFeature([DateFormat]), ɵngcc0.ɵɵNgOnChangesFeature], decls: 5, vars: 2, consts: [["cellspacing", "0", "cellpadding", "0", 1, "el-date-table"], [4, "ngFor", "ngForOf"], ["class", "el-date-table__row", 4, "ngFor", "ngForOf"], [1, "el-date-table__row"], [3, "available", "next-month", "prev-month", "normal", "today", "current", "click", 4, "ngFor", "ngForOf"], [3, "click"]], template: function ElDateTable_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "table", 0);
        ɵngcc0.ɵɵelementStart(1, "tbody");
        ɵngcc0.ɵɵelementStart(2, "tr");
        ɵngcc0.ɵɵtemplate(3, ElDateTable_th_3_Template, 2, 1, "th", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(4, ElDateTable_tr_4_Template, 2, 1, "tr", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.weeks);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.tableRows);
    } }, directives: [ɵngcc1.NgForOf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElDateTable, [{
        type: Component,
        args: [{
                selector: 'el-date-table',
                providers: [DateFormat],
                template: "\n    <table class=\"el-date-table\" cellspacing=\"0\" cellpadding=\"0\">\n      <tbody>\n      <tr>\n        <th *ngFor=\"let week of weeks\">{{week}}</th>\n      </tr>\n      <tr class=\"el-date-table__row\"\n          *ngFor=\"let row of tableRows\">\n        <td *ngFor=\"let item of row\"\n          [class.available]=\"item.monthOffset === 0\"\n          [class.next-month]=\"item.monthOffset === 1\"\n          [class.prev-month]=\"item.monthOffset === -1\"\n          [class.normal]=\"item.monthOffset === 0\"\n          [class.today]=\"isToday(item)\"\n          [class.current]=\"isTargetDay(item)\"\n          (click)=\"clickHandle(item)\">\n          <div>\n            <span>{{item.day}}</span>\n          </div>\n        </td>\n      </tr>\n      </tbody>\n    </table>\n  "
            }]
    }], function () { return []; }, { modelChange: [{
            type: Output
        }], model: [{
            type: Input
        }], disabledDate: [{
            type: Input,
            args: ['disabled-date']
        }] }); })();
    return ElDateTable;
}());
export { ElDateTable };
function ElDateTable_tsickle_Closure_declarations() {
    /** @type {?} */
    ElDateTable.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElDateTable.ctorParameters;
    /** @type {?} */
    ElDateTable.propDecorators;
    /** @type {?} */
    ElDateTable.prototype.model;
    /** @type {?} */
    ElDateTable.prototype.disabledDate;
    /** @type {?} */
    ElDateTable.prototype.modelChange;
    /** @type {?} */
    ElDateTable.prototype.weeks;
    /** @type {?} */
    ElDateTable.prototype.tableRows;
    /** @type {?} */
    ElDateTable.prototype.targetDay;
    /** @type {?} */
    ElDateTable.prototype.targetMonthOffset;
    /** @type {?} */
    ElDateTable.prototype.date;
    /** @type {?} */
    ElDateTable.prototype.today;
    /** @type {?} */
    ElDateTable.prototype.currentMonthOffset;
}
