// element-angular/__ivy_ngcc__/release/date-picker/children/month-table.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function EMonthTable_tr_2_td_1_Template(rf, ctx) { if (rf & 1) {
    var _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 3);
    ɵngcc0.ɵɵlistener("click", function EMonthTable_tr_2_td_1_Template_td_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r8); var k_r5 = ctx.index; var i_r2 = ɵngcc0.ɵɵnextContext().index; var ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.clickHandle(i_r2, k_r5); });
    ɵngcc0.ɵɵelementStart(1, "a", 4);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r4 = ctx.$implicit;
    var k_r5 = ctx.index;
    var i_r2 = ɵngcc0.ɵɵnextContext().index;
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("current", ctx_r3.isCurrentMonth(i_r2, k_r5));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r4);
} }
function EMonthTable_tr_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵtemplate(1, EMonthTable_tr_2_td_1_Template, 3, 3, "td", 2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var row_r1 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", row_r1);
} }
var EMonthTable = /** @class */ (function () {
    function EMonthTable() {
        this.showWeekNumber = false;
        this.modelChange = new EventEmitter();
        this.monthRows = [
            ['一月', '二月', '三月', '四月'],
            ['五月', '六月', '七月', '八月'],
            ['九月', '十月', '十一月', '十二月'],
        ];
    }
    /**
     * @param {?} i
     * @param {?} k
     * @return {?}
     */
    EMonthTable.prototype.clickHandle = function (i, k) {
        var /** @type {?} */ monthID = 4 * i + k;
        this.currentMonth = monthID;
        this.date.setMonth(monthID);
        this.modelChange.emit(this.date.getTime());
    };
    /**
     * @param {?} i
     * @param {?} k
     * @return {?}
     */
    EMonthTable.prototype.isCurrentMonth = function (i, k) {
        return this.currentMonth === 4 * i + k;
    };
    /**
     * @return {?}
     */
    EMonthTable.prototype.ngOnInit = function () {
        this.date = new Date(this.model);
        this.currentMonth = this.date.getMonth();
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    EMonthTable.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.model)
            return;
        // first change
        if (changes.model.isFirstChange())
            return;
        this.model = changes.model.currentValue;
        this.date = new Date(this.model);
        this.currentMonth = this.date.getMonth();
    };
    /**
     * @nocollapse
     */
    EMonthTable.ctorParameters = function () { return []; };
    EMonthTable.propDecorators = {
        'showWeekNumber': [{ type: Input },],
        'model': [{ type: Input },],
        'disabledDate': [{ type: Input, args: ['disabled-date',] },],
        'modelChange': [{ type: Output },],
    };
EMonthTable.ɵfac = function EMonthTable_Factory(t) { return new (t || EMonthTable)(); };
EMonthTable.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EMonthTable, selectors: [["el-month-table"]], inputs: { showWeekNumber: "showWeekNumber", model: "model", disabledDate: ["disabled-date", "disabledDate"] }, outputs: { modelChange: "modelChange" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 1, consts: [[1, "el-month-table"], [4, "ngFor", "ngForOf"], ["class", "available", 3, "current", "click", 4, "ngFor", "ngForOf"], [1, "available", 3, "click"], [1, "cell"]], template: function EMonthTable_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "table", 0);
        ɵngcc0.ɵɵelementStart(1, "tbody");
        ɵngcc0.ɵɵtemplate(2, EMonthTable_tr_2_Template, 2, 1, "tr", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.monthRows);
    } }, directives: [ɵngcc1.NgForOf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EMonthTable, [{
        type: Component,
        args: [{
                selector: 'el-month-table',
                template: "\n    <table class=\"el-month-table\">\n      <tbody>\n      <tr *ngFor=\"let row of monthRows; let i = index;\">\n        <td class=\"available\"\n            *ngFor=\"let item of row; let k = index;\"\n            [class.current]=\"isCurrentMonth(i, k)\"\n            (click)=\"clickHandle(i, k)\">\n          <a class=\"cell\">{{ item }}</a>\n        </td>\n      </tr>\n      </tbody>\n    </table>\n  "
            }]
    }], function () { return []; }, { showWeekNumber: [{
            type: Input
        }], modelChange: [{
            type: Output
        }], model: [{
            type: Input
        }], disabledDate: [{
            type: Input,
            args: ['disabled-date']
        }] }); })();
    return EMonthTable;
}());
export { EMonthTable };
function EMonthTable_tsickle_Closure_declarations() {
    /** @type {?} */
    EMonthTable.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    EMonthTable.ctorParameters;
    /** @type {?} */
    EMonthTable.propDecorators;
    /** @type {?} */
    EMonthTable.prototype.showWeekNumber;
    /** @type {?} */
    EMonthTable.prototype.model;
    /** @type {?} */
    EMonthTable.prototype.disabledDate;
    /** @type {?} */
    EMonthTable.prototype.modelChange;
    /** @type {?} */
    EMonthTable.prototype.currentMonth;
    /** @type {?} */
    EMonthTable.prototype.date;
    /** @type {?} */
    EMonthTable.prototype.monthRows;
}
