// element-angular/__ivy_ngcc__/release/select/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ElTagsModule } from '../tag/module';
import { ElInputsModule } from '../input/module';
import { ElSelectDropdown } from './select-dropdown';
import { ElOption } from './option';
import { ElSelect } from './select';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../tag/tag';
import * as ɵngcc3 from '../input/input';
var ElSelectModule = /** @class */ (function () {
    function ElSelectModule() {
    }
    /**
     * @return {?}
     */
    ElSelectModule.forRoot = function () {
        return { ngModule: ElSelectModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElSelectModule.ctorParameters = function () { return []; };
ElSelectModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElSelectModule });
ElSelectModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElSelectModule_Factory(t) { return new (t || ElSelectModule)(); }, imports: [[CommonModule, FormsModule, ElTagsModule, ElInputsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElSelectModule, { declarations: function () { return [ElSelectDropdown, ElOption, ElSelect]; }, imports: function () { return [CommonModule, FormsModule, ElTagsModule, ElInputsModule]; }, exports: function () { return [ElSelectDropdown, ElOption, ElSelect]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElSelectModule, [{
        type: NgModule,
        args: [{
                declarations: [ElSelectDropdown, ElOption, ElSelect],
                exports: [ElSelectDropdown, ElOption, ElSelect],
                imports: [CommonModule, FormsModule, ElTagsModule, ElInputsModule],
                entryComponents: [ElSelectDropdown, ElOption, ElSelect]
            }]
    }], function () { return []; }, null); })();
ɵngcc0.ɵɵsetComponentScope(ElSelect, [ɵngcc1.NgIf, ɵngcc1.NgForOf, ɵngcc2.ElTag, ɵngcc3.ElInput, ElSelectDropdown], []);
    return ElSelectModule;
}());
export { ElSelectModule };
function ElSelectModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElSelectModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElSelectModule.ctorParameters;
}
