// element-angular/__ivy_ngcc__/release/select/select.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../shared/services/index';

var _c0 = ["tags"];
var _c1 = ["input"];
function ElSelect_div_1_el_tag_2_Template(rf, ctx) { if (rf & 1) {
    var _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-tag", 9);
    ɵngcc0.ɵɵlistener("close", function ElSelect_div_1_el_tag_2_Template_el_tag_close_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r7); var tag_r4 = ctx.$implicit; var i_r5 = ctx.index; var ctx_r6 = ɵngcc0.ɵɵnextContext(2); $event.stopPropagation(); return ctx_r6.changeLabel(tag_r4, ctx_r6.model[i_r5]); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var tag_r4 = ctx.$implicit;
    var ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("closable", !ctx_r3.elDisabled)("size", ctx_r3.size);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(tag_r4);
} }
function ElSelect_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6, 7);
    ɵngcc0.ɵɵtemplate(2, ElSelect_div_1_el_tag_2_Template, 2, 3, "el-tag", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.multipleLabels);
} }
var _c2 = ["*"];
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, ElementRef, Renderer2, forwardRef, ViewChild, } from '@angular/core';
import { ElSelectPoprs } from './select-props';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { WindowWrapper } from '../shared/services/index';
var ElSelect = /** @class */ (function (_super) {
    __extends(ElSelect, _super);
    /**
     * @param {?} el
     * @param {?} renderer
     * @param {?} window
     */
    function ElSelect(el, renderer, window) {
        var _this = _super.call(this) || this;
        _this.el = el;
        _this.renderer = renderer;
        _this.window = window;
        _this.subscriber = [];
        _this.multipleLabels = [];
        _this.multiplePlaceholder = _this.placeholder;
        _this.dropdownActive = false;
        _this.iconClass = 'arrow-up';
        _this.selectOptions = [];
        _this.controlChange = function () { };
        _this.controlTouch = function () { };
        return _this;
    }
    /**
     * @param {?=} isEnter
     * @return {?}
     */
    ElSelect.prototype.mouseHandle = function (isEnter) {
        if (isEnter === void 0) { isEnter = false; }
        this.clearable = this.clearable && !this.multiple;
        if (!this.clearable || !this.model)
            return;
        this.iconClass = "" + (isEnter ? 'circle-close is-reverse' : 'arrow-up');
    };
    /**
     * @param {?=} event
     * @return {?}
     */
    ElSelect.prototype.toggleHandle = function (event) {
        this.clearable = this.clearable && !this.multiple;
        if (this.elDisabled)
            return;
        event && event.stopPropagation();
        this.dropdownActive = !this.dropdownActive;
        var /** @type {?} */ nextClass = 'arrow-up' + (this.dropdownActive ? ' is-reverse' : '');
        this.iconClass = !this.clearable ? nextClass : this.iconClass;
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ElSelect.prototype.clearSelected = function (event) {
        this.clearable = this.clearable && !this.multiple;
        if (!this.clearable)
            return;
        event.stopPropagation();
        // reset icon
        this.mouseHandle(false);
        // reset selected label
        this.changeLabel(null, null);
        // reset model
        this.model = null;
        this.modelChange.emit(this.model);
        this.controlChange(this.model);
        this.subscriber.forEach(function (sub) { return sub(); });
        // close dropdown menu
        this.dropdownActive = false;
    };
    /**
     * @param {?} nextLabel
     * @param {?=} nextValue
     * @return {?}
     */
    ElSelect.prototype.changeLabel = function (nextLabel, nextValue) {
        this.dropdownActive && this.toggleHandle();
        // only update label
        this.selectedLabel = this.multiple ? '' : nextLabel;
        if (!nextValue || this.model === nextValue)
            return;
        if (this.multiple) {
            this.updateValueWithMultipleMode(nextLabel, nextValue);
            this.updateLayoutWithMultipleMode();
        }
        else {
            this.model = nextValue;
        }
        this.modelChange.emit(this.model);
        this.controlChange(this.model);
        this.subscriber.forEach(function (sub) { return sub(); });
    };
    /**
     * @param {?} item
     * @return {?}
     */
    ElSelect.prototype.appendOptions = function (item) {
        this.selectOptions.push(item);
    };
    /**
     * @return {?}
     */
    ElSelect.prototype.ngOnInit = function () {
        var _this = this;
        var /** @type {?} */ timer = window.setTimeout(function () {
            _this.selfWidth = _this.el.nativeElement.getBoundingClientRect().width;
            clearTimeout(timer);
        }, 0);
        this.globalListener = this.renderer.listen('document', 'click', function () {
            _this.dropdownActive && _this.toggleHandle();
        });
        this.updatePlaceholderWithMultipleMode();
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElSelect.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.model)
            return;
        if (changes.model.isFirstChange())
            return;
        // reset model
        if (!changes.model.currentValue) {
            this.selectedLabel = changes.model.currentValue;
            this.model = changes.model.currentValue;
            this.modelChange.emit(changes.model.currentValue);
            this.controlChange(this.model);
        }
        this.subscriber.forEach(function (sub) { return sub(); });
    };
    /**
     * @return {?}
     */
    ElSelect.prototype.ngOnDestroy = function () {
        this.globalListener && this.globalListener();
    };
    /**
     * @return {?}
     */
    ElSelect.prototype.ngAfterViewInit = function () {
        var _this = this;
        var /** @type {?} */ timer = this.window.setTimeout(function () {
            _this.initModelWithMultipleMode();
            _this.window.clearTimeout(timer);
        }, 0);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    ElSelect.prototype.writeValue = function (value) {
        this.model = value;
        this.initModelWithMultipleMode();
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElSelect.prototype.registerOnChange = function (fn) {
        this.controlChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElSelect.prototype.registerOnTouched = function (fn) {
        this.controlTouch = fn;
    };
    /**
     * @return {?}
     */
    ElSelect.prototype.updateLayoutWithMultipleMode = function () {
        var _this = this;
        var /** @type {?} */ updateHandle = function () {
            if (!_this.tags)
                return;
            var /** @type {?} */ children = _this.tags.nativeElement && _this.tags.nativeElement.children;
            var /** @type {?} */ inputEl = _this.input.el.nativeElement;
            if (!children || !children.length || !inputEl)
                return;
            var /** @type {?} */ inputWidth = inputEl.getBoundingClientRect().width;
            var /** @type {?} */ unit = inputWidth - 34;
            var /** @type {?} */ row = 1;
            Array.from(children).reduce(function (count, el) {
                var /** @type {?} */ currentWidth = el.getBoundingClientRect().width || 80;
                if (count + currentWidth < unit)
                    return count + currentWidth;
                // add a row
                row++;
                return currentWidth;
            }, 0);
            var /** @type {?} */ el = inputEl.querySelector('.el-input__inner');
            _this.renderer.setStyle(el, 'height', Math.ceil(row) * 40 + "px");
        };
        var /** @type {?} */ timer = this.window.setTimeout(function () {
            updateHandle();
            _this.window.clearTimeout(timer);
        }, 0);
    };
    /**
     * @param {?} nextLabel
     * @param {?=} nextValue
     * @return {?}
     */
    ElSelect.prototype.updateValueWithMultipleMode = function (nextLabel, nextValue) {
        var /** @type {?} */ model = Array.isArray(this.model)
            ? (this.model.indexOf(nextValue) > -1 ? this.model.filter(function (v) { return v !== nextValue; }) : this.model.concat(nextValue))
            : [nextValue];
        this.model = model.filter(function (r) { return r !== undefined; });
        this.multipleLabels = !nextLabel || this.multipleLabels.indexOf(nextLabel) > -1
            ? this.multipleLabels.filter(function (v) { return v !== nextLabel; })
            : this.multipleLabels.concat(nextLabel);
        this.updatePlaceholderWithMultipleMode();
    };
    /**
     * @return {?}
     */
    ElSelect.prototype.updatePlaceholderWithMultipleMode = function () {
        if (!this.multiple)
            return;
        this.multiplePlaceholder = this.model && this.model.length ? '' : this.placeholder;
    };
    /**
     * @return {?}
     */
    ElSelect.prototype.initModelWithMultipleMode = function () {
        var _this = this;
        if (!this.model || !this.multiple || !this.selectOptions.length)
            return;
        if (!Array.isArray(this.model))
            this.model = [this.model];
        this.multipleLabels = this.model
            .map(function (item) {
            var /** @type {?} */ option = _this.selectOptions.find(function (option) { return option.value === item; });
            if (option && option.label)
                return option.label;
            return null;
        })
            .filter(function (r) { return !!r; });
        this.updateLayoutWithMultipleMode();
    };
    /**
     * @nocollapse
     */
    ElSelect.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: Renderer2, },
        { type: WindowWrapper, },
    ]; };
    ElSelect.propDecorators = {
        'tags': [{ type: ViewChild, args: ['tags',] },],
        'input': [{ type: ViewChild, args: ['input',] },],
    };
ElSelect.ɵfac = function ElSelect_Factory(t) { return new (t || ElSelect)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.WindowWrapper)); };
ElSelect.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElSelect, selectors: [["el-select"]], viewQuery: function ElSelect_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
        ɵngcc0.ɵɵviewQuery(_c1, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.tags = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.input = _t.first);
    } }, features: [ɵngcc0.ɵɵProvidersFeature([{
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return ElSelect; }),
                multi: true
            }]), ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c2, decls: 8, vars: 9, consts: [[1, "el-select", 3, "click"], ["class", "el-select__tags el-select__tags__padding", 4, "ngIf"], ["iconClass", "el-select__caret", 3, "model", "placeholder", "icon", "name", "size", "elDisabled", "readonly", "mouseenter", "mouseleave", "icon-click"], ["input", ""], [3, "isActived"], [1, "el-scrollbar__view", "el-select-dropdown__list"], [1, "el-select__tags", "el-select__tags__padding"], ["tags", ""], ["type", "info", 3, "closable", "size", "close", 4, "ngFor", "ngForOf"], ["type", "info", 3, "closable", "size", "close"]], template: function ElSelect_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("click", function ElSelect_Template_div_click_0_listener($event) { return ctx.toggleHandle($event); });
        ɵngcc0.ɵɵtemplate(1, ElSelect_div_1_Template, 3, 1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "el-input", 2, 3);
        ɵngcc0.ɵɵlistener("mouseenter", function ElSelect_Template_el_input_mouseenter_2_listener() { return ctx.mouseHandle(true); })("mouseleave", function ElSelect_Template_el_input_mouseleave_2_listener() { return ctx.mouseHandle(false); })("icon-click", function ElSelect_Template_el_input_icon_click_2_listener($event) { return ctx.clearSelected($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementContainerStart(4);
        ɵngcc0.ɵɵelementStart(5, "el-select-dropdown", 4);
        ɵngcc0.ɵɵelementStart(6, "ul", 5);
        ɵngcc0.ɵɵprojection(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.multiple && ctx.model && ctx.model.length);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("model", ctx.selectedLabel)("placeholder", ctx.multiple ? ctx.multiplePlaceholder : ctx.placeholder)("icon", ctx.iconClass)("name", ctx.name)("size", ctx.size)("elDisabled", ctx.elDisabled)("readonly", true);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("isActived", ctx.dropdownActive);
    } }, styles: [".el-select-dropdown__list[_ngcontent-%COMP%] { overflow: hidden; }\n    .el-select__tags__padding[_ngcontent-%COMP%] { padding-right: 30px; }"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElSelect, [{
        type: Component,
        args: [{
                selector: 'el-select',
                providers: [{
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(function () { return ElSelect; }),
                        multi: true
                    }],
                styles: ["\n    .el-select-dropdown__list { overflow: hidden; }\n    .el-select__tags__padding { padding-right: 30px; }\n  "],
                template: "\n    <div class=\"el-select\" (click)=\"toggleHandle($event)\">\n      <div class=\"el-select__tags el-select__tags__padding\" *ngIf=\"multiple && model && model.length\" #tags>\n        <el-tag *ngFor=\"let tag of multipleLabels; let i = index\"\n          [closable]=\"!elDisabled\"\n          [size]=\"size\"\n          (close)=\"$event.stopPropagation();changeLabel(tag, model[i])\"\n          type=\"info\">{{tag}}</el-tag>\n      </div>\n      \n      <el-input iconClass=\"el-select__caret\" #input\n        [model]=\"selectedLabel\"\n        [placeholder]=\"multiple ? multiplePlaceholder : placeholder\"\n        [icon]=\"iconClass\"\n        [name]=\"name\"\n        [size]=\"size\"\n        [elDisabled]=\"elDisabled\" [readonly]=\"true\"\n        (mouseenter)=\"mouseHandle(true)\" (mouseleave)=\"mouseHandle(false)\"\n        (icon-click)=\"clearSelected($event)\">\n      </el-input>\n      <ng-container>\n        <el-select-dropdown [isActived]=\"dropdownActive\">\n          <ul class=\"el-scrollbar__view el-select-dropdown__list\">\n            <ng-content></ng-content>\n          </ul>\n        </el-select-dropdown>\n      </ng-container>\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc1.WindowWrapper }]; }, { tags: [{
            type: ViewChild,
            args: ['tags']
        }], input: [{
            type: ViewChild,
            args: ['input']
        }] }); })();
    return ElSelect;
}(ElSelectPoprs));
export { ElSelect };
function ElSelect_tsickle_Closure_declarations() {
    /** @type {?} */
    ElSelect.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElSelect.ctorParameters;
    /** @type {?} */
    ElSelect.propDecorators;
    /** @type {?} */
    ElSelect.prototype.tags;
    /** @type {?} */
    ElSelect.prototype.input;
    /** @type {?} */
    ElSelect.prototype.selfWidth;
    /** @type {?} */
    ElSelect.prototype.subscriber;
    /** @type {?} */
    ElSelect.prototype.multipleLabels;
    /** @type {?} */
    ElSelect.prototype.multiplePlaceholder;
    /** @type {?} */
    ElSelect.prototype.dropdownActive;
    /** @type {?} */
    ElSelect.prototype.selectedLabel;
    /** @type {?} */
    ElSelect.prototype.iconClass;
    /** @type {?} */
    ElSelect.prototype.globalListener;
    /** @type {?} */
    ElSelect.prototype.selectOptions;
    /** @type {?} */
    ElSelect.prototype.controlChange;
    /** @type {?} */
    ElSelect.prototype.controlTouch;
    /** @type {?} */
    ElSelect.prototype.el;
    /** @type {?} */
    ElSelect.prototype.renderer;
    /** @type {?} */
    ElSelect.prototype.window;
}
