// element-angular/__ivy_ngcc__/release/button/button-group.js
import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var ElButtonGroup = /** @class */ (function () {
    function ElButtonGroup() {
        this.nativeClass = '';
    }
    /**
     * @nocollapse
     */
    ElButtonGroup.ctorParameters = function () { return []; };
    ElButtonGroup.propDecorators = {
        'nativeClass': [{ type: Input, args: ['class',] },],
    };
ElButtonGroup.ɵfac = function ElButtonGroup_Factory(t) { return new (t || ElButtonGroup)(); };
ElButtonGroup.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElButtonGroup, selectors: [["el-button-group"]], inputs: { nativeClass: ["class", "nativeClass"] }, ngContentSelectors: _c0, decls: 2, vars: 2, template: function ElButtonGroup_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-button-group " + ctx.nativeClass);
    } }, encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElButtonGroup, [{
        type: Component,
        args: [{
                selector: 'el-button-group',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: "\n    <div [class]=\"'el-button-group ' + nativeClass\">\n      <ng-content></ng-content>\n    </div>\n  "
            }]
    }], function () { return []; }, { nativeClass: [{
            type: Input,
            args: ['class']
        }] }); })();
    return ElButtonGroup;
}());
export { ElButtonGroup };
function ElButtonGroup_tsickle_Closure_declarations() {
    /** @type {?} */
    ElButtonGroup.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElButtonGroup.ctorParameters;
    /** @type {?} */
    ElButtonGroup.propDecorators;
    /** @type {?} */
    ElButtonGroup.prototype.nativeClass;
}
