// element-angular/__ivy_ngcc__/release/button/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElButton } from './button';
import { ElButtonGroup } from './button-group';
import * as ɵngcc0 from '@angular/core';
var ElButtonsModule = /** @class */ (function () {
    function ElButtonsModule() {
    }
    /**
     * @return {?}
     */
    ElButtonsModule.forRoot = function () {
        return { ngModule: ElButtonsModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElButtonsModule.ctorParameters = function () { return []; };
ElButtonsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElButtonsModule });
ElButtonsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElButtonsModule_Factory(t) { return new (t || ElButtonsModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElButtonsModule, { declarations: function () { return [ElButton, ElButtonGroup]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElButton, ElButtonGroup]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElButtonsModule, [{
        type: NgModule,
        args: [{
                declarations: [ElButton, ElButtonGroup],
                exports: [ElButton, ElButtonGroup],
                imports: [CommonModule],
                entryComponents: [ElButton, ElButtonGroup]
            }]
    }], function () { return []; }, null); })();
    return ElButtonsModule;
}());
export { ElButtonsModule };
function ElButtonsModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElButtonsModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElButtonsModule.ctorParameters;
}
