// element-angular/__ivy_ngcc__/release/carousel/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElCarousel } from './carousel';
import { ElCarouselItem } from './carousel-item';
import * as ɵngcc0 from '@angular/core';
var ElCarouselModule = /** @class */ (function () {
    function ElCarouselModule() {
    }
    /**
     * @return {?}
     */
    ElCarouselModule.forRoot = function () {
        return { ngModule: ElCarouselModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElCarouselModule.ctorParameters = function () { return []; };
ElCarouselModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElCarouselModule });
ElCarouselModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElCarouselModule_Factory(t) { return new (t || ElCarouselModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElCarouselModule, { declarations: function () { return [ElCarousel, ElCarouselItem]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElCarousel, ElCarouselItem]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCarouselModule, [{
        type: NgModule,
        args: [{
                declarations: [ElCarousel, ElCarouselItem],
                exports: [ElCarousel, ElCarouselItem],
                imports: [CommonModule],
                entryComponents: [ElCarousel]
            }]
    }], function () { return []; }, null); })();
    return ElCarouselModule;
}());
export { ElCarouselModule };
function ElCarouselModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCarouselModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCarouselModule.ctorParameters;
}
