// element-angular/__ivy_ngcc__/release/table/table.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../shared/services/index';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from './children/header';
import * as ɵngcc4 from './children/body';
import * as ɵngcc5 from '../shared/pipe/css-value';

var _c0 = ["headerRef"];
var _c1 = function (a0) { return { width: a0 }; };
function ElTable_div_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 8);
    ɵngcc0.ɵɵelementStart(1, "span", 9);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(2, _c1, ctx_r1.layout.bodyWidth + "px"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.placeholder);
} }
var _c2 = function (a0) { return { height: a0 }; };
var _c3 = function (a0) { return { "overflow-x": a0 }; };
var _c4 = ["*"];
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
import { Component, ElementRef, KeyValueDiffers, Renderer2, ViewChild, } from '@angular/core';
import { DocumentWrapper, WindowWrapper } from '../shared/services/index';
import { ElTableProps } from './table.props';
import { ElTableFormat } from './utils/format';
var ElTable = /** @class */ (function (_super) {
    __extends(ElTable, _super);
    /**
     * @param {?} el
     * @param {?} renderer
     * @param {?} document
     * @param {?} window
     * @param {?} differs
     */
    function ElTable(el, renderer, document, window, differs) {
        var _this = _super.call(this) || this;
        _this.el = el;
        _this.renderer = renderer;
        _this.document = document;
        _this.window = window;
        _this.differs = differs;
        _this.columnsWithLevel = [];
        _this.layout = {
            headerHeight: 40,
            bodyHeight: 'auto',
            bodyWidth: 'auto',
            scrollBarWidth: 0,
        };
        _this.columnsWidth = [];
        _this.columns = [];
        _this.widthCount = 0;
        _this.differ = _this.differs.find([]).create();
        return _this;
    }
    /**
     * @return {?}
     */
    ElTable.GEN_TEMPLATE_KEY = function () {
        return Math.random().toString(16).substr(2, 8);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ElTable.prototype.bodyScroll = function (event) {
        if (!this.scrollX)
            return;
        var /** @type {?} */ el = ((event.target));
        if (el.scrollLeft === undefined)
            return;
        this.headerRef.nativeElement.scrollLeft = el.scrollLeft;
    };
    /**
     * @param {?} column
     * @return {?}
     */
    ElTable.prototype.updateColumns = function (column) {
        var /** @type {?} */ next = Object.assign(column, { index: this.columns.length });
        this.columns.push(next);
    };
    /**
     * @return {?}
     */
    ElTable.prototype.updateBodyHeight = function () {
        var _this = this;
        var /** @type {?} */ height = ElTableFormat.getCSSValue(this.height);
        var /** @type {?} */ header = this.headerRef.nativeElement;
        if (!header || !height)
            return;
        var /** @type {?} */ timer = setTimeout(function () {
            var /** @type {?} */ headerHeight = header.offsetHeight;
            if (headerHeight) {
                _this.layout.headerHeight = headerHeight;
                _this.layout.bodyHeight = height - _this.layout.headerHeight;
                _this.layout.scrollBarWidth = _this.window.innerWidth - _this.document.body.clientWidth;
            }
            clearTimeout(timer);
        }, 0);
    };
    /**
     * @return {?}
     */
    ElTable.prototype.updateLayout = function () {
        var /** @type {?} */ elTable = this.el.nativeElement.children[0];
        this.layout.bodyWidth = this.widthCount || elTable.clientWidth;
        if (this.widthCount) {
            this.renderer.setStyle(elTable, 'width', this.widthCount + "px");
        }
    };
    /**
     * @param {?} widthItem
     * @return {?}
     */
    ElTable.prototype.updateColumnsWidth = function (widthItem) {
        this.columnsWidth.push(widthItem);
    };
    /**
     * @param {?} columnsWidth
     * @return {?}
     */
    ElTable.prototype.computeColumnsWidth = function (columnsWidth) {
        var /** @type {?} */ auto = 0, /** @type {?} */ count = 0;
        columnsWidth.forEach(function (item) {
            if (item.auto)
                return auto++;
            if (Number.isNaN(item.width)) {
                item.auto = true;
                return auto++;
            }
            count += item.width;
        });
        // if user has set the width, use fixed width
        // update layout
        if (!auto) {
            this.widthCount = count;
            this.updateLayout();
        }
        var /** @type {?} */ average = (this.layout.bodyWidth - count) / auto;
        return columnsWidth.map(function (item) {
            return item.auto ? Object.assign(item, { width: average }) : item;
        });
    };
    /**
     * @return {?}
     */
    ElTable.prototype.transformColumnsData = function () {
        var _this = this;
        // order by deep
        this.columns = this.columns.map(function (column) {
            if (!Array.isArray(_this.columnsWithLevel[column.level])) {
                _this.columnsWithLevel[column.level] = [];
            }
            _this.columnsWithLevel[column.level].push(column);
            if (column.deep === 0)
                return column;
            return null;
        }).filter(function (r) { return r; });
        this.columnsWithLevel.reverse();
        this.columnsWidth = this.computeColumnsWidth(this.columnsWidth);
        // distribution template
        this.columns = this.columns.map(function (column) {
            if (!column.slot)
                return column;
            var /** @type {?} */ TEMPLATE_KEY = ElTable.GEN_TEMPLATE_KEY();
            _this.modelStorge = _this.model.map(function (v) {
                return Object.assign(v, (_a = {}, _a[TEMPLATE_KEY] = column.slot, _a));
                var _a;
            });
            return Object.assign(column, { modelKey: TEMPLATE_KEY });
        });
        this.orderMap = this.columns.reduce(function (pre, next) {
            return Object.assign(pre, (_a = {}, _a[next.modelKey] = next, _a));
            var _a;
        }, {});
        this.transformModelData();
    };
    /**
     * @return {?}
     */
    ElTable.prototype.transformModelData = function () {
        var /** @type {?} */ orderMap = this.orderMap;
        // add index, width, value
        if (!this.modelStorge) {
            this.modelStorge = this.model;
        }
        var /** @type {?} */ modelWithIndex = this.modelStorge.map(function (row) {
            return Object.keys(__assign({}, row, orderMap)).map(function (v) {
                var /** @type {?} */ item = orderMap[v] || {};
                return _a = {
                        hidden: !item.width,
                        value: row[v]
                    },
                    _a[v] = row[v],
                    _a.index = ~~item.index,
                    _a.width = item.width,
                    _a._renderHTML = item._renderHTML,
                    _a;
                var _a;
            })
                .filter(function (r) { return !!r; });
        });
        // column sort
        this.columnsData = modelWithIndex.map(function (row) {
            return row.sort(function (pre, next) { return pre.index - next.index; });
        });
        this.updateBodyHeight();
    };
    /**
     * @return {?}
     */
    ElTable.prototype.ngOnInit = function () {
        var _this = this;
        this.updateLayout();
        this.updateBodyHeight();
        this.globalListenFunc = this.renderer.listen('window', 'resize', function () {
            _this.updateLayout();
            _this.columnsWidth = _this.computeColumnsWidth(_this.columnsWidth);
        });
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElTable.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.model)
            return;
        // first change
        if (changes.model.isFirstChange())
            return;
        this.model = changes.model.currentValue;
        this.transformModelData();
    };
    /**
     * @return {?}
     */
    ElTable.prototype.ngDoCheck = function () {
        var _this = this;
        var /** @type {?} */ change = this.differ.diff(this.model);
        if (!change || !this.orderMap)
            return;
        // distribution template
        this.modelStorge = this.model;
        var /** @type {?} */ nextColumns = this.columns.map(function (column) {
            if (!column.slot)
                return column;
            var /** @type {?} */ TEMPLATE_KEY = ElTable.GEN_TEMPLATE_KEY();
            _this.modelStorge = _this.model.map(function (v) {
                return Object.assign(v, (_a = {}, _a[TEMPLATE_KEY] = column.slot, _a));
                var _a;
            });
            return Object.assign(column, { modelKey: TEMPLATE_KEY });
        });
        this.orderMap = nextColumns.reduce(function (pre, next) {
            return Object.assign(pre, (_a = {}, _a[next.modelKey] = next, _a));
            var _a;
        }, {});
        this.transformModelData();
    };
    /**
     * @return {?}
     */
    ElTable.prototype.ngOnDestroy = function () {
        this.globalListenFunc && this.globalListenFunc();
    };
    /**
     * @nocollapse
     */
    ElTable.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: Renderer2, },
        { type: DocumentWrapper, },
        { type: WindowWrapper, },
        { type: KeyValueDiffers, },
    ]; };
    ElTable.propDecorators = {
        'headerRef': [{ type: ViewChild, args: ['headerRef',] },],
    };
ElTable.ɵfac = function ElTable_Factory(t) { return new (t || ElTable)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DocumentWrapper), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.WindowWrapper), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.KeyValueDiffers)); };
ElTable.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElTable, selectors: [["el-table"]], viewQuery: function ElTable_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.headerRef = _t.first);
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c4, decls: 11, vars: 37, consts: [[1, "el-table", 3, "ngStyle"], [1, "hidden-columns"], [1, "el-table__header-wrapper", "el-table__header-scroll", 3, "hidden", "ngStyle"], ["headerRef", ""], [3, "model", "layout", "columns-width", "border", "height", "center"], [1, "el-table__body-wrapper", 3, "ngStyle", "scroll"], [3, "model", "stripe", "layout", "row-class-name", "center", "ngStyle"], ["class", "el-table__empty-block", 3, "ngStyle", 4, "ngIf"], [1, "el-table__empty-block", 3, "ngStyle"], [1, "el-table__empty-text"]], template: function ElTable_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵpipe(1, "cssValue");
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵprojection(3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "div", 2, 3);
        ɵngcc0.ɵɵelement(6, "el-table-header", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 5);
        ɵngcc0.ɵɵlistener("scroll", function ElTable_Template_div_scroll_7_listener($event) { return ctx.bodyScroll($event); });
        ɵngcc0.ɵɵpipe(8, "cssValue");
        ɵngcc0.ɵɵelement(9, "el-table-body", 6);
        ɵngcc0.ɵɵtemplate(10, ElTable_div_10_Template, 3, 4, "div", 7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("el-table--enable-row-transition", true)("el-table--striped", ctx.stripe)("el-table--border", ctx.border)("el-table--hidden", false);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(29, _c2, ɵngcc0.ɵɵpipeBind1(1, 25, ctx.height)));
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("hidden", !ctx.showHeader)("ngStyle", ɵngcc0.ɵɵpureFunction1(31, _c3, ctx.scrollX ? "auto" : "hidden"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("model", ctx.columnsWithLevel)("layout", ctx.layout)("columns-width", ctx.columnsWidth)("border", ctx.border)("height", ctx.height)("center", ctx.center === "header" || ctx.center === "all");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(33, _c2, ɵngcc0.ɵɵpipeBind1(8, 27, ctx.layout.bodyHeight)));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("model", ctx.columnsData)("stripe", ctx.stripe)("layout", ctx.layout)("row-class-name", ctx.rowClassName)("center", ctx.center === "all")("ngStyle", ɵngcc0.ɵɵpureFunction1(35, _c1, ctx.layout.bodyWidth + "px"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.model || ctx.model.length === 0);
    } }, directives: [ɵngcc2.NgStyle, ɵngcc3.ElTableHeader, ɵngcc4.ElTableBody, ɵngcc2.NgIf], pipes: [ɵngcc5.ElCSSValuePipe], styles: [".el-table__header-scroll[_ngcontent-%COMP%]::-webkit-scrollbar { visibility: hidden; }"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElTable, [{
        type: Component,
        args: [{
                selector: 'el-table',
                styles: ["\n  .el-table__header-scroll::-webkit-scrollbar { visibility: hidden; }\n  "],
                template: "\n    <div class=\"el-table\"\n      [ngStyle]=\"{ height: height | cssValue }\"\n      [class.el-table--enable-row-transition]=\"true\"\n      [class.el-table--striped]=\"stripe\"\n      [class.el-table--border]=\"border\"\n      [class.el-table--hidden]=\"false\">\n      <div class=\"hidden-columns\"><ng-content></ng-content></div>\n      <div class=\"el-table__header-wrapper el-table__header-scroll\" [hidden]=\"!showHeader\"\n        [ngStyle]=\"{'overflow-x': (scrollX ? 'auto' : 'hidden')}\" #headerRef>\n        <el-table-header [model]=\"columnsWithLevel\" [layout]=\"layout\" [columns-width]=\"columnsWidth\"\n          [border]=\"border\" [height]=\"height\" [center]=\"center === 'header' || center === 'all'\">\n        </el-table-header>\n      </div>\n      <div class=\"el-table__body-wrapper\" [ngStyle]=\"{ height: layout.bodyHeight | cssValue }\"\n        (scroll)=\"bodyScroll($event)\">\n        <el-table-body [model]=\"columnsData\" [stripe]=\"stripe\"\n          [layout]=\"layout\" [row-class-name]=\"rowClassName\"\n          [center]=\"center === 'all'\"\n          [ngStyle]=\"{ width: layout.bodyWidth + 'px' }\">\n        </el-table-body>\n        <div [ngStyle]=\"{width: layout.bodyWidth + 'px'}\" class=\"el-table__empty-block\" *ngIf=\"!model || model.length === 0\">\n          <span class=\"el-table__empty-text\">{{placeholder}}</span>\n        </div>\n      </div>\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc1.DocumentWrapper }, { type: ɵngcc1.WindowWrapper }, { type: ɵngcc0.KeyValueDiffers }]; }, { headerRef: [{
            type: ViewChild,
            args: ['headerRef']
        }] }); })();
    return ElTable;
}(ElTableProps));
export { ElTable };
function ElTable_tsickle_Closure_declarations() {
    /** @type {?} */
    ElTable.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElTable.ctorParameters;
    /** @type {?} */
    ElTable.propDecorators;
    /** @type {?} */
    ElTable.prototype.headerRef;
    /** @type {?} */
    ElTable.prototype.columnsData;
    /** @type {?} */
    ElTable.prototype.columnsWithLevel;
    /** @type {?} */
    ElTable.prototype.layout;
    /** @type {?} */
    ElTable.prototype.columnsWidth;
    /** @type {?} */
    ElTable.prototype.columns;
    /** @type {?} */
    ElTable.prototype.globalListenFunc;
    /** @type {?} */
    ElTable.prototype.orderMap;
    /** @type {?} */
    ElTable.prototype.modelStorge;
    /** @type {?} */
    ElTable.prototype.differ;
    /** @type {?} */
    ElTable.prototype.widthCount;
    /** @type {?} */
    ElTable.prototype.el;
    /** @type {?} */
    ElTable.prototype.renderer;
    /** @type {?} */
    ElTable.prototype.document;
    /** @type {?} */
    ElTable.prototype.window;
    /** @type {?} */
    ElTable.prototype.differs;
}
