// element-angular/__ivy_ngcc__/release/badge/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElBadge } from './badge';
import * as ɵngcc0 from '@angular/core';
var ElBadgesModule = /** @class */ (function () {
    function ElBadgesModule() {
    }
    /**
     * @return {?}
     */
    ElBadgesModule.forRoot = function () {
        return { ngModule: ElBadgesModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElBadgesModule.ctorParameters = function () { return []; };
ElBadgesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElBadgesModule });
ElBadgesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElBadgesModule_Factory(t) { return new (t || ElBadgesModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElBadgesModule, { declarations: function () { return [ElBadge]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElBadge]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElBadgesModule, [{
        type: NgModule,
        args: [{
                declarations: [ElBadge],
                exports: [ElBadge],
                imports: [CommonModule],
                entryComponents: [ElBadge]
            }]
    }], function () { return []; }, null); })();
    return ElBadgesModule;
}());
export { ElBadgesModule };
function ElBadgesModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElBadgesModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElBadgesModule.ctorParameters;
}
