// element-angular/__ivy_ngcc__/release/shared/pipe/css-value.js
import { Pipe } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var ElCSSValuePipe = /** @class */ (function () {
    function ElCSSValuePipe() {
    }
    /**
     * @param {?} value
     * @return {?}
     */
    ElCSSValuePipe.prototype.transform = function (value) {
        if (typeof value === 'number')
            return value + "px";
        if (Number.isNaN(+value))
            return String(value);
        return value + "px";
    };
    /**
     * @nocollapse
     */
    ElCSSValuePipe.ctorParameters = function () { return []; };
ElCSSValuePipe.ɵfac = function ElCSSValuePipe_Factory(t) { return new (t || ElCSSValuePipe)(); };
ElCSSValuePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "cssValue", type: ElCSSValuePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCSSValuePipe, [{
        type: Pipe,
        args: [{ name: 'cssValue' }]
    }], function () { return []; }, null); })();
    return ElCSSValuePipe;
}());
export { ElCSSValuePipe };
function ElCSSValuePipe_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCSSValuePipe.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCSSValuePipe.ctorParameters;
}
