// element-angular/__ivy_ngcc__/release/rate/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElRate } from './rate';
import * as ɵngcc0 from '@angular/core';
var ElRateModule = /** @class */ (function () {
    function ElRateModule() {
    }
    /**
     * @return {?}
     */
    ElRateModule.forRoot = function () {
        return { ngModule: ElRateModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElRateModule.ctorParameters = function () { return []; };
ElRateModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElRateModule });
ElRateModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElRateModule_Factory(t) { return new (t || ElRateModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElRateModule, { declarations: function () { return [ElRate]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElRate]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRateModule, [{
        type: NgModule,
        args: [{
                declarations: [ElRate],
                exports: [ElRate],
                imports: [CommonModule],
                entryComponents: [ElRate]
            }]
    }], function () { return []; }, null); })();
    return ElRateModule;
}());
export { ElRateModule };
function ElRateModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRateModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRateModule.ctorParameters;
}
