// element-angular/__ivy_ngcc__/release/form/form.js
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component } from '@angular/core';
import { ElFormProps } from './form.props';
var ElForm = /** @class */ (function (_super) {
    __extends(ElForm, _super);
    function ElForm() {
        return _super.call(this) || this;
    }
    /**
     * @nocollapse
     */
    ElForm.ctorParameters = function () { return []; };
ElForm.ɵfac = function ElForm_Factory(t) { return new (t || ElForm)(); };
ElForm.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElForm, selectors: [["el-form"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 2, vars: 2, template: function ElForm_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "form");
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-form " + (ctx.labelPosition ? "el-form--label-" + ctx.labelPosition : "") + (ctx.inline ? " el-form--inline" : ""));
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElForm, [{
        type: Component,
        args: [{
                selector: 'el-form',
                template: "\n    <form [class]=\"'el-form ' + (labelPosition ? 'el-form--label-' + labelPosition : '')\n      + (inline ? ' el-form--inline' : '')\">\n     <ng-content></ng-content>\n    </form>\n  "
            }]
    }], function () { return []; }, null); })();
    return ElForm;
}(ElFormProps));
export { ElForm };
function ElForm_tsickle_Closure_declarations() {
    /** @type {?} */
    ElForm.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElForm.ctorParameters;
}
