// element-angular/__ivy_ngcc__/release/loading/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElSharedModule } from '../shared/module';
import { ElLoadingDirective } from './loading.directive';
import * as ɵngcc0 from '@angular/core';
var ElLoadingModule = /** @class */ (function () {
    function ElLoadingModule() {
    }
    /**
     * @return {?}
     */
    ElLoadingModule.forRoot = function () {
        return { ngModule: ElLoadingModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElLoadingModule.ctorParameters = function () { return []; };
ElLoadingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElLoadingModule });
ElLoadingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElLoadingModule_Factory(t) { return new (t || ElLoadingModule)(); }, imports: [[CommonModule, ElSharedModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElLoadingModule, { declarations: function () { return [ElLoadingDirective]; }, imports: function () { return [CommonModule, ElSharedModule]; }, exports: function () { return [ElLoadingDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElLoadingModule, [{
        type: NgModule,
        args: [{
                declarations: [ElLoadingDirective],
                exports: [ElLoadingDirective],
                imports: [CommonModule, ElSharedModule],
                entryComponents: []
            }]
    }], function () { return []; }, null); })();
    return ElLoadingModule;
}());
export { ElLoadingModule };
function ElLoadingModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElLoadingModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElLoadingModule.ctorParameters;
}
