// element-angular/__ivy_ngcc__/release/pagination/children/button.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output, } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function ElPaginationButton_button_0_Template(rf, ctx) { if (rf & 1) {
    var _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 2);
    ɵngcc0.ɵɵlistener("click", function ElPaginationButton_button_0_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r3); var ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.clickHandle(-1); });
    ɵngcc0.ɵɵelement(1, "i", 3);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("disabled", ctx_r0.disabled);
} }
function ElPaginationButton_button_1_Template(rf, ctx) { if (rf & 1) {
    var _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 4);
    ɵngcc0.ɵɵlistener("click", function ElPaginationButton_button_1_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r5); var ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.clickHandle(1); });
    ɵngcc0.ɵɵelement(1, "i", 5);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("disabled", ctx_r1.disabled);
} }
var ElPaginationButton = /** @class */ (function () {
    function ElPaginationButton() {
        this.disabled = false;
        this.next = new EventEmitter();
    }
    /**
     * @param {?} step
     * @return {?}
     */
    ElPaginationButton.prototype.clickHandle = function (step) {
        if (this.disabled)
            return;
        this.next.emit(step);
    };
    /**
     * @nocollapse
     */
    ElPaginationButton.ctorParameters = function () { return []; };
    ElPaginationButton.propDecorators = {
        'dir': [{ type: Input },],
        'disabled': [{ type: Input },],
        'next': [{ type: Output },],
    };
ElPaginationButton.ɵfac = function ElPaginationButton_Factory(t) { return new (t || ElPaginationButton)(); };
ElPaginationButton.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElPaginationButton, selectors: [["el-pagination-btn"]], inputs: { disabled: "disabled", dir: "dir" }, outputs: { next: "next" }, decls: 2, vars: 2, consts: [["type", "button", "class", "btn-prev", 3, "disabled", "click", 4, "ngIf"], ["type", "button", "class", "btn-next", 3, "disabled", "click", 4, "ngIf"], ["type", "button", 1, "btn-prev", 3, "click"], [1, "el-icon", "el-icon-arrow-left"], ["type", "button", 1, "btn-next", 3, "click"], [1, "el-icon", "el-icon-arrow-right"]], template: function ElPaginationButton_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, ElPaginationButton_button_0_Template, 2, 2, "button", 0);
        ɵngcc0.ɵɵtemplate(1, ElPaginationButton_button_1_Template, 2, 2, "button", 1);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.dir === "left");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.dir === "right");
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElPaginationButton, [{
        type: Component,
        args: [{
                selector: 'el-pagination-btn',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: "\n    <button type=\"button\" class=\"btn-prev\"\n      *ngIf=\"dir === 'left'\"\n      [class.disabled]=\"disabled\"\n      (click)=\"clickHandle(-1)\">\n      <i class=\"el-icon el-icon-arrow-left\"></i>\n    </button>\n    <button type=\"button\" class=\"btn-next\"\n      *ngIf=\"dir === 'right'\"\n      [class.disabled]=\"disabled\"\n      (click)=\"clickHandle(1)\">\n    <i class=\"el-icon el-icon-arrow-right\"></i>\n    </button>\n  "
            }]
    }], function () { return []; }, { disabled: [{
            type: Input
        }], next: [{
            type: Output
        }], dir: [{
            type: Input
        }] }); })();
    return ElPaginationButton;
}());
export { ElPaginationButton };
function ElPaginationButton_tsickle_Closure_declarations() {
    /** @type {?} */
    ElPaginationButton.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElPaginationButton.ctorParameters;
    /** @type {?} */
    ElPaginationButton.propDecorators;
    /** @type {?} */
    ElPaginationButton.prototype.dir;
    /** @type {?} */
    ElPaginationButton.prototype.disabled;
    /** @type {?} */
    ElPaginationButton.prototype.next;
}
