// element-angular/__ivy_ngcc__/release/pagination/children/jump.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/forms';
var ElPaginationJump = /** @class */ (function () {
    function ElPaginationJump() {
        this.next = new EventEmitter();
    }
    /**
     * @param {?} page
     * @param {?} max
     * @return {?}
     */
    ElPaginationJump.nextPageNumber = function (page, max) {
        if (page <= 1)
            return 1;
        if (page >= max)
            return max;
        return page;
    };
    /**
     * @param {?} nextValue
     * @return {?}
     */
    ElPaginationJump.prototype.changeHandle = function (nextValue) {
        if (Number.isNaN(+nextValue))
            return;
        var /** @type {?} */ next = ElPaginationJump.nextPageNumber(+nextValue, this.max);
        var /** @type {?} */ pre = this.model;
        this.model = Math.round(next);
        this.next.emit(this.model - pre);
    };
    /**
     * @nocollapse
     */
    ElPaginationJump.ctorParameters = function () { return []; };
    ElPaginationJump.propDecorators = {
        'model': [{ type: Input },],
        'max': [{ type: Input },],
        'next': [{ type: Output },],
    };
ElPaginationJump.ɵfac = function ElPaginationJump_Factory(t) { return new (t || ElPaginationJump)(); };
ElPaginationJump.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElPaginationJump, selectors: [["el-pagination-jump"]], inputs: { model: "model", max: "max" }, outputs: { next: "next" }, decls: 6, vars: 3, consts: [[1, "el-pagination__jump"], [1, "el-input", "el-pagination__editor", "is-in-pagination"], ["type", "number", "autocomplete", "off", "rows", "2", 1, "el-input__inner", 3, "min", "max", "ngModel", "keyup.enter", "blur"], ["input", ""]], template: function ElPaginationJump_Template(rf, ctx) { if (rf & 1) {
        var _r1 = ɵngcc0.ɵɵgetCurrentView();
        ɵngcc0.ɵɵelementStart(0, "span", 0);
        ɵngcc0.ɵɵtext(1, " \u524D\u5F80 ");
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵelementStart(3, "input", 2, 3);
        ɵngcc0.ɵɵlistener("keyup.enter", function ElPaginationJump_Template_input_keyup_enter_3_listener() { ɵngcc0.ɵɵrestoreView(_r1); var _r0 = ɵngcc0.ɵɵreference(4); return ctx.changeHandle(_r0.value); })("blur", function ElPaginationJump_Template_input_blur_3_listener() { ɵngcc0.ɵɵrestoreView(_r1); var _r0 = ɵngcc0.ɵɵreference(4); return ctx.changeHandle(_r0.value); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtext(5, " \u9875 ");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("min", 1)("max", ctx.max)("ngModel", ctx.model);
    } }, directives: [ɵngcc1.NumberValueAccessor, ɵngcc1.DefaultValueAccessor, ɵngcc1.NgControlStatus, ɵngcc1.NgModel], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElPaginationJump, [{
        type: Component,
        args: [{
                selector: 'el-pagination-jump',
                template: "\n    <span class=\"el-pagination__jump\">\n      \u524D\u5F80\n    <div class=\"el-input el-pagination__editor is-in-pagination\">\n      <input class=\"el-input__inner\" type=\"number\" #input\n        autocomplete=\"off\" rows=\"2\"\n        [min]=\"1\" [max]=\"max\" [ngModel]=\"model\"\n        (keyup.enter)=\"changeHandle(input.value)\" (blur)=\"changeHandle(input.value)\"/>\n    </div>\n      \u9875\n    </span>\n  "
            }]
    }], function () { return []; }, { next: [{
            type: Output
        }], model: [{
            type: Input
        }], max: [{
            type: Input
        }] }); })();
    return ElPaginationJump;
}());
export { ElPaginationJump };
function ElPaginationJump_tsickle_Closure_declarations() {
    /** @type {?} */
    ElPaginationJump.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElPaginationJump.ctorParameters;
    /** @type {?} */
    ElPaginationJump.propDecorators;
    /** @type {?} */
    ElPaginationJump.prototype.model;
    /** @type {?} */
    ElPaginationJump.prototype.max;
    /** @type {?} */
    ElPaginationJump.prototype.next;
}
