// element-angular/__ivy_ngcc__/release/pagination/children/pager.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function ElPaginationPager_li_1_Template(rf, ctx) { if (rf & 1) {
    var _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 4);
    ɵngcc0.ɵɵlistener("click", function ElPaginationPager_li_1_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.clickHandle(1); });
    ɵngcc0.ɵɵtext(1, "1");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("active", ctx_r0.current === 1);
} }
function ElPaginationPager_li_2_Template(rf, ctx) { if (rf & 1) {
    var _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function ElPaginationPager_li_2_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r8); var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.jumpHandle(-5); })("mouseenter", function ElPaginationPager_li_2_Template_li_mouseenter_0_listener() { ɵngcc0.ɵɵrestoreView(_r8); var ctx_r9 = ɵngcc0.ɵɵnextContext(); return ctx_r9.quickprevIconClass = "el-icon-d-arrow-left"; })("mouseleave", function ElPaginationPager_li_2_Template_li_mouseleave_0_listener() { ɵngcc0.ɵɵrestoreView(_r8); var ctx_r10 = ɵngcc0.ɵɵnextContext(); return ctx_r10.quickprevIconClass = "el-icon-more"; });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap("el-icon more btn-quickprev " + ctx_r1.quickprevIconClass);
} }
function ElPaginationPager_li_3_Template(rf, ctx) { if (rf & 1) {
    var _r13 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 4);
    ɵngcc0.ɵɵlistener("click", function ElPaginationPager_li_3_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r13); var pager_r11 = ctx.$implicit; var ctx_r12 = ɵngcc0.ɵɵnextContext(); return ctx_r12.clickHandle(pager_r11); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var pager_r11 = ctx.$implicit;
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("active", ctx_r2.current === pager_r11);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(pager_r11);
} }
function ElPaginationPager_li_4_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function ElPaginationPager_li_4_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.jumpHandle(5); })("mouseenter", function ElPaginationPager_li_4_Template_li_mouseenter_0_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16.quicknextIconClass = "el-icon-d-arrow-right"; })("mouseleave", function ElPaginationPager_li_4_Template_li_mouseleave_0_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r17 = ɵngcc0.ɵɵnextContext(); return ctx_r17.quicknextIconClass = "el-icon-more"; });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap("el-icon more btn-quicknext " + ctx_r3.quicknextIconClass);
} }
function ElPaginationPager_li_5_Template(rf, ctx) { if (rf & 1) {
    var _r19 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 4);
    ɵngcc0.ɵɵlistener("click", function ElPaginationPager_li_5_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r19); var ctx_r18 = ɵngcc0.ɵɵnextContext(); return ctx_r18.clickHandle(ctx_r18.count); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("active", ctx_r4.current === ctx_r4.count);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r4.count);
} }
var ElPaginationPager = /** @class */ (function () {
    function ElPaginationPager() {
        this.current = 1;
        this.next = new EventEmitter();
        this.showPrevMore = false;
        this.showNextMore = false;
        this.quicknextIconClass = 'el-icon-more';
        this.quickprevIconClass = 'el-icon-more';
    }
    /**
     * @param {?} minValue
     * @return {?}
     */
    ElPaginationPager.pagerGenerator = function (minValue) {
        var /** @type {?} */ target = new Array(5).fill('').map(function (v, i) { return i + minValue; });
        return target;
    };
    /**
     * Always show 5 numbers, excluding head and foot
     * like: 1 ... < 5, 6, 7, 8, 9 > ... 100
     *
     * jump page button is [showPrevMore] and [showNextMore]
     *
     * @param {?} current
     * @param {?} count
     * @return {?} number[], like: [2, 3, 4, 5, 6]
     *
     */
    ElPaginationPager.prototype.makePagers = function (current, count) {
        var /** @type {?} */ pagerCount = 7;
        if (count < pagerCount) {
            this.setMoreBtn(false, false);
            var /** @type {?} */ target = ElPaginationPager.pagerGenerator(2);
            target.length = count - 2 >= 0 ? count - 2 : 0;
            return target;
        }
        var /** @type {?} */ max = current + 2;
        var /** @type {?} */ min = current - 2;
        if (max >= count) {
            this.setMoreBtn(true, false);
            return ElPaginationPager.pagerGenerator(count - 5);
        }
        if (min <= 2) {
            this.setMoreBtn(false, true);
            return ElPaginationPager.pagerGenerator(2);
        }
        this.setMoreBtn(true, true);
        return ElPaginationPager.pagerGenerator(min);
    };
    /**
     * @param {?} prev
     * @param {?} next
     * @return {?}
     */
    ElPaginationPager.prototype.setMoreBtn = function (prev, next) {
        this.showPrevMore = prev;
        this.showNextMore = next;
    };
    /**
     * @param {?} to
     * @return {?}
     */
    ElPaginationPager.prototype.clickHandle = function (to) {
        var /** @type {?} */ step = to - this.current;
        this.next.emit(step);
    };
    /**
     * @param {?} step
     * @return {?}
     */
    ElPaginationPager.prototype.jumpHandle = function (step) {
        this.next.emit(step);
    };
    /**
     * @return {?}
     */
    ElPaginationPager.prototype.ngOnInit = function () {
        this.pagers = this.makePagers(this.current, this.count);
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElPaginationPager.prototype.ngOnChanges = function (changes) {
        if (!changes)
            return;
        this.pagers = this.makePagers(this.current, this.count);
    };
    /**
     * @nocollapse
     */
    ElPaginationPager.ctorParameters = function () { return []; };
    ElPaginationPager.propDecorators = {
        'current': [{ type: Input },],
        'count': [{ type: Input },],
        'next': [{ type: Output },],
    };
ElPaginationPager.ɵfac = function ElPaginationPager_Factory(t) { return new (t || ElPaginationPager)(); };
ElPaginationPager.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElPaginationPager, selectors: [["el-pagination-pager"]], inputs: { current: "current", count: "count" }, outputs: { next: "next" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 6, vars: 5, consts: [[1, "el-pager"], ["class", "number", 3, "active", "click", 4, "ngIf"], [3, "class", "click", "mouseenter", "mouseleave", 4, "ngIf"], ["class", "number", 3, "active", "click", 4, "ngFor", "ngForOf"], [1, "number", 3, "click"], [3, "click", "mouseenter", "mouseleave"]], template: function ElPaginationPager_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "ul", 0);
        ɵngcc0.ɵɵtemplate(1, ElPaginationPager_li_1_Template, 2, 2, "li", 1);
        ɵngcc0.ɵɵtemplate(2, ElPaginationPager_li_2_Template, 1, 2, "li", 2);
        ɵngcc0.ɵɵtemplate(3, ElPaginationPager_li_3_Template, 2, 3, "li", 3);
        ɵngcc0.ɵɵtemplate(4, ElPaginationPager_li_4_Template, 1, 2, "li", 2);
        ɵngcc0.ɵɵtemplate(5, ElPaginationPager_li_5_Template, 2, 3, "li", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.count > 0);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showPrevMore);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.pagers);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showNextMore);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.count > 1);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgForOf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElPaginationPager, [{
        type: Component,
        args: [{
                selector: 'el-pagination-pager',
                template: "\n    <ul class=\"el-pager\">\n      <li class=\"number\" [class.active]=\"current === 1\"\n        (click)=\"clickHandle(1)\"\n        *ngIf=\"count > 0\">1</li>\n      <li *ngIf=\"showPrevMore\"\n        [class]=\"'el-icon more btn-quickprev ' + quickprevIconClass\"\n        (click)=\"jumpHandle(-5)\"\n        (mouseenter)=\"quickprevIconClass = 'el-icon-d-arrow-left'\"\n        (mouseleave)=\"quickprevIconClass = 'el-icon-more'\"></li>\n      <li class=\"number\"\n        *ngFor=\"let pager of pagers\"\n        (click)=\"clickHandle(pager)\"\n        [class.active]=\"current === pager\">{{ pager }}</li>\n      <li [class]=\"'el-icon more btn-quicknext ' + quicknextIconClass\"\n        *ngIf=\"showNextMore\"\n        (click)=\"jumpHandle(5)\"\n        (mouseenter)=\"quicknextIconClass = 'el-icon-d-arrow-right'\"\n        (mouseleave)=\"quicknextIconClass = 'el-icon-more'\"></li>\n      <li class=\"number\"\n        [class.active]=\"current === count\"\n        (click)=\"clickHandle(count)\"\n        *ngIf=\"count > 1\">{{ count }}</li>\n    </ul>\n  "
            }]
    }], function () { return []; }, { current: [{
            type: Input
        }], next: [{
            type: Output
        }], count: [{
            type: Input
        }] }); })();
    return ElPaginationPager;
}());
export { ElPaginationPager };
function ElPaginationPager_tsickle_Closure_declarations() {
    /** @type {?} */
    ElPaginationPager.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElPaginationPager.ctorParameters;
    /** @type {?} */
    ElPaginationPager.propDecorators;
    /** @type {?} */
    ElPaginationPager.prototype.current;
    /** @type {?} */
    ElPaginationPager.prototype.count;
    /** @type {?} */
    ElPaginationPager.prototype.next;
    /** @type {?} */
    ElPaginationPager.prototype.pagers;
    /** @type {?} */
    ElPaginationPager.prototype.showPrevMore;
    /** @type {?} */
    ElPaginationPager.prototype.showNextMore;
    /** @type {?} */
    ElPaginationPager.prototype.quicknextIconClass;
    /** @type {?} */
    ElPaginationPager.prototype.quickprevIconClass;
}
