// element-angular/__ivy_ngcc__/release/rate/rate.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';
import * as ɵngcc2 from '@angular/common';

var _c0 = ["rateIcon"];
var _c1 = function (a0) { return { cursor: a0 }; };
function ElRate_span_1_Template(rf, ctx) { if (rf & 1) {
    var _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span", 3);
    ɵngcc0.ɵɵlistener("mousemove", function ElRate_span_1_Template_span_mousemove_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r6); var i_r3 = ctx.index; var ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.hoverToggle($event, i_r3); })("mouseleave", function ElRate_span_1_Template_span_mouseleave_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r6); var i_r3 = ctx.index; var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.hoverToggle($event, i_r3, true); })("click", function ElRate_span_1_Template_span_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r6); var i_r3 = ctx.index; var ctx_r8 = ɵngcc0.ɵɵnextContext(); return ctx_r8.selectValue(i_r3); });
    ɵngcc0.ɵɵelement(1, "i", 4, 5);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var i_r3 = ctx.index;
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(7, _c1, ctx_r0.elDisabled ? "auto" : "pointer"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵstyleMap(ctx_r0.makeIconStyles(i_r3));
    ɵngcc0.ɵɵclassMap(ctx_r0.makeIconClasses(i_r3));
    ɵngcc0.ɵɵclassProp("hover", i_r3);
} }
var _c2 = function (a0) { return { color: a0 }; };
function ElRate_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(2, _c2, ctx_r1.textColor));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r1.texts[ctx_r1.model], " ");
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, ViewChild, Renderer2, forwardRef } from '@angular/core';
import { ElRateProps } from './rate.props';
import { DomSanitizer } from '@angular/platform-browser';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var ElRate = /** @class */ (function (_super) {
    __extends(ElRate, _super);
    /**
     * @param {?} sanitizer
     * @param {?} renderer
     */
    function ElRate(sanitizer, renderer) {
        var _this = _super.call(this) || this;
        _this.sanitizer = sanitizer;
        _this.renderer = renderer;
        _this.scores = [];
        _this.controlChange = function () { };
        _this.controlTouch = function () { };
        return _this;
    }
    /**
     * @param {?} __0
     * @param {?=} index
     * @param {?=} reset
     * @return {?}
     */
    ElRate.prototype.hoverToggle = function (_a, index, reset) {
        var srcElement = _a.srcElement;
        if (reset === void 0) { reset = false; }
        if (this.elDisabled)
            return;
        var /** @type {?} */ iconElement = srcElement.tagName === 'I' ? srcElement : srcElement.children[0];
        if (reset) {
            this.model = this.backupModel;
            this.renderer.removeClass(iconElement, 'hover');
        }
        else {
            this.model = index;
            this.renderer.addClass(iconElement, 'hover');
        }
    };
    /**
     * @param {?} index
     * @return {?}
     */
    ElRate.prototype.selectValue = function (index) {
        if (this.elDisabled)
            return;
        this.model = this.backupModel = index;
        this.modelChange.emit(index);
        this.controlChange;
    };
    /**
     * @param {?} index
     * @return {?}
     */
    ElRate.prototype.makeIconClasses = function (index) {
        var /** @type {?} */ voidClass = this.elDisabled ? this.rateMap.elDisabled.class : this.rateMap.void.class;
        var /** @type {?} */ activeItem = this.findCurrentType(this.model, this.rateMap);
        var /** @type {?} */ classes = index <= this.model ? activeItem.class : voidClass;
        return 'el-rate__icon ' + classes;
    };
    /**
     * @param {?} index
     * @return {?}
     */
    ElRate.prototype.makeIconStyles = function (index) {
        var /** @type {?} */ voidColor = this.elDisabled ? this.rateMap.elDisabled.color : this.rateMap.void.color;
        var /** @type {?} */ activeItem = this.findCurrentType(this.model, this.rateMap);
        var /** @type {?} */ styles = "color: " + (index <= this.model ? activeItem.color : voidColor) + ";";
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @param {?} index
     * @param {?} rateMap
     * @return {?}
     */
    ElRate.prototype.findCurrentType = function (index, rateMap) {
        if (index <= this.lowThreshold)
            return rateMap.low;
        if (index >= this.highThreshold)
            return rateMap.high;
        return rateMap.medium;
    };
    /**
     * @return {?}
     */
    ElRate.prototype.ngOnInit = function () {
        this.scores = new Array(this.max).fill('');
        this.backupModel = this.model;
        this.rateMap = {
            low: { color: this.colors[0], class: this.iconClasses[0] },
            medium: { color: this.colors[1], class: this.iconClasses[1] },
            high: { color: this.colors[2], class: this.iconClasses[2] },
            void: { color: this.voidColor, class: this.voidIconClass },
            elDisabled: { color: this.disabledVoidColor, class: this.disabledVoidIconClass },
        };
    };
    /**
     * @param {?} value
     * @return {?}
     */
    ElRate.prototype.writeValue = function (value) {
        this.model = value;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElRate.prototype.registerOnChange = function (fn) {
        this.controlChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElRate.prototype.registerOnTouched = function (fn) {
        this.controlTouch = fn;
    };
    /**
     * @nocollapse
     */
    ElRate.ctorParameters = function () { return [
        { type: DomSanitizer, },
        { type: Renderer2, },
    ]; };
    ElRate.propDecorators = {
        'rateIcon': [{ type: ViewChild, args: ['rateIcon',] },],
    };
ElRate.ɵfac = function ElRate_Factory(t) { return new (t || ElRate)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2)); };
ElRate.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElRate, selectors: [["el-rate"]], viewQuery: function ElRate_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.rateIcon = _t.first);
    } }, features: [ɵngcc0.ɵɵProvidersFeature([{
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return ElRate; }),
                multi: true
            }]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 2, consts: [["role", "slider", 1, "el-rate"], ["class", "el-rate__item", 3, "ngStyle", "mousemove", "mouseleave", "click", 4, "ngFor", "ngForOf"], ["class", "el-rate__text", 3, "ngStyle", 4, "ngIf"], [1, "el-rate__item", 3, "ngStyle", "mousemove", "mouseleave", "click"], [1, "el-rate__icon"], ["rateIcon", ""], [1, "el-rate__text", 3, "ngStyle"]], template: function ElRate_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, ElRate_span_1_Template, 3, 9, "span", 1);
        ɵngcc0.ɵɵtemplate(2, ElRate_span_2_Template, 2, 4, "span", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.scores);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showText);
    } }, directives: [ɵngcc2.NgForOf, ɵngcc2.NgIf, ɵngcc2.NgStyle], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElRate, [{
        type: Component,
        args: [{
                selector: 'el-rate',
                providers: [{
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(function () { return ElRate; }),
                        multi: true
                    }],
                template: "\n    <div class=\"el-rate\" role=\"slider\">\n      <span *ngFor=\"let s of scores; let i = index\" class=\"el-rate__item\"\n        [ngStyle]=\"{cursor: elDisabled ? 'auto' : 'pointer'}\"\n        (mousemove)=\"hoverToggle($event, i)\"\n        (mouseleave)=\"hoverToggle($event, i, true)\"\n        (click)=\"selectValue(i)\">\n        <i class=\"el-rate__icon\" [style]=\"makeIconStyles(i)\"\n          [class]=\"makeIconClasses(i)\"\n          [class.hover]=\"i\"\n          #rateIcon></i>\n      </span>\n      <span *ngIf=\"showText\" class=\"el-rate__text\" [ngStyle]=\"{ color: textColor }\">\n        {{ texts[model] }}\n      </span>\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.DomSanitizer }, { type: ɵngcc0.Renderer2 }]; }, { rateIcon: [{
            type: ViewChild,
            args: ['rateIcon']
        }] }); })();
    return ElRate;
}(ElRateProps));
export { ElRate };
function ElRate_tsickle_Closure_declarations() {
    /** @type {?} */
    ElRate.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElRate.ctorParameters;
    /** @type {?} */
    ElRate.propDecorators;
    /** @type {?} */
    ElRate.prototype.rateIcon;
    /** @type {?} */
    ElRate.prototype.scores;
    /** @type {?} */
    ElRate.prototype.rateMap;
    /** @type {?} */
    ElRate.prototype.backupModel;
    /** @type {?} */
    ElRate.prototype.controlChange;
    /** @type {?} */
    ElRate.prototype.controlTouch;
    /** @type {?} */
    ElRate.prototype.sanitizer;
    /** @type {?} */
    ElRate.prototype.renderer;
}
