// element-angular/__ivy_ngcc__/release/col/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElColDirective } from './col.directive';
import * as ɵngcc0 from '@angular/core';
var ElColModule = /** @class */ (function () {
    function ElColModule() {
    }
    /**
     * @return {?}
     */
    ElColModule.forRoot = function () {
        return { ngModule: ElColModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElColModule.ctorParameters = function () { return []; };
ElColModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElColModule });
ElColModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElColModule_Factory(t) { return new (t || ElColModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElColModule, { declarations: function () { return [ElColDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElColDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElColModule, [{
        type: NgModule,
        args: [{
                declarations: [ElColDirective],
                exports: [ElColDirective],
                imports: [CommonModule],
                entryComponents: []
            }]
    }], function () { return []; }, null); })();
    return ElColModule;
}());
export { ElColModule };
function ElColModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElColModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElColModule.ctorParameters;
}
