// element-angular/__ivy_ngcc__/release/tree/tree.js
import * as ɵngcc0 from '@angular/core';

function ElTree_el_tree_item_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "el-tree-item", 3);
} if (rf & 2) {
    var item_r2 = ctx.$implicit;
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("model", item_r2)("indent", ctx_r0.indent);
} }
function ElTree_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵelementStart(1, "span", 5);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.emptyText);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, Input } from '@angular/core';
import { ModelStandard } from './utils';
import { ElTreeProps } from './tree-props';
var ElTree = /** @class */ (function (_super) {
    __extends(ElTree, _super);
    function ElTree() {
        var _this = _super.call(this) || this;
        _this.userSafeHooks = function () { return ({
            findAllChecked: _this.findAllChecked.bind(_this),
            updateItemChecked: _this.updateChecked.bind(_this),
            updateItemExpanded: _this.updateExpanded.bind(_this),
            removeAllChecked: _this.removeAllChecked.bind(_this),
        }); };
        return _this;
    }
    Object.defineProperty(ElTree.prototype, "model", {
        /**
         * @param {?} val
         * @return {?}
         */
        set: function (val) {
            var /** @type {?} */ standardTool = new ModelStandard({
                initDepth: 0,
                defaultExpandAll: this.defaultExpandAll,
                defaultExpandedKeys: this.defaultExpandedKeys,
                defaultCheckedKeys: this.defaultCheckedKeys,
            });
            this.identModel = standardTool.filterModel(val);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    ElTree.prototype.findAllChecked = function () {
        if (!this.showCheckbox)
            return [];
        return ModelStandard.FindAllChecked(this.identModel);
    };
    /**
     * @return {?}
     */
    ElTree.prototype.removeAllChecked = function () {
        ModelStandard.LoopRemoveChecked(this.identModel);
    };
    /**
     * @param {?} id
     * @return {?}
     */
    ElTree.prototype.updateExpanded = function (id) {
        this.identModel = ModelStandard.DeepUpdateExpanded(id, this.identModel, this.accordion);
    };
    /**
     * @param {?} id
     * @return {?}
     */
    ElTree.prototype.updateChecked = function (id) {
        if (!this.showCheckbox)
            return;
        this.identModel = ModelStandard.DeepUpdateChecked(id, this.identModel);
    };
    /**
     * @param {?} next
     * @return {?}
     */
    ElTree.prototype.emitter = function (next) {
        this.modelChange.emit(next);
    };
    /**
     * @nocollapse
     */
    ElTree.ctorParameters = function () { return []; };
    ElTree.propDecorators = {
        'model': [{ type: Input },],
    };
ElTree.ɵfac = function ElTree_Factory(t) { return new (t || ElTree)(); };
ElTree.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElTree, selectors: [["el-tree"]], inputs: { model: "model" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 2, consts: [["role", "tree", 1, "el-tree"], [3, "model", "indent", 4, "ngFor", "ngForOf"], ["class", "el-tree__empty-block", 4, "ngIf"], [3, "model", "indent"], [1, "el-tree__empty-block"], [1, "el-tree__empty-text"]], template: function ElTree_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, ElTree_el_tree_item_1_Template, 1, 2, "el-tree-item", 1);
        ɵngcc0.ɵɵtemplate(2, ElTree_div_2_Template, 3, 1, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.identModel);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.identModel && !ctx.identModel.length);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElTree, [{
        type: Component,
        args: [{
                selector: 'el-tree',
                template: "\n    <div class=\"el-tree\" role=\"tree\">\n      <el-tree-item *ngFor=\"let item of identModel\"\n        [model]=\"item\" [indent]=\"indent\">\n      </el-tree-item>\n      <div class=\"el-tree__empty-block\" *ngIf=\"!identModel && !identModel.length\">\n        <span class=\"el-tree__empty-text\">{{ emptyText }}</span>\n      </div>\n    </div>\n  "
            }]
    }], function () { return []; }, { model: [{
            type: Input
        }] }); })();
    return ElTree;
}(ElTreeProps));
export { ElTree };
function ElTree_tsickle_Closure_declarations() {
    /** @type {?} */
    ElTree.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElTree.ctorParameters;
    /** @type {?} */
    ElTree.propDecorators;
    /** @type {?} */
    ElTree.prototype.userSafeHooks;
}
