// element-angular/__ivy_ngcc__/release/date-picker/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ElInputsModule } from '../input/module';
import { ElDataPicker } from './picker';
import { ElDatePickerPanel } from './picker-panel';
import { ElDateTable } from './children/date-table';
import { ElYearTable } from './children/year-table';
import { EMonthTable } from './children/month-table';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../input/input';
var ElDateModule = /** @class */ (function () {
    function ElDateModule() {
    }
    /**
     * @return {?}
     */
    ElDateModule.forRoot = function () {
        return { ngModule: ElDateModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElDateModule.ctorParameters = function () { return []; };
ElDateModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElDateModule });
ElDateModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElDateModule_Factory(t) { return new (t || ElDateModule)(); }, imports: [[CommonModule, FormsModule, ElInputsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElDateModule, { declarations: function () { return [ElDataPicker, ElDatePickerPanel, ElDateTable, ElYearTable, EMonthTable]; }, imports: function () { return [CommonModule, FormsModule, ElInputsModule]; }, exports: function () { return [ElDataPicker, ElDatePickerPanel, ElDateTable, ElYearTable, EMonthTable]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElDateModule, [{
        type: NgModule,
        args: [{
                declarations: [ElDataPicker, ElDatePickerPanel, ElDateTable, ElYearTable, EMonthTable],
                exports: [ElDataPicker, ElDatePickerPanel, ElDateTable, ElYearTable, EMonthTable],
                imports: [CommonModule, FormsModule, ElInputsModule],
                entryComponents: [ElDataPicker]
            }]
    }], function () { return []; }, null); })();
ɵngcc0.ɵɵsetComponentScope(ElDataPicker, [ɵngcc1.ElInput, ElDatePickerPanel], []);
    return ElDateModule;
}());
export { ElDateModule };
function ElDateModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElDateModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElDateModule.ctorParameters;
}
