// element-angular/__ivy_ngcc__/release/collapse/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElCollapse } from './collapse';
import { ElCollapseItem } from './collapse-item';
import * as ɵngcc0 from '@angular/core';
var ElCollapseModule = /** @class */ (function () {
    function ElCollapseModule() {
    }
    /**
     * @return {?}
     */
    ElCollapseModule.forRoot = function () {
        return { ngModule: ElCollapseModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElCollapseModule.ctorParameters = function () { return []; };
ElCollapseModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElCollapseModule });
ElCollapseModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElCollapseModule_Factory(t) { return new (t || ElCollapseModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElCollapseModule, { declarations: function () { return [ElCollapse, ElCollapseItem]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElCollapse, ElCollapseItem]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCollapseModule, [{
        type: NgModule,
        args: [{
                declarations: [ElCollapse, ElCollapseItem],
                exports: [ElCollapse, ElCollapseItem],
                imports: [CommonModule],
                entryComponents: [ElCollapse]
            }]
    }], function () { return []; }, null); })();
    return ElCollapseModule;
}());
export { ElCollapseModule };
function ElCollapseModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCollapseModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCollapseModule.ctorParameters;
}
