// element-angular/__ivy_ngcc__/release/carousel/carousel.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function ElCarousel_button_3_Template(rf, ctx) { if (rf & 1) {
    var _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 6, 7);
    ɵngcc0.ɵɵlistener("mouseenter", function ElCarousel_button_3_Template_button_mouseenter_0_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.btnActionHandle(ctx_r5.model - 1, "hover"); })("click", function ElCarousel_button_3_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.btnActionHandle(ctx_r7.model - 1, "click"); });
    ɵngcc0.ɵɵelement(2, "i", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵproperty("@carouselBtnLeftAnimation", ctx_r1.arrow === "always" || _r0.hover);
} }
function ElCarousel_button_4_Template(rf, ctx) { if (rf & 1) {
    var _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 9, 10);
    ɵngcc0.ɵɵlistener("mouseenter", function ElCarousel_button_4_Template_button_mouseenter_0_listener() { ɵngcc0.ɵɵrestoreView(_r10); var ctx_r9 = ɵngcc0.ɵɵnextContext(); return ctx_r9.btnActionHandle(ctx_r9.model + 1, "hover"); })("click", function ElCarousel_button_4_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r10); var ctx_r11 = ɵngcc0.ɵɵnextContext(); return ctx_r11.btnActionHandle(ctx_r11.model + 1, "click"); });
    ɵngcc0.ɵɵelement(2, "i", 11);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵproperty("@carouselBtnRightAnimation", ctx_r2.arrow === "always" || _r0.hover);
} }
function ElCarousel_ul_6_li_1_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r13 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(item_r13);
} }
function ElCarousel_ul_6_li_1_Template(rf, ctx) { if (rf & 1) {
    var _r18 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 14);
    ɵngcc0.ɵɵlistener("mouseenter", function ElCarousel_ul_6_li_1_Template_li_mouseenter_0_listener() { ɵngcc0.ɵɵrestoreView(_r18); var i_r14 = ctx.index; var ctx_r17 = ɵngcc0.ɵɵnextContext(2); return ctx_r17.indicatorActionHandle(i_r14, "hover"); })("click", function ElCarousel_ul_6_li_1_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r18); var i_r14 = ctx.index; var ctx_r19 = ɵngcc0.ɵɵnextContext(2); return ctx_r19.indicatorActionHandle(i_r14, "click"); });
    ɵngcc0.ɵɵelementStart(1, "button", 15);
    ɵngcc0.ɵɵtemplate(2, ElCarousel_ul_6_li_1_span_2_Template, 2, 1, "span", 16);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var i_r14 = ctx.index;
    var ctx_r12 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassProp("is-active", i_r14 === ctx_r12.model);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r12.hasLabel);
} }
function ElCarousel_ul_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ul", 12);
    ɵngcc0.ɵɵtemplate(1, ElCarousel_ul_6_li_1_Template, 3, 3, "li", 13);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("el-carousel__indicators--labels", ctx_r3.hasLabel)("el-carousel__indicators--outside", ctx_r3.indicatorPosition === "outside" || ctx_r3.type === "card");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r3.items);
} }
var _c0 = function (a0) { return { height: a0 }; };
var _c1 = ["*"];
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, ContentChildren, forwardRef, } from '@angular/core';
import { ElCarouselItem } from './carousel-item';
import { ElCarouselProps } from './carousel-props';
import { carouselBtnLeftAnimation, carouselBtnRightAnimation } from './animations';
var ElCarousel = /** @class */ (function (_super) {
    __extends(ElCarousel, _super);
    function ElCarousel() {
        var _this = _super.call(this) || this;
        _this.subscriber = [];
        _this.items = [];
        _this.hasLabel = false;
        return _this;
    }
    /**
     * @param {?} nextValue
     * @param {?} eventType
     * @return {?}
     */
    ElCarousel.prototype.btnActionHandle = function (nextValue, eventType) {
        if (this.trigger !== eventType)
            return;
        this.autoplay && this.resetInterval();
        this.setActiveItem(nextValue);
    };
    /**
     * @param {?} nextValue
     * @param {?} eventType
     * @return {?}
     */
    ElCarousel.prototype.indicatorActionHandle = function (nextValue, eventType) {
        if (this.indicatorTrigger !== eventType)
            return;
        this.autoplay && this.resetInterval();
        this.setActiveItem(nextValue);
    };
    /**
     * @param {?} index
     * @return {?}
     */
    ElCarousel.prototype.setActiveItem = function (index) {
        if (!this.children)
            return;
        var /** @type {?} */ len = this.children.length;
        var /** @type {?} */ nextValue = index >= len ? 0 : index < 0 ? len - 1 : index;
        this.model = nextValue;
        this.subscriber.forEach(function (func) { return func(); });
        this.modelChange.emit(this.model);
    };
    /**
     * @return {?}
     */
    ElCarousel.prototype.resetInterval = function () {
        var _this = this;
        this.timer && clearInterval(this.timer);
        this.timer = setInterval(function () {
            _this.setActiveItem(_this.model + 1);
        }, this.interval);
    };
    /**
     * @return {?}
     */
    ElCarousel.prototype.ngAfterContentChecked = function () {
        var _this = this;
        var /** @type {?} */ timer = setTimeout(function () {
            _this.children.forEach(function (item, index) {
                item.index = index;
                item.updateActive();
                item.updateStyles();
            });
            // all labels are validated
            _this.hasLabel = !_this.items.some(function (item) { return !item; });
            // auto play
            _this.autoplay && _this.resetInterval();
            clearTimeout(timer);
        }, 0);
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElCarousel.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.model)
            return;
        this.setActiveItem(changes.model.currentValue);
    };
    /**
     * @return {?}
     */
    ElCarousel.prototype.ngOnDestroy = function () {
        this.timer && clearInterval(this.timer);
    };
    /**
     * @nocollapse
     */
    ElCarousel.ctorParameters = function () { return []; };
    ElCarousel.propDecorators = {
        'children': [{ type: ContentChildren, args: [forwardRef(function () { return ElCarouselItem; }),] },],
    };
ElCarousel.ɵfac = function ElCarousel_Factory(t) { return new (t || ElCarousel)(); };
ElCarousel.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElCarousel, selectors: [["el-carousel"]], contentQueries: function ElCarousel_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, ElCarouselItem, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.children = _t);
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c1, decls: 7, vars: 8, consts: [[1, "el-carousel", 3, "mouseenter", "mouseleave"], ["carousel", ""], [1, "el-carousel__container", 3, "ngStyle"], ["class", "el-carousel__arrow el-carousel__arrow--left", 3, "mouseenter", "click", 4, "ngIf"], ["class", "el-carousel__arrow el-carousel__arrow--right", 3, "mouseenter", "click", 4, "ngIf"], ["class", "el-carousel__indicators", 3, "el-carousel__indicators--labels", "el-carousel__indicators--outside", 4, "ngIf"], [1, "el-carousel__arrow", "el-carousel__arrow--left", 3, "mouseenter", "click"], ["leftBtn", ""], [1, "el-icon-arrow-left"], [1, "el-carousel__arrow", "el-carousel__arrow--right", 3, "mouseenter", "click"], ["rightBtn", ""], [1, "el-icon-arrow-right"], [1, "el-carousel__indicators"], ["class", "el-carousel__indicator", 3, "is-active", "mouseenter", "click", 4, "ngFor", "ngForOf"], [1, "el-carousel__indicator", 3, "mouseenter", "click"], [1, "el-carousel__button"], [4, "ngIf"]], template: function ElCarousel_Template(rf, ctx) { if (rf & 1) {
        var _r20 = ɵngcc0.ɵɵgetCurrentView();
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0, 1);
        ɵngcc0.ɵɵlistener("mouseenter", function ElCarousel_Template_div_mouseenter_0_listener() { ɵngcc0.ɵɵrestoreView(_r20); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.hover = true; })("mouseleave", function ElCarousel_Template_div_mouseleave_0_listener() { ɵngcc0.ɵɵrestoreView(_r20); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.hover = false; });
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵtemplate(3, ElCarousel_button_3_Template, 3, 1, "button", 3);
        ɵngcc0.ɵɵtemplate(4, ElCarousel_button_4_Template, 3, 1, "button", 4);
        ɵngcc0.ɵɵprojection(5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(6, ElCarousel_ul_6_Template, 2, 5, "ul", 5);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("el-carousel--card", ctx.type === "card");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(6, _c0, ctx.height));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.arrow !== "never");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.arrow !== "never");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.indicatorPosition !== "none");
    } }, directives: [ɵngcc1.NgStyle, ɵngcc1.NgIf, ɵngcc1.NgForOf], encapsulation: 2, data: { animation: [carouselBtnLeftAnimation, carouselBtnRightAnimation] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCarousel, [{
        type: Component,
        args: [{
                selector: 'el-carousel',
                animations: [carouselBtnLeftAnimation, carouselBtnRightAnimation],
                template: "\n    <div class=\"el-carousel\"\n      #carousel\n      [class.el-carousel--card]=\"type === 'card'\"\n      (mouseenter)=\"carousel.hover = true\"\n      (mouseleave)=\"carousel.hover = false\">\n      <div class=\"el-carousel__container\" [ngStyle]=\"{height: height}\">\n        <button class=\"el-carousel__arrow el-carousel__arrow--left\"\n          #leftBtn\n          *ngIf=\"arrow !== 'never'\"\n          [@carouselBtnLeftAnimation]=\"arrow === 'always' || carousel.hover\"\n          (mouseenter)=\"btnActionHandle(model - 1,'hover')\"\n          (click)=\"btnActionHandle(model - 1, 'click')\">\n          <i class=\"el-icon-arrow-left\"></i>\n        </button>\n        <button class=\"el-carousel__arrow el-carousel__arrow--right\"\n          #rightBtn\n          *ngIf=\"arrow !== 'never'\"\n          [@carouselBtnRightAnimation]=\"arrow === 'always' || carousel.hover\"\n          (mouseenter)=\"btnActionHandle(model + 1, 'hover')\"\n          (click)=\"btnActionHandle(model + 1, 'click')\">\n          <i class=\"el-icon-arrow-right\"></i>\n        </button>\n        <ng-content></ng-content>\n      </div>\n      <ul class=\"el-carousel__indicators\" *ngIf=\"indicatorPosition !== 'none'\"\n        [class.el-carousel__indicators--labels]=\"hasLabel\"\n        [class.el-carousel__indicators--outside]=\"indicatorPosition === 'outside' || type === 'card'\">\n        <li *ngFor=\"let item of items; let i = index\"\n          class=\"el-carousel__indicator\"\n          [class.is-active]=\"i === model\"\n          (mouseenter)=\"indicatorActionHandle(i, 'hover')\"\n          (click)=\"indicatorActionHandle(i, 'click')\">\n          <button class=\"el-carousel__button\">\n            <span *ngIf=\"hasLabel\">{{item}}</span>\n          </button>\n        </li>\n      </ul>\n    </div>\n  "
            }]
    }], function () { return []; }, { children: [{
            type: ContentChildren,
            args: [forwardRef(function () { return ElCarouselItem; })]
        }] }); })();
    return ElCarousel;
}(ElCarouselProps));
export { ElCarousel };
function ElCarousel_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCarousel.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCarousel.ctorParameters;
    /** @type {?} */
    ElCarousel.propDecorators;
    /** @type {?} */
    ElCarousel.prototype.children;
    /** @type {?} */
    ElCarousel.prototype.subscriber;
    /** @type {?} */
    ElCarousel.prototype.items;
    /** @type {?} */
    ElCarousel.prototype.hasLabel;
    /** @type {?} */
    ElCarousel.prototype.timer;
}
