// element-angular/__ivy_ngcc__/release/notification/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElNotificationContainer } from './notification';
import { ElNotificationService } from './notification.service';
import * as ɵngcc0 from '@angular/core';
var ElNotificationModule = /** @class */ (function () {
    function ElNotificationModule() {
    }
    /**
     * @return {?}
     */
    ElNotificationModule.forRoot = function () {
        return { ngModule: ElNotificationModule, providers: [
                ElNotificationService,
            ] };
    };
    /**
     * @nocollapse
     */
    ElNotificationModule.ctorParameters = function () { return []; };
ElNotificationModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElNotificationModule });
ElNotificationModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElNotificationModule_Factory(t) { return new (t || ElNotificationModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElNotificationModule, { declarations: function () { return [ElNotificationContainer]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ElNotificationContainer]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElNotificationModule, [{
        type: NgModule,
        args: [{
                declarations: [ElNotificationContainer],
                exports: [ElNotificationContainer],
                imports: [CommonModule],
                entryComponents: [ElNotificationContainer]
            }]
    }], function () { return []; }, null); })();
    return ElNotificationModule;
}());
export { ElNotificationModule };
function ElNotificationModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElNotificationModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElNotificationModule.ctorParameters;
}
