// element-angular/__ivy_ngcc__/release/slider/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElTooltipModule } from '../tooltip/module';
import { ElSliderButton } from './slider-button';
import { ElSlider } from './slider';
import * as ɵngcc0 from '@angular/core';
var ElSliderModule = /** @class */ (function () {
    function ElSliderModule() {
    }
    /**
     * @return {?}
     */
    ElSliderModule.forRoot = function () {
        return { ngModule: ElSliderModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElSliderModule.ctorParameters = function () { return []; };
ElSliderModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElSliderModule });
ElSliderModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElSliderModule_Factory(t) { return new (t || ElSliderModule)(); }, imports: [[CommonModule, ElTooltipModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElSliderModule, { declarations: function () { return [ElSlider, ElSliderButton]; }, imports: function () { return [CommonModule, ElTooltipModule]; }, exports: function () { return [ElSlider, ElSliderButton]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElSliderModule, [{
        type: NgModule,
        args: [{
                declarations: [ElSlider, ElSliderButton],
                exports: [ElSlider, ElSliderButton],
                imports: [CommonModule, ElTooltipModule],
                entryComponents: [ElSlider]
            }]
    }], function () { return []; }, null); })();
ɵngcc0.ɵɵsetComponentScope(ElSlider, [ElSliderButton], []);
    return ElSliderModule;
}());
export { ElSliderModule };
function ElSliderModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElSliderModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElSliderModule.ctorParameters;
}
