// element-angular/__ivy_ngcc__/release/slider/slider.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/platform-browser';

var _c0 = ["runway"];
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, forwardRef, ViewChild } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { ElSliderProps } from './slider.props';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var ElSlider = /** @class */ (function (_super) {
    __extends(ElSlider, _super);
    /**
     * @param {?} sanitizer
     */
    function ElSlider(sanitizer) {
        var _this = _super.call(this) || this;
        _this.sanitizer = sanitizer;
        _this.start = 0;
        _this.isDragging = false;
        _this.controlChange = function () { };
        _this.controlTouch = function () { };
        return _this;
    }
    /**
     * @return {?}
     */
    ElSlider.prototype.makeRunwayStyle = function () {
        var /** @type {?} */ styles = this.vertical ? "height: " + this.height + "px;" : '';
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @return {?}
     */
    ElSlider.prototype.makeBarStyle = function () {
        var /** @type {?} */ val = (this.model - this.min) * 100 / (this.max - this.min);
        var /** @type {?} */ styles = this.vertical ? "height: " + val + "%; bottom: " + this.start + "%;"
            : "width: " + val + "%; left: " + this.start + "%;";
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @return {?}
     */
    ElSlider.prototype.resetSize = function () {
        if (!this.runwayElement)
            return;
        this.size = this.runwayElement.nativeElement["client" + (this.vertical ? 'Height' : 'Width')];
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ElSlider.prototype.onSliderClick = function (event) {
        if (this.elDisabled || this.isDragging)
            return;
        this.resetSize();
        var /** @type {?} */ val = this.vertical ? event.clientY : event.clientX;
        var _a = this.runwayElement.nativeElement.getBoundingClientRect(), left = _a.left, bottom = _a.bottom;
        var /** @type {?} */ offset = Math.abs(val - (this.vertical ? bottom : left));
        // update value
        this.model = Math.round((offset / this.size) * (this.max - this.min)) + this.min;
        this.moveChange(this.model);
        this.makeBarStyle();
    };
    /**
     * @param {?} nextValue
     * @return {?}
     */
    ElSlider.prototype.moveChange = function (nextValue) {
        this.model = nextValue;
        this.modelChange.emit(nextValue);
        this.controlChange(nextValue);
    };
    /**
     * @return {?}
     */
    ElSlider.prototype.ngOnInit = function () {
        if (!this.model) {
            this.model = this.min;
        }
    };
    /**
     * @return {?}
     */
    ElSlider.prototype.ngAfterViewInit = function () {
        this.resetSize();
    };
    /**
     * @param {?} value
     * @return {?}
     */
    ElSlider.prototype.writeValue = function (value) {
        this.model = value;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElSlider.prototype.registerOnChange = function (fn) {
        this.controlChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElSlider.prototype.registerOnTouched = function (fn) {
        this.controlTouch = fn;
    };
    /**
     * @nocollapse
     */
    ElSlider.ctorParameters = function () { return [
        { type: DomSanitizer, },
    ]; };
    ElSlider.propDecorators = {
        'runwayElement': [{ type: ViewChild, args: ['runway',] },],
    };
ElSlider.ɵfac = function ElSlider_Factory(t) { return new (t || ElSlider)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DomSanitizer)); };
ElSlider.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElSlider, selectors: [["el-slider"]], viewQuery: function ElSlider_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.runwayElement = _t.first);
    } }, features: [ɵngcc0.ɵɵProvidersFeature([{
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return ElSlider; }),
                multi: true
            }]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 5, vars: 14, consts: [[1, "el-slider"], [1, "el-slider__runway", 3, "click"], ["runway", ""], [1, "el-slider__bar"], [3, "vertical", "model", "min", "max", "format-tooltip", "elDisabled", "modelChange"]], template: function ElSlider_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1, 2);
        ɵngcc0.ɵɵlistener("click", function ElSlider_Template_div_click_1_listener($event) { return ctx.onSliderClick($event); });
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelementStart(4, "el-slider-button", 4);
        ɵngcc0.ɵɵlistener("modelChange", function ElSlider_Template_el_slider_button_modelChange_4_listener($event) { return ctx.moveChange($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("is-vertical", ctx.vertical);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleMap(ctx.makeRunwayStyle());
        ɵngcc0.ɵɵclassProp("disabled", ctx.elDisabled);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleMap(ctx.makeBarStyle());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("vertical", ctx.vertical)("model", ctx.model)("min", ctx.min)("max", ctx.max)("format-tooltip", ctx.formatTooltip)("elDisabled", !ctx.showTooltip);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElSlider, [{
        type: Component,
        args: [{
                selector: 'el-slider',
                providers: [{
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(function () { return ElSlider; }),
                        multi: true
                    }],
                template: "\n    <div class=\"el-slider\" [class.is-vertical]=\"vertical\">\n      <div class=\"el-slider__runway\" [class.disabled]=\"elDisabled\"\n        [style]=\"makeRunwayStyle()\" (click)=\"onSliderClick($event)\" #runway>\n        <div class=\"el-slider__bar\" [style]=\"makeBarStyle()\"></div>\n        <el-slider-button [vertical]=\"vertical\"\n          [model]=\"model\"\n          (modelChange)=\"moveChange($event)\"\n          [min]=\"min\" [max]=\"max\"\n          [format-tooltip]=\"formatTooltip\"\n          [elDisabled]=\"!showTooltip\">\n        </el-slider-button>\n      </div>\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.DomSanitizer }]; }, { runwayElement: [{
            type: ViewChild,
            args: ['runway']
        }] }); })();
    return ElSlider;
}(ElSliderProps));
export { ElSlider };
function ElSlider_tsickle_Closure_declarations() {
    /** @type {?} */
    ElSlider.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElSlider.ctorParameters;
    /** @type {?} */
    ElSlider.propDecorators;
    /** @type {?} */
    ElSlider.prototype.runwayElement;
    /** @type {?} */
    ElSlider.prototype.size;
    /** @type {?} */
    ElSlider.prototype.start;
    /** @type {?} */
    ElSlider.prototype.isDragging;
    /** @type {?} */
    ElSlider.prototype.controlChange;
    /** @type {?} */
    ElSlider.prototype.controlTouch;
    /** @type {?} */
    ElSlider.prototype.sanitizer;
}
