// element-angular/__ivy_ngcc__/release/checkbox/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ElCheckbox } from './checkbox';
import { ElCheckboxButton } from './checkbox-button';
import { ElCheckboxGroup } from './checkbox-group';
import * as ɵngcc0 from '@angular/core';
var ElCheckboxsModule = /** @class */ (function () {
    function ElCheckboxsModule() {
    }
    /**
     * @return {?}
     */
    ElCheckboxsModule.forRoot = function () {
        return { ngModule: ElCheckboxsModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElCheckboxsModule.ctorParameters = function () { return []; };
ElCheckboxsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElCheckboxsModule });
ElCheckboxsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElCheckboxsModule_Factory(t) { return new (t || ElCheckboxsModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElCheckboxsModule, { declarations: function () { return [ElCheckbox, ElCheckboxButton, ElCheckboxGroup]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [ElCheckbox, ElCheckboxButton, ElCheckboxGroup]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCheckboxsModule, [{
        type: NgModule,
        args: [{
                declarations: [ElCheckbox, ElCheckboxButton, ElCheckboxGroup],
                exports: [ElCheckbox, ElCheckboxButton, ElCheckboxGroup],
                imports: [CommonModule, FormsModule],
                entryComponents: [ElCheckbox, ElCheckboxButton, ElCheckboxGroup]
            }]
    }], function () { return []; }, null); })();
    return ElCheckboxsModule;
}());
export { ElCheckboxsModule };
function ElCheckboxsModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCheckboxsModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCheckboxsModule.ctorParameters;
}
