// element-angular/__ivy_ngcc__/release/container/directives/footer.directive.js
import { Directive, Input, Self } from '@angular/core';
import { NgStyle } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
var ElFooterDirective = /** @class */ (function () {
    /**
     * @param {?} ngStyle
     */
    function ElFooterDirective(ngStyle) {
        this.ngStyle = ngStyle;
        this.height = '60px';
    }
    /**
     * @return {?}
     */
    ElFooterDirective.prototype.ngOnChanges = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElFooterDirective.prototype.ngOnInit = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElFooterDirective.prototype.colletClasses = function () {
        this.hostStyles = {
            height: this.height,
        };
        this.ngStyle.ngStyle = this.hostStyles;
        this.ngStyle.ngDoCheck();
    };
    /**
     * @nocollapse
     */
    ElFooterDirective.ctorParameters = function () { return [
        { type: NgStyle, decorators: [{ type: Self },] },
    ]; };
    ElFooterDirective.propDecorators = {
        'height': [{ type: Input },],
    };
ElFooterDirective.ɵfac = function ElFooterDirective_Factory(t) { return new (t || ElFooterDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NgStyle, 2)); };
ElFooterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElFooterDirective, selectors: [["", "el-footer", ""]], hostAttrs: [1, "el-footer"], inputs: { height: "height" }, features: [ɵngcc0.ɵɵProvidersFeature([NgStyle]), ɵngcc0.ɵɵNgOnChangesFeature] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElFooterDirective, [{
        type: Directive,
        args: [{
                selector: '[el-footer]',
                providers: [NgStyle],
                host: { class: 'el-footer' }
            }]
    }], function () { return [{ type: ɵngcc1.NgStyle, decorators: [{
                type: Self
            }] }]; }, { height: [{
            type: Input
        }] }); })();
    return ElFooterDirective;
}());
export { ElFooterDirective };
function ElFooterDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    ElFooterDirective.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElFooterDirective.ctorParameters;
    /** @type {?} */
    ElFooterDirective.propDecorators;
    /** @type {?} */
    ElFooterDirective.prototype.height;
    /** @type {?} */
    ElFooterDirective.prototype.hostStyles;
    /** @type {?} */
    ElFooterDirective.prototype.ngStyle;
}
