// element-angular/__ivy_ngcc__/release/container/directives/main.directive.js
import { Directive } from '@angular/core';
import { NgClass, NgStyle } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
var ElMainDirective = /** @class */ (function () {
    function ElMainDirective() {
    }
    /**
     * @nocollapse
     */
    ElMainDirective.ctorParameters = function () { return []; };
ElMainDirective.ɵfac = function ElMainDirective_Factory(t) { return new (t || ElMainDirective)(); };
ElMainDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElMainDirective, selectors: [["", "el-main", ""]], hostAttrs: [1, "el-main"], features: [ɵngcc0.ɵɵProvidersFeature([NgClass, NgStyle])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElMainDirective, [{
        type: Directive,
        args: [{
                selector: '[el-main]',
                providers: [NgClass, NgStyle],
                host: { class: 'el-main' }
            }]
    }], function () { return []; }, null); })();
    return ElMainDirective;
}());
export { ElMainDirective };
function ElMainDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    ElMainDirective.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElMainDirective.ctorParameters;
}
