// element-angular/__ivy_ngcc__/release/pagination/module.js
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { ElSelectModule } from '../select/module';
import { ElPaginationButton, ElPaginationPager, ElPaginationSize, ElPaginationJump } from './children/index';
import { ElPagination } from './pagination';
import * as ɵngcc0 from '@angular/core';
var ElPaginationModule = /** @class */ (function () {
    function ElPaginationModule() {
    }
    /**
     * @return {?}
     */
    ElPaginationModule.forRoot = function () {
        return { ngModule: ElPaginationModule, providers: [] };
    };
    /**
     * @nocollapse
     */
    ElPaginationModule.ctorParameters = function () { return []; };
ElPaginationModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElPaginationModule });
ElPaginationModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElPaginationModule_Factory(t) { return new (t || ElPaginationModule)(); }, imports: [[CommonModule, FormsModule, ElSelectModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElPaginationModule, { declarations: function () { return [ElPagination,
        ElPaginationButton,
        ElPaginationPager,
        ElPaginationSize,
        ElPaginationJump]; }, imports: function () { return [CommonModule, FormsModule, ElSelectModule]; }, exports: function () { return [ElPagination]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElPaginationModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    ElPagination,
                    ElPaginationButton,
                    ElPaginationPager,
                    ElPaginationSize,
                    ElPaginationJump,
                ],
                exports: [ElPagination],
                imports: [CommonModule, FormsModule, ElSelectModule],
                entryComponents: [ElPagination]
            }]
    }], function () { return []; }, null); })();
    return ElPaginationModule;
}());
export { ElPaginationModule };
function ElPaginationModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElPaginationModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElPaginationModule.ctorParameters;
}
