// element-angular/__ivy_ngcc__/release/pagination/pagination.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from './children/size';
import * as ɵngcc3 from './children/button';
import * as ɵngcc4 from './children/pager';
import * as ɵngcc5 from './children/jump';

function ElPagination_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 7);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("\u5171 ", ctx_r0.total, " \u6761");
} }
function ElPagination_el_pagination_size_2_Template(rf, ctx) { if (rf & 1) {
    var _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-pagination-size", 8);
    ɵngcc0.ɵɵlistener("modelChange", function ElPagination_el_pagination_size_2_Template_el_pagination_size_modelChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r7); var ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.updatePageSize($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("model", ctx_r1.pageSize)("sizes", ctx_r1.pageSizes);
} }
function ElPagination_el_pagination_btn_3_Template(rf, ctx) { if (rf & 1) {
    var _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-pagination-btn", 9);
    ɵngcc0.ɵɵlistener("next", function ElPagination_el_pagination_btn_3_Template_el_pagination_btn_next_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r9); var ctx_r8 = ɵngcc0.ɵɵnextContext(); return ctx_r8.nextHandle($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("disabled", ctx_r2.model <= 1);
} }
function ElPagination_el_pagination_pager_4_Template(rf, ctx) { if (rf & 1) {
    var _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-pagination-pager", 10);
    ɵngcc0.ɵɵlistener("next", function ElPagination_el_pagination_pager_4_Template_el_pagination_pager_next_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r11); var ctx_r10 = ɵngcc0.ɵɵnextContext(); return ctx_r10.nextHandle($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("current", ctx_r3.model)("count", ctx_r3.pageCount);
} }
function ElPagination_el_pagination_btn_5_Template(rf, ctx) { if (rf & 1) {
    var _r13 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-pagination-btn", 11);
    ɵngcc0.ɵɵlistener("next", function ElPagination_el_pagination_btn_5_Template_el_pagination_btn_next_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r13); var ctx_r12 = ɵngcc0.ɵɵnextContext(); return ctx_r12.nextHandle($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("disabled", ctx_r4.model === ctx_r4.pageCount);
} }
function ElPagination_el_pagination_jump_6_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "el-pagination-jump", 12);
    ɵngcc0.ɵɵlistener("next", function ElPagination_el_pagination_jump_6_Template_el_pagination_jump_next_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.nextHandle($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("model", ctx_r5.model)("max", ctx_r5.pageCount);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component } from '@angular/core';
import { ElPaginationProps } from './pagination.props';
var ElPagination = /** @class */ (function (_super) {
    __extends(ElPagination, _super);
    function ElPagination() {
        var _this = _super.call(this) || this;
        _this.showPager = true;
        _this.showPrev = true;
        _this.showNext = true;
        _this.showTotal = true;
        _this.showSize = true;
        _this.showJumper = true;
        return _this;
    }
    /**
     * @param {?} ElName
     * @param {?} layout
     * @return {?}
     */
    ElPagination.showLayout = function (ElName, layout) {
        var /** @type {?} */ member = layout.find(function (name) { return name === ElName; });
        return !!member;
    };
    /**
     * @param {?} step
     * @return {?}
     */
    ElPagination.prototype.nextHandle = function (step) {
        var /** @type {?} */ nextPage = this.model + step;
        this.model = nextPage < 1 ? 1 : nextPage > this.total ? this.total : nextPage;
        this.modelChange.emit(this.model);
    };
    /**
     * @param {?} nextPageSize
     * @return {?}
     */
    ElPagination.prototype.updatePageSize = function (nextPageSize) {
        this.pageCount = Math.ceil(this.total / nextPageSize);
        this.model = 1;
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElPagination.prototype.ngOnChanges = function (changes) {
        if (!changes)
            return;
        if (changes.total && changes.total.isFirstChange())
            return;
        this.updateLayout();
    };
    /**
     * @return {?}
     */
    ElPagination.prototype.ngOnInit = function () {
        if (!this.pageCount && this.total === undefined) {
            return console.warn('el-pagination required [total]');
        }
        this.updateLayout();
    };
    /**
     * @return {?}
     */
    ElPagination.prototype.updateLayout = function () {
        if (this.total === undefined || this.total === null) {
            this.total = Math.round(this.pageCount * this.pageSize);
        }
        this.pageCount = Math.ceil(this.total / this.pageSize) || 1;
        this.showPager = ElPagination.showLayout('pager', this.layout);
        this.showPrev = ElPagination.showLayout('prev', this.layout);
        this.showNext = ElPagination.showLayout('next', this.layout);
        this.showTotal = ElPagination.showLayout('total', this.layout);
        this.showSize = ElPagination.showLayout('size', this.layout);
        this.showJumper = ElPagination.showLayout('jumper', this.layout);
    };
    /**
     * @nocollapse
     */
    ElPagination.ctorParameters = function () { return []; };
ElPagination.ɵfac = function ElPagination_Factory(t) { return new (t || ElPagination)(); };
ElPagination.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElPagination, selectors: [["el-pagination"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], decls: 7, vars: 8, consts: [[1, "el-pagination", 2, "display", "inline-table"], ["class", "el-pagination__total", 4, "ngIf"], [3, "model", "sizes", "modelChange", 4, "ngIf"], ["dir", "left", 3, "disabled", "next", 4, "ngIf"], [3, "current", "count", "next", 4, "ngIf"], ["dir", "right", 3, "disabled", "next", 4, "ngIf"], [3, "model", "max", "next", 4, "ngIf"], [1, "el-pagination__total"], [3, "model", "sizes", "modelChange"], ["dir", "left", 3, "disabled", "next"], [3, "current", "count", "next"], ["dir", "right", 3, "disabled", "next"], [3, "model", "max", "next"]], template: function ElPagination_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, ElPagination_span_1_Template, 2, 1, "span", 1);
        ɵngcc0.ɵɵtemplate(2, ElPagination_el_pagination_size_2_Template, 1, 2, "el-pagination-size", 2);
        ɵngcc0.ɵɵtemplate(3, ElPagination_el_pagination_btn_3_Template, 1, 1, "el-pagination-btn", 3);
        ɵngcc0.ɵɵtemplate(4, ElPagination_el_pagination_pager_4_Template, 1, 2, "el-pagination-pager", 4);
        ɵngcc0.ɵɵtemplate(5, ElPagination_el_pagination_btn_5_Template, 1, 1, "el-pagination-btn", 5);
        ɵngcc0.ɵɵtemplate(6, ElPagination_el_pagination_jump_6_Template, 1, 2, "el-pagination-jump", 6);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("el-pagination--small", ctx.small);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showTotal);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showSize);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showPrev);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showPager);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showNext);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showJumper);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.ElPaginationSize, ɵngcc3.ElPaginationButton, ɵngcc4.ElPaginationPager, ɵngcc5.ElPaginationJump], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElPagination, [{
        type: Component,
        args: [{
                selector: 'el-pagination',
                template: "\n    <div class=\"el-pagination\"\n      [class.el-pagination--small]=\"small\"\n      style=\"display: inline-table;\">\n      <span class=\"el-pagination__total\" *ngIf=\"showTotal\">\u5171 {{total}} \u6761</span>\n      <el-pagination-size *ngIf=\"showSize\"\n        [model]=\"pageSize\" [sizes]=\"pageSizes\"\n        (modelChange)=\"updatePageSize($event)\">\n      </el-pagination-size>\n      \n      <el-pagination-btn dir=\"left\" (next)=\"nextHandle($event)\"\n        *ngIf=\"showPrev\"\n        [disabled]=\"model <= 1\">\n      </el-pagination-btn>\n      <el-pagination-pager [current]=\"model\" [count]=\"pageCount\"\n        *ngIf=\"showPager\"\n        (next)=\"nextHandle($event)\">\n      </el-pagination-pager>\n      <el-pagination-btn dir=\"right\" (next)=\"nextHandle($event)\"\n        *ngIf=\"showNext\"\n        [disabled]=\"model === pageCount\">\n      </el-pagination-btn>\n\n      <el-pagination-jump *ngIf=\"showJumper\"\n        [model]=\"model\" [max]=\"pageCount\"\n        (next)=\"nextHandle($event)\">\n      </el-pagination-jump>\n    </div>\n  "
            }]
    }], function () { return []; }, null); })();
    return ElPagination;
}(ElPaginationProps));
export { ElPagination };
function ElPagination_tsickle_Closure_declarations() {
    /** @type {?} */
    ElPagination.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElPagination.ctorParameters;
    /** @type {?} */
    ElPagination.prototype.showPager;
    /** @type {?} */
    ElPagination.prototype.showPrev;
    /** @type {?} */
    ElPagination.prototype.showNext;
    /** @type {?} */
    ElPagination.prototype.showTotal;
    /** @type {?} */
    ElPagination.prototype.showSize;
    /** @type {?} */
    ElPagination.prototype.showJumper;
}
