// element-angular/__ivy_ngcc__/release/container/directives/aside.directive.js
import { Directive, Input, Self } from '@angular/core';
import { NgStyle } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
var ElAsideDirective = /** @class */ (function () {
    /**
     * @param {?} ngStyle
     */
    function ElAsideDirective(ngStyle) {
        this.ngStyle = ngStyle;
        this.width = '300px';
    }
    /**
     * @return {?}
     */
    ElAsideDirective.prototype.ngOnChanges = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElAsideDirective.prototype.ngOnInit = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElAsideDirective.prototype.colletClasses = function () {
        this.hostStyles = {
            width: this.width,
        };
        this.ngStyle.ngStyle = this.hostStyles;
        this.ngStyle.ngDoCheck();
    };
    /**
     * @nocollapse
     */
    ElAsideDirective.ctorParameters = function () { return [
        { type: NgStyle, decorators: [{ type: Self },] },
    ]; };
    ElAsideDirective.propDecorators = {
        'width': [{ type: Input },],
    };
ElAsideDirective.ɵfac = function ElAsideDirective_Factory(t) { return new (t || ElAsideDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NgStyle, 2)); };
ElAsideDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElAsideDirective, selectors: [["", "el-aside", ""]], hostAttrs: [1, "el-aside"], inputs: { width: "width" }, features: [ɵngcc0.ɵɵProvidersFeature([NgStyle]), ɵngcc0.ɵɵNgOnChangesFeature] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElAsideDirective, [{
        type: Directive,
        args: [{
                selector: '[el-aside]',
                host: { class: 'el-aside' },
                providers: [NgStyle]
            }]
    }], function () { return [{ type: ɵngcc1.NgStyle, decorators: [{
                type: Self
            }] }]; }, { width: [{
            type: Input
        }] }); })();
    return ElAsideDirective;
}());
export { ElAsideDirective };
function ElAsideDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    ElAsideDirective.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElAsideDirective.ctorParameters;
    /** @type {?} */
    ElAsideDirective.propDecorators;
    /** @type {?} */
    ElAsideDirective.prototype.width;
    /** @type {?} */
    ElAsideDirective.prototype.hostStyles;
    /** @type {?} */
    ElAsideDirective.prototype.ngStyle;
}
