// element-angular/__ivy_ngcc__/release/container/directives/header.directive.js
import { Directive, Input, Self } from '@angular/core';
import { NgStyle } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
var ElHeaderDirective = /** @class */ (function () {
    /**
     * @param {?} ngStyle
     */
    function ElHeaderDirective(ngStyle) {
        this.ngStyle = ngStyle;
        this.height = '60px';
    }
    /**
     * @return {?}
     */
    ElHeaderDirective.prototype.ngOnChanges = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElHeaderDirective.prototype.ngOnInit = function () {
        this.colletClasses();
    };
    /**
     * @return {?}
     */
    ElHeaderDirective.prototype.colletClasses = function () {
        this.hostStyles = {
            height: this.height,
        };
        this.ngStyle.ngStyle = this.hostStyles;
        this.ngStyle.ngDoCheck();
    };
    /**
     * @nocollapse
     */
    ElHeaderDirective.ctorParameters = function () { return [
        { type: NgStyle, decorators: [{ type: Self },] },
    ]; };
    ElHeaderDirective.propDecorators = {
        'height': [{ type: Input },],
    };
ElHeaderDirective.ɵfac = function ElHeaderDirective_Factory(t) { return new (t || ElHeaderDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NgStyle, 2)); };
ElHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ElHeaderDirective, selectors: [["", "el-header", ""]], hostAttrs: [1, "el-header"], inputs: { height: "height" }, features: [ɵngcc0.ɵɵProvidersFeature([NgStyle]), ɵngcc0.ɵɵNgOnChangesFeature] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElHeaderDirective, [{
        type: Directive,
        args: [{
                selector: '[el-header]',
                host: { class: 'el-header' },
                providers: [NgStyle]
            }]
    }], function () { return [{ type: ɵngcc1.NgStyle, decorators: [{
                type: Self
            }] }]; }, { height: [{
            type: Input
        }] }); })();
    return ElHeaderDirective;
}());
export { ElHeaderDirective };
function ElHeaderDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    ElHeaderDirective.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElHeaderDirective.ctorParameters;
    /** @type {?} */
    ElHeaderDirective.propDecorators;
    /** @type {?} */
    ElHeaderDirective.prototype.height;
    /** @type {?} */
    ElHeaderDirective.prototype.hostStyles;
    /** @type {?} */
    ElHeaderDirective.prototype.ngStyle;
}
