// element-angular/__ivy_ngcc__/release/steps/step.js
import { Component, Input, Optional, ElementRef, ViewChild } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { removeNgTag } from '../shared/utils/index';
import { ElSteps } from './steps';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './steps';
import * as ɵngcc2 from '@angular/platform-browser';
import * as ɵngcc3 from '@angular/common';

var _c0 = ["titleRef"];
function ElStep_ng_container_5_i_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "i");
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassMap("el-step__icon-inner el-icon-" + ctx_r4.icon);
} }
function ElStep_ng_container_5_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 8);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r5.index + 1);
} }
function ElStep_ng_container_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, ElStep_ng_container_5_i_1_Template, 1, 2, "i", 3);
    ɵngcc0.ɵɵtemplate(2, ElStep_ng_container_5_div_2_Template, 2, 1, "div", 7);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.icon && !ctx_r0.root.simple);
} }
function ElStep_i_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "i");
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap("el-icon-" + (ctx_r1.currentStatus() === "success" ? "check" : "close") + " el-step__icon-inner is-status");
} }
function ElStep_div_12_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 9);
} }
var _c1 = function (a0, a1) { return { "margin-right": a0, "display": a1 }; };
var ElStep = /** @class */ (function () {
    /**
     * @param {?} root
     * @param {?} el
     * @param {?} sanitizer
     */
    function ElStep(root, el, sanitizer) {
        this.root = root;
        this.el = el;
        this.sanitizer = sanitizer;
        this.index = 1;
        this.mainOffset = '0px';
    }
    /**
     * @return {?}
     */
    ElStep.prototype.currentStatus = function () {
        if (this.root.active > this.index) {
            return this.root.finishStatus;
        }
        if (this.root.active === this.index) {
            return this.root.processStatus;
        }
        return 'wait';
    };
    /**
     * @return {?}
     */
    ElStep.prototype.makeStepStyles = function () {
        var /** @type {?} */ space = this.root.space;
        var /** @type {?} */ width = typeof space === 'number' ? space + "px" : space
            ? String(space) : 100 / (this.root.stepsLength - 1) + "%";
        var /** @type {?} */ lastStyles = this.isLast()
            ? "max-width: " + 100 / this.root.stepsLength + "%;"
            : "margin-right: " + this.root.offset + "px;";
        var /** @type {?} */ styles = "flex-basis: " + width + "; " + lastStyles;
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @return {?}
     */
    ElStep.prototype.makeLineStyles = function () {
        var /** @type {?} */ step = this.index === this.root.active - 1
            ? (this.currentStatus() !== 'error' ? 50 : 0)
            : this.currentStatus() === 'wait' ? 0 : 100;
        var /** @type {?} */ delay = (this.root.processStatus === 'wait' ? -1 : 1) * 150 * this.index + 'ms';
        var /** @type {?} */ key = this.root.direction === 'vertical' ? 'height' : 'width';
        var /** @type {?} */ styles = "border-width: " + (step ? '1px' : 0) + "; " + key + ": " + step + "%; transition-delay: " + delay + ";";
        return this.sanitizer.bypassSecurityTrustStyle(styles);
    };
    /**
     * @return {?}
     */
    ElStep.prototype.isLast = function () {
        return this.root.stepsLength - 1 === this.index;
    };
    /**
     * @return {?}
     */
    ElStep.prototype.isVertical = function () {
        return this.root.direction === 'vertical';
    };
    /**
     * @return {?}
     */
    ElStep.prototype.ngOnInit = function () {
        this.index = +this.el.nativeElement.getAttribute('el-index');
        if (this.root.direction === 'horizontal' && this.root.alignCenter) {
            var /** @type {?} */ width = this.titleRef.nativeElement.getBoundingClientRect().width;
            this.mainOffset = width / 2 + 16 + 'px';
        }
        removeNgTag(this.el.nativeElement);
    };
    /**
     * @nocollapse
     */
    ElStep.ctorParameters = function () { return [
        { type: ElSteps, decorators: [{ type: Optional },] },
        { type: ElementRef, },
        { type: DomSanitizer, },
    ]; };
    ElStep.propDecorators = {
        'titleRef': [{ type: ViewChild, args: ['titleRef',] },],
        'title': [{ type: Input },],
        'description': [{ type: Input },],
        'icon': [{ type: Input },],
        'status': [{ type: Input },],
    };
ElStep.ɵfac = function ElStep_Factory(t) { return new (t || ElStep)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ElSteps, 8), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.DomSanitizer)); };
ElStep.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElStep, selectors: [["el-step"]], viewQuery: function ElStep_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.titleRef = _t.first);
    } }, inputs: { title: "title", description: "description", icon: "icon", status: "status" }, decls: 16, vars: 29, consts: [[1, "el-step__line", 3, "ngStyle"], [1, "el-step__line-inner"], [4, "ngIf"], [3, "class", 4, "ngIf"], [1, "el-step__main"], ["titleRef", ""], ["class", "el-step__arrow", 4, "ngIf"], ["class", "el-step__icon-inner", 4, "ngIf"], [1, "el-step__icon-inner"], [1, "el-step__arrow"]], template: function ElStep_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "div");
        ɵngcc0.ɵɵelementStart(2, "div", 0);
        ɵngcc0.ɵɵelement(3, "i", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "span");
        ɵngcc0.ɵɵtemplate(5, ElStep_ng_container_5_Template, 3, 2, "ng-container", 2);
        ɵngcc0.ɵɵtemplate(6, ElStep_i_6_Template, 1, 2, "i", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelementStart(8, "div", null, 5);
        ɵngcc0.ɵɵelementContainerStart(10);
        ɵngcc0.ɵɵtext(11);
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(12, ElStep_div_12_Template, 1, 0, "div", 6);
        ɵngcc0.ɵɵelementStart(13, "div");
        ɵngcc0.ɵɵelementContainerStart(14);
        ɵngcc0.ɵɵtext(15);
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleMap(ctx.makeStepStyles());
        ɵngcc0.ɵɵclassMap("el-step " + (!ctx.root.simple ? "is-" + ctx.root.direction : ""));
        ɵngcc0.ɵɵclassProp("is-simple", ctx.root.simple)("is-center", ctx.root.alignCenter && !ctx.root.simple && !ctx.isVertical())("is-flex", ctx.isLast() && !ctx.root.alignCenter && !ctx.root.simple);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMap("el-step__head is-" + ctx.currentStatus());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction2(26, _c1, ctx.isLast() ? "" : ctx.root.offset + "px", ctx.isLast() ? "none" : "block"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleMap(ctx.makeLineStyles());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMap("el-step__icon " + (ctx.icon ? "is-icon" : "is-text"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentStatus() !== "success" && ctx.currentStatus() !== "error");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentStatus() === "success" || ctx.currentStatus() === "error");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵclassMap("el-step__title " + "is-" + ctx.currentStatus());
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ctx.title);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.root.simple);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMap("el-step__description " + "is-" + ctx.currentStatus());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.description);
    } }, directives: [ɵngcc3.NgStyle, ɵngcc3.NgIf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElStep, [{
        type: Component,
        args: [{
                selector: 'el-step',
                template: "\n    <div [class]=\"'el-step ' + (!root.simple ? 'is-' + root.direction : '')\"\n      [class.is-simple]=\"root.simple\"\n      [class.is-center]=\"root.alignCenter && !root.simple && !isVertical()\"\n      [class.is-flex]=\"isLast() && !root.alignCenter && !root.simple\"\n      [style]=\"makeStepStyles()\">\n      <div [class]=\"'el-step__head is-' + currentStatus()\">\n        <div class=\"el-step__line\" [ngStyle]=\"{\n          'margin-right': isLast() ? '' : root.offset + 'px',\n          'display': isLast() ? 'none' : 'block'}\">\n          <i class=\"el-step__line-inner\" [style]=\"makeLineStyles()\"></i>\n        </div>\n\n        <span [class]=\"'el-step__icon ' + (icon ? 'is-icon' : 'is-text')\">\n          <ng-container *ngIf=\"currentStatus() !== 'success' && currentStatus() !== 'error'\">\n            <i *ngIf=\"icon\" [class]=\"'el-step__icon-inner el-icon-' + icon\"></i>\n            <div *ngIf=\"!icon && !root.simple\" class=\"el-step__icon-inner\" >{{ index + 1 }}</div>\n          </ng-container>\n          <i *ngIf=\"currentStatus() === 'success' || currentStatus() === 'error'\"\n            [class]=\"'el-icon-' + (currentStatus() === 'success' ? 'check' : 'close') + ' el-step__icon-inner is-status'\">\n          </i>\n        </span>\n      </div>\n      <div class=\"el-step__main\">\n        <div [class]=\"'el-step__title ' + 'is-' + currentStatus()\" #titleRef>\n          <ng-container>{{ title }}</ng-container>\n        </div>\n        <div *ngIf=\"root.simple\" class=\"el-step__arrow\"></div>\n        <div [class]=\"'el-step__description ' + 'is-' + currentStatus()\">\n          <ng-container>{{ description }}</ng-container>\n        </div>\n      </div>\n    </div>\n  "
            }]
    }], function () { return [{ type: ɵngcc1.ElSteps, decorators: [{
                type: Optional
            }] }, { type: ɵngcc0.ElementRef }, { type: ɵngcc2.DomSanitizer }]; }, { titleRef: [{
            type: ViewChild,
            args: ['titleRef']
        }], title: [{
            type: Input
        }], description: [{
            type: Input
        }], icon: [{
            type: Input
        }], status: [{
            type: Input
        }] }); })();
    return ElStep;
}());
export { ElStep };
function ElStep_tsickle_Closure_declarations() {
    /** @type {?} */
    ElStep.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElStep.ctorParameters;
    /** @type {?} */
    ElStep.propDecorators;
    /** @type {?} */
    ElStep.prototype.titleRef;
    /** @type {?} */
    ElStep.prototype.title;
    /** @type {?} */
    ElStep.prototype.description;
    /** @type {?} */
    ElStep.prototype.icon;
    /** @type {?} */
    ElStep.prototype.status;
    /** @type {?} */
    ElStep.prototype.index;
    /** @type {?} */
    ElStep.prototype.mainOffset;
    /** @type {?} */
    ElStep.prototype.root;
    /** @type {?} */
    ElStep.prototype.el;
    /** @type {?} */
    ElStep.prototype.sanitizer;
}
