// element-angular/__ivy_ngcc__/release/table/module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ElTableHeader, ElTableColumn, ElTableBody } from './children/index';
import { ElTable } from './table';
import { ElTableFormat } from './utils/format';
import { ElSharedModule } from '../shared/module';
import * as ɵngcc0 from '@angular/core';
var ElTableModule = /** @class */ (function () {
    function ElTableModule() {
    }
    /**
     * @return {?}
     */
    ElTableModule.forRoot = function () {
        return { ngModule: ElTableModule, providers: [ElTableFormat] };
    };
    /**
     * @nocollapse
     */
    ElTableModule.ctorParameters = function () { return []; };
ElTableModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ElTableModule });
ElTableModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ElTableModule_Factory(t) { return new (t || ElTableModule)(); }, imports: [[CommonModule, ElSharedModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ElTableModule, { declarations: function () { return [ElTable,
        ElTableHeader,
        ElTableColumn,
        ElTableBody]; }, imports: function () { return [CommonModule, ElSharedModule]; }, exports: function () { return [ElTable,
        ElTableColumn]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElTableModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    ElTable,
                    ElTableHeader,
                    ElTableColumn,
                    ElTableBody,
                ],
                exports: [
                    ElTable,
                    ElTableColumn,
                ],
                imports: [CommonModule, ElSharedModule],
                entryComponents: [ElTable]
            }]
    }], function () { return []; }, null); })();
    return ElTableModule;
}());
export { ElTableModule };
function ElTableModule_tsickle_Closure_declarations() {
    /** @type {?} */
    ElTableModule.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElTableModule.ctorParameters;
}
