// element-angular/__ivy_ngcc__/release/date-picker/children/year-table.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function ElYearTable_tr_2_td_1_Template(rf, ctx) { if (rf & 1) {
    var _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 3);
    ɵngcc0.ɵɵlistener("click", function ElYearTable_tr_2_td_1_Template_td_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r5); var year_r3 = ctx.$implicit; var ctx_r4 = ɵngcc0.ɵɵnextContext(2); return ctx_r4.clickHandle(year_r3); });
    ɵngcc0.ɵɵelementStart(1, "a", 4);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var year_r3 = ctx.$implicit;
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassProp("current", year_r3 === ctx_r2.currentYear);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(year_r3);
} }
function ElYearTable_tr_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵtemplate(1, ElYearTable_tr_2_td_1_Template, 3, 3, "td", 2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var years_r1 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", years_r1);
} }
var ElYearTable = /** @class */ (function () {
    function ElYearTable() {
        this.showWeekNumber = false;
        this.modelChange = new EventEmitter();
    }
    /**
     * @param {?} year
     * @return {?}
     */
    ElYearTable.prototype.clickHandle = function (year) {
        this.currentYear = year;
        this.date.setFullYear(year);
        this.modelChange.emit(this.date.getTime());
    };
    /**
     * @param {?} currentYear
     * @return {?}
     */
    ElYearTable.prototype.updateYearRow = function (currentYear) {
        var /** @type {?} */ startYear = ~~(currentYear / 10) * 10;
        return [[], [], []].map(function (v, index) {
            return [0, 1, 2, 3].map(function (num) { return startYear + (index * 4) + num; });
        });
    };
    /**
     * @return {?}
     */
    ElYearTable.prototype.ngOnInit = function () {
        this.date = new Date(this.model);
        this.currentYear = this.date.getFullYear();
        this.yearRows = this.updateYearRow(this.currentYear);
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ElYearTable.prototype.ngOnChanges = function (changes) {
        // not include model
        if (!changes || !changes.model)
            return;
        // first change
        if (changes.model.isFirstChange())
            return;
        this.model = changes.model.currentValue;
        this.date = new Date(this.model);
        this.currentYear = this.date.getFullYear();
        this.yearRows = this.updateYearRow(this.currentYear);
    };
    /**
     * @nocollapse
     */
    ElYearTable.ctorParameters = function () { return []; };
    ElYearTable.propDecorators = {
        'showWeekNumber': [{ type: Input },],
        'model': [{ type: Input },],
        'disabledDate': [{ type: Input, args: ['disabled-date',] },],
        'modelChange': [{ type: Output },],
    };
ElYearTable.ɵfac = function ElYearTable_Factory(t) { return new (t || ElYearTable)(); };
ElYearTable.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElYearTable, selectors: [["el-year-table"]], inputs: { showWeekNumber: "showWeekNumber", model: "model", disabledDate: ["disabled-date", "disabledDate"] }, outputs: { modelChange: "modelChange" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 1, consts: [[1, "el-year-table"], [4, "ngFor", "ngForOf"], ["class", "available", 3, "current", "click", 4, "ngFor", "ngForOf"], [1, "available", 3, "click"], [1, "cell"]], template: function ElYearTable_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "table", 0);
        ɵngcc0.ɵɵelementStart(1, "tbody");
        ɵngcc0.ɵɵtemplate(2, ElYearTable_tr_2_Template, 2, 1, "tr", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.yearRows);
    } }, directives: [ɵngcc1.NgForOf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElYearTable, [{
        type: Component,
        args: [{
                selector: 'el-year-table',
                template: "\n    <table class=\"el-year-table\">\n      <tbody>\n      <tr *ngFor=\"let years of yearRows\">\n        <td class=\"available\" *ngFor=\"let year of years\"\n          [class.current]=\"year === currentYear\"\n          (click)=\"clickHandle(year)\">\n          <a class=\"cell\">{{year}}</a>\n        </td>\n      </tr>\n      </tbody>\n    </table>\n  "
            }]
    }], function () { return []; }, { showWeekNumber: [{
            type: Input
        }], modelChange: [{
            type: Output
        }], model: [{
            type: Input
        }], disabledDate: [{
            type: Input,
            args: ['disabled-date']
        }] }); })();
    return ElYearTable;
}());
export { ElYearTable };
function ElYearTable_tsickle_Closure_declarations() {
    /** @type {?} */
    ElYearTable.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElYearTable.ctorParameters;
    /** @type {?} */
    ElYearTable.propDecorators;
    /** @type {?} */
    ElYearTable.prototype.showWeekNumber;
    /** @type {?} */
    ElYearTable.prototype.model;
    /** @type {?} */
    ElYearTable.prototype.disabledDate;
    /** @type {?} */
    ElYearTable.prototype.modelChange;
    /** @type {?} */
    ElYearTable.prototype.date;
    /** @type {?} */
    ElYearTable.prototype.yearRows;
    /** @type {?} */
    ElYearTable.prototype.currentYear;
}
