// element-angular/__ivy_ngcc__/release/cascader/cascader.js
import * as ɵngcc0 from '@angular/core';

function ElCascader_span_2_ng_container_1_ng_container_1_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1, " / ");
    ɵngcc0.ɵɵelementEnd();
} }
function ElCascader_span_2_ng_container_1_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, ElCascader_span_2_ng_container_1_ng_container_1_span_2_Template, 2, 0, "span", 4);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var item_r4 = ctx.$implicit;
    var i_r5 = ctx.index;
    var ctx_r3 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", item_r4.label, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", i_r5 < ctx_r3.currentLabels.length - 1);
} }
function ElCascader_span_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, ElCascader_span_2_ng_container_1_ng_container_1_Template, 3, 2, "ng-container", 5);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.currentLabels);
} }
function ElCascader_span_2_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r2.currentLabels[ctx_r2.currentLabels.length - 1].label, " ");
} }
function ElCascader_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 3);
    ɵngcc0.ɵɵtemplate(1, ElCascader_span_2_ng_container_1_Template, 2, 1, "ng-container", 4);
    ɵngcc0.ɵɵtemplate(2, ElCascader_span_2_ng_container_2_Template, 2, 1, "ng-container", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.allLevels);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.allLevels);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, forwardRef, Renderer2 } from '@angular/core';
import { ElCascaderPoprs } from './cascader-props';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var ElCascader = /** @class */ (function (_super) {
    __extends(ElCascader, _super);
    /**
     * @param {?} renderer
     */
    function ElCascader(renderer) {
        var _this = _super.call(this) || this;
        _this.renderer = renderer;
        _this.steps = [];
        _this.menuVisible = false;
        _this.inputHover = false;
        _this.currentLabels = [];
        _this.controlChange = function () { };
        _this.controlTouch = function () { };
        return _this;
    }
    /**
     * @return {?}
     */
    ElCascader.prototype.close = function () {
        this.menuVisible = false;
        this.globalListenFunc && this.globalListenFunc();
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ElCascader.prototype.clickHandle = function (event) {
        var _this = this;
        event.stopPropagation();
        if (this.elDisabled)
            return;
        var /** @type {?} */ element = (event.target);
        var /** @type {?} */ isSelfTrigger = ['SPAN', 'I', 'INPUT'].find(function (v) { return v === element.tagName; });
        if (!isSelfTrigger)
            return;
        this.menuVisible = !this.menuVisible;
        if (this.menuVisible) {
            this.globalListenFunc = this.renderer.listen('document', 'click', function () {
                _this.menuVisible = false;
            });
        }
        else {
            this.globalListenFunc && this.globalListenFunc();
        }
    };
    /**
     * @return {?}
     */
    ElCascader.prototype.changeLabels = function () {
        var /** @type {?} */ nextValue = [];
        this.steps.forEach(function (items) {
            var /** @type {?} */ steps = items.filter(function (item) { return item.active; });
            nextValue.push(steps[0]);
        });
        this.currentLabels = nextValue;
        var /** @type {?} */ next = nextValue.map(function (item) { return item.value; });
        this.model = next;
        this.modelChange.emit(next);
        this.controlChange(next);
    };
    /**
     * @param {?=} event
     * @return {?}
     */
    ElCascader.prototype.clearValue = function (event) {
        event && event.stopPropagation();
        this.currentLabels = [];
        var /** @type {?} */ step1 = this.options.map(function (option) {
            return Object.assign(option, {
                active: false,
            });
        });
        this.steps = [step1];
        this.menuVisible = false;
    };
    /**
     * @param {?} event
     * @param {?} step
     * @param {?} index
     * @return {?}
     */
    ElCascader.prototype.selectHandle = function (event, step, index) {
        event.stopPropagation();
        if (this.steps[step][index].elDisabled)
            return;
        this.steps[step] = this.steps[step].map(function (item, i) {
            return Object.assign(item, { active: i === index });
        });
        // reset steps
        this.steps.length = step + 1;
        var /** @type {?} */ next = this.steps[step][index].children;
        // go next
        if (next && Array.isArray(next)) {
            // change on select (props)
            this.changeOnSelect && this.changeLabels();
            var /** @type {?} */ nativeNext = next.map(function (item) { return Object.assign(item, { active: false }); });
            return this.steps.push(nativeNext);
        }
        // last step
        this.changeLabels();
        this.menuVisible = false;
    };
    /**
     * @return {?}
     */
    ElCascader.prototype.showClearIcon = function () {
        return !!(this.clearable && this.inputHover && this.currentLabels.length);
    };
    /**
     * @return {?}
     */
    ElCascader.prototype.ngOnInit = function () {
        this.clearValue();
        if (this.model && this.model.length) {
            var /** @type {?} */ getLabel_1 = function (options, val) {
                var /** @type {?} */ item = options.filter(function (item) { return item.value === val; })[0];
                return { children: item.children, val: item };
            };
            var /** @type {?} */ options_1 = this.options;
            var /** @type {?} */ val = this.model.map(function (v) {
                var _a = getLabel_1(options_1, v), children = _a.children, val = _a.val;
                options_1 = children;
                return val;
            });
            this.currentLabels = val.filter(function (v) { return !!v; });
        }
    };
    /**
     * @return {?}
     */
    ElCascader.prototype.ngOnDestroy = function () {
        this.globalListenFunc && this.globalListenFunc();
    };
    /**
     * @param {?} value
     * @return {?}
     */
    ElCascader.prototype.writeValue = function (value) {
        this.model = value;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElCascader.prototype.registerOnChange = function (fn) {
        this.controlChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElCascader.prototype.registerOnTouched = function (fn) {
        this.controlTouch = fn;
    };
    /**
     * @nocollapse
     */
    ElCascader.ctorParameters = function () { return [
        { type: Renderer2, },
    ]; };
ElCascader.ɵfac = function ElCascader_Factory(t) { return new (t || ElCascader)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2)); };
ElCascader.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ElCascader, selectors: [["el-cascader"]], features: [ɵngcc0.ɵɵProvidersFeature([{
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return ElCascader; }),
                multi: true
            }]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 4, vars: 9, consts: [[3, "click", "mouseenter", "mouseleave"], [3, "readonly", "placeholder", "size", "elDisabled", "icon", "iconClass", "icon-click"], ["class", "el-cascader__label", 4, "ngIf"], [1, "el-cascader__label"], [4, "ngIf"], [4, "ngFor", "ngForOf"]], template: function ElCascader_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "span", 0);
        ɵngcc0.ɵɵlistener("click", function ElCascader_Template_span_click_0_listener($event) { return ctx.clickHandle($event); })("mouseenter", function ElCascader_Template_span_mouseenter_0_listener() { return ctx.inputHover = true; })("mouseleave", function ElCascader_Template_span_mouseleave_0_listener() { return ctx.inputHover = false; });
        ɵngcc0.ɵɵelementStart(1, "el-input", 1);
        ɵngcc0.ɵɵlistener("icon-click", function ElCascader_Template_el_input_icon_click_1_listener($event) { return ctx.clearValue($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(2, ElCascader_span_2_Template, 3, 2, "span", 2);
        ɵngcc0.ɵɵelement(3, "el-cascader-menu");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap("el-cascader " + (ctx.menuVisible ? "is-opened " : "") + (ctx.elDisabled ? "is-disabled " : "") + (ctx.size ? "el-cascader--" + ctx.size : ""));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("readonly", true)("placeholder", ctx.currentLabels.length ? "" : ctx.placeholder)("size", ctx.size)("elDisabled", ctx.elDisabled)("icon", ctx.showClearIcon() ? "circle-close" : "caret-bottom")("iconClass", ctx.showClearIcon() ? "el-cascader__clearIcon" : ctx.menuVisible ? "is-reverse" : "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.currentLabels.length);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ElCascader, [{
        type: Component,
        args: [{
                selector: 'el-cascader',
                providers: [{
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(function () { return ElCascader; }),
                        multi: true
                    }],
                template: "\n    <span [class]=\"'el-cascader ' +  (menuVisible ? 'is-opened ' : '') + (elDisabled ? 'is-disabled ' : '')\n      + (size ? 'el-cascader--' + size : '')\"\n      (click)=\"clickHandle($event)\"\n      (mouseenter)=\"inputHover = true\" (mouseleave)=\"inputHover = false\">\n      <el-input [readonly]=\"true\"\n        [placeholder]=\"currentLabels.length ? '' : placeholder\"\n        [size]=\"size\" [elDisabled]=\"elDisabled\"\n        [icon]=\"showClearIcon() ? 'circle-close' : 'caret-bottom'\"\n        [iconClass]=\"showClearIcon() ? 'el-cascader__clearIcon' : (menuVisible ? 'is-reverse' : '')\"\n        (icon-click)=\"clearValue($event)\">\n      </el-input>\n    \n      <span class=\"el-cascader__label\" *ngIf=\"currentLabels.length\">\n        <ng-container *ngIf=\"allLevels\">\n          <ng-container *ngFor=\"let item of currentLabels; let i = index\">\n            {{ item.label }}\n            <span *ngIf=\"i < currentLabels.length - 1\"> / </span>\n          </ng-container>\n        </ng-container>\n        <ng-container *ngIf=\"!allLevels\">\n          {{ currentLabels[currentLabels.length - 1].label }}\n        </ng-container>\n      </span>\n      <el-cascader-menu></el-cascader-menu>\n  </span>\n  "
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }]; }, null); })();
    return ElCascader;
}(ElCascaderPoprs));
export { ElCascader };
function ElCascader_tsickle_Closure_declarations() {
    /** @type {?} */
    ElCascader.decorators;
    /**
     * @nocollapse
     * @type {?}
     */
    ElCascader.ctorParameters;
    /** @type {?} */
    ElCascader.prototype.steps;
    /** @type {?} */
    ElCascader.prototype.menuVisible;
    /** @type {?} */
    ElCascader.prototype.inputHover;
    /** @type {?} */
    ElCascader.prototype.currentLabels;
    /** @type {?} */
    ElCascader.prototype.globalListenFunc;
    /** @type {?} */
    ElCascader.prototype.controlChange;
    /** @type {?} */
    ElCascader.prototype.controlTouch;
    /** @type {?} */
    ElCascader.prototype.renderer;
}
